--- 1.32/BBS_BSNEF.bbs	2023-05-24 20:54:42.719931568 +0200
+++ 1.33/BBS_BSNEF.bbs	2023-08-23 04:22:58.547844486 +0200
@@ -1,2671 +1,2672 @@
 beginState: s32'dmy' {
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'okasi' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'cmn_okasi'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'bsn_okasi'
 } endState: 
 
 beginState: s32'wtn600cs' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		linkPawnWithAnimation: s32'WTN', s32'WTN600cs'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	spriteEnd: 
 } endState: 
 
 beginState: s32'mnp600cs' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		linkPawnWithAnimation: s32'MNP', s32'MNP_BSN600cs_01'
 	} endUpon: 
 	sprite: s32'null', 450
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_5A_eff01' {
 	upon: (IMMEDIATE) {
 		addXPosition: 140000
 		addYPosition: 250000
 	} endUpon: 
 	sprite: s32'null', 60
 	createParticle: s32'bsn200_eff01', (ZERO)
 } endState: 
 
 beginState: s32'bsn_5A_eff02' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		addXPosition: 250000
 		addYPosition: 225000
 	} endUpon: 
 	sprite: s32'null', 60
 	createParticle: s32'bsn201_eff01', (ZERO)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_Small-A'
 } endState: 
 
 beginState: s32'bsn_5A_eff031' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: 50000
 		addYPosition: 100000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bsn202_eff01'
 } endState: 
 
 beginState: s32'bsn_5A_eff032' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: -40000
 		addYPosition: -30000
 		addZPosition: -50000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bsn202_eff02'
 } endState: 
 
 beginState: s32'bsn_5C_eff_aura01' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'bsn204_Aura01'
 } endState: 
 
 beginState: s32'bsn_5C_eff_aura01_end' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		ignoreFinishStop: 1
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'bsn204_Aura01_end'
 } endState: 
 
 beginState: s32'bsn_5C_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: 140000
 		addYPosition: 325000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bsn204_eff01'
 } endState: 
 
 beginState: s32'bsn_2B_eff00' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: 50000
 		addYPosition: 100000
 	} endUpon: 
 	sprite: s32'null', 15
 	linkParticle: s32'bsn231_eff02'
 } endState: 
 
 beginState: s32'bsn_2B_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 15
 	linkParticle: s32'bsn231_eff01'
 } endState: 
 
 beginState: s32'bsn_2B_eff02' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		ignoreWorldStopByFriend: 1
 		addXPosition: 200000
 		setScale: 500
 		linkParticle: s32'bg_bsn231_smoke'
 	} endUpon: 
 	sprite: s32'null', 10
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 } endState: 
 
 beginState: s32'bsn_2C_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: 160000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bsn232_eff01'
 } endState: 
 
 beginState: s32'bsn_2C_wind01' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		addYPosition: 200000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'cmn_strike00'
 } endState: 
 
 beginState: s32'bsn_5Cair_wind01' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		setAngle: 15000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'cmn_strike00'
 } endState: 
 
 beginState: s32'bsn_5Cair_eff02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: -200000
 		addYPosition: 200000
 		setAngle: 15000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bsn263_eff01'
 	setZPosition: -150000
 } endState: 
 
 beginState: s32'bsn_5Cair_wind02' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		addXPosition: 400000
 		setAngle: 90000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'cmn_strike00'
 } endState: 
 
 beginState: s32'kidan_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectChangeToEliminate: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 		upon: (NO_STOP_IDLING) {
 			moveToPosType: (PARENT), (EX_POINT_0)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 3
 	setZPosition: 0
 	sprite: s32'null', 3
 	setZPosition: 0
 	sprite: s32'null', 3
 	setZPosition: -20000
 	sprite: s32'null', 3
 	setZPosition: 100000
 } endState: 
 
 beginState: s32'kidan_start_aura' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 	} endUpon: 
 	sprite: s32'null', 10
 	setScale: 100
 	setScaleSpeed: 60
 	sprite: s32'null', 300
 	setScale: 700
 	setScaleSpeed: 0
 } endState: 
 
 beginState: s32'kidan_renda_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_renda_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 4
 	setZPosition: -150000
 	addXPosition: 30000
 	sprite: s32'null', 4
 	addXPosition: 120000
 	addYPosition: -20000
 	setZPosition: -100000
 } endState: 
 
 beginState: s32'kidan_renda_start_aura' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 	} endUpon: 
 	sprite: s32'null', 4
 	setScale: 200
 	setScaleSpeed: 100
 	sprite: s32'null', 300
 	setScale: 600
 	setScaleSpeed: 0
 } endState: 
 
 beginState: s32'kidan_renda2_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_renda2_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 4
 	setZPosition: -80000
 	sprite: s32'null', 4
 	setZPosition: 100000
 } endState: 
 
 beginState: s32'kidan_renda2_start_aura' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 	} endUpon: 
 	sprite: s32'null', 4
 	setScale: 200
 	setScaleSpeed: 100
 	sprite: s32'null', 300
 	setScale: 600
 	setScaleSpeed: 0
 } endState: 
 
 beginState: s32'crouchkidan_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'crouchkidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 2
 	setZPosition: -40000
 	sprite: s32'null', 2
 	setZPosition: -130000
 	addXPosition: 10000
 	addYPosition: 10000
 	sprite: s32'null', 4
 	addXPosition: 40000
 	addYPosition: 30000
 	sprite: s32'null', 2
 	addXPosition: -40000
 	addYPosition: -20000
 	setZPosition: -110000
 	sprite: s32'null', 2
 	addXPosition: 70000
 	addYPosition: 70000
 	setZPosition: 70000
 	spriteEnd: 
 } endState: 
 
 beginState: s32'crouchkidan_start_aura' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 		setZPosition: 50000
 	} endUpon: 
 	sprite: s32'null', 10
 	setScale: 100
 	setScaleSpeed: 60
 	sprite: s32'null', 300
 	setScale: 700
 	setScaleSpeed: 0
 } endState: 
 
 beginState: s32'jumpkidan_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PLAYER)
 		createObject: s32'jumpkidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 4
 	setZPosition: -90000
 	sprite: s32'null', 4
 	addXPosition: -20000
 	addYPosition: 60000
 	setZPosition: -70000
 	sprite: s32'null', 4
 	addXPosition: 90000
 	addYPosition: 30000
 	setZPosition: 60000
 	spriteEnd: 
 } endState: 
 
 beginState: s32'jumpkidan_start_aura' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 	} endUpon: 
 	sprite: s32'null', 10
 	setScale: 100
 	setScaleSpeed: 65
 	sprite: s32'null', 300
 	setScale: 750
 	setScaleSpeed: 0
 } endState: 
 
 beginState: s32'bsn_hajiki01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addYPosition: -24000
 	} endUpon: 
 	sprite: s32'null', 30
 	characterSfx: s32'ARC_BTL_BSN_seisei_Chrg_Short'
 	linkParticle: s32'bsn301_eff01'
 } endState: 
 
 beginState: s32'bsn_hajiki01_end' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		addYPosition: -24000
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'bsn301_eff01_end'
 } endState: 
 
 beginState: s32'bsn_kyusyu_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bsn301_eff01'
 } endState: 
 
 beginState: s32'bsn_kidan_charge' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 8
 	linkParticle: s32'bsn205_kidan01_charge'
 	cmnSfx: s32'ARC_BTL_CMN_kidan_Chrg'
 } endState: 
 
 beginState: s32'bsn_kidan_mazzle' {
 	upon: (IMMEDIATE) {
 		setScale: 1200
 	} endUpon: 
 	sprite: s32'null', 100
 	linkParticle: s32'bsn205_kidan01_mazzle'
 } endState: 
 
 beginState: s32'kidan_5D' {
 	upon: (IMMEDIATE) {
 		normalAttack: 
 		callSubroutine: s32'cmnAtkLevel_2_AtkInit'
 		callSubroutine: s32'cmn5DKidan_AtkInit'
 		callSubroutine: s32'cmnKidan_Init'
 		attackDamageEffect: 4, s32'cmn_BomHit'
 		damage: 500
 		airKnockbackEffect: (AIR_FACE_UP)
 		groundKnockbackEffect: (AIR_FACE_UP)
 		attackXKnockback: 45000
 		attackYKnockback: 20000
 		naguri_Naguru: 23
 		xSpeedPerFrame: 106
 		linkParticle: s32'bsn205_kidan01'
 		setActionTimeMax: 30
 		upon: (Z_REVERSAL) {
 			clearRegisteredUponCode: (Z_REVERSAL)
 			linkParticle: s32''
 			createObject: s32'bsn_reversalkidan', (ZERO)
 			copyVar: var(act7), int(1)
 			requestDestroy: (SELF)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'kidan', 1
 	enableHitbox: 
 	setColorAlpha: 0
 	spriteEnd: 
 	sprite: s32'kidan', 30
 	setColorAlpha: 255
 	worldOutToEliminate: 1
 	setSpeedByCurrentAngle: 70000, 0
 	label: s32'Koware'
 	sprite: s32'null', 10
 	resetSpeed: 1
 	linkParticle: s32''
 } endState: 
 
 beginState: s32'kidan_smoke' {
 	upon: (IMMEDIATE) {
 		addXPosition: -27000
 	} endUpon: 
 	sprite: s32'null', 1
 	createParticle: s32'bg_undersmoke00', (ZERO)
 } endState: 
 
 beginState: s32'jumpkidan' {
 	upon: (IMMEDIATE) {
 		normalAttack: 
 		callSubroutine: s32'cmnAtkLevel_2_AtkInit'
 		callSubroutine: s32'cmnAir5DKidan_AtkInit'
 		callSubroutine: s32'cmnKidan_Init'
 		damage: 330
 		guardPushback: 50
 		attackYKnockback: 2000
 		naguri_Naguru: 23
 		attackDamageEffect: 4, s32'cmn_BomHit'
 		setZPosition: 80000
 		linkParticle: s32'bsn205_kidan01'
 		upon: (Z_REVERSAL) {
 			clearRegisteredUponCode: (Z_REVERSAL)
 			linkParticle: s32''
 			createObject: s32'bsn_reversalkidan', (ZERO)
 			copyVar: var(act7), int(1)
 			requestDestroy: (SELF)
 		} endUpon: 
 		upon: (SIGNAL_NAME) {
 			checkSignalName: s32'jumpkidan'
 			if: var(Result) {
 				clearRegisteredUponCode: (SIGNAL_NAME)
 				clearRegisteredUponCode: (IDLING)
 				upon: (ATTACK_COLLISION) {
 					createObject: s32'cmn_kidan_end', (ZERO)
 				} endUpon: 
 			} endIf: 
 		} endUpon: 
 		checkCurrentStateName: (PLAYER), s32'NmlAtkAir5D'
 		if: var(Result) {
 			copyVarFromObject: (SELF), var(Result), (PLAYER), var(act0)
 			ifNot: var(Result) {
 				groundKnockbackEffect: (AIR_FACE_UP)
 				airKnockbackEffect: (AIR_FACE_UP)
 				attackXKnockback: 20000
 				attackYKnockback: 20000
 			} endIf: 
 		} endIf: 
 	} endUpon: 
 	sprite: s32'kidan_nanamesita0', 1
 	enableHitbox: 
 	displayModel: 0
 	if: var(ObjectParameter1) {
 		addXPosition: -40000
 		addYPosition: -27000
 	} endIf: 
 	sprite: s32'kidan_nanamesita', 30
 	if: var(ObjectParameter1) {
 		addXPosition: 40000
 		addYPosition: 27000
 	} endIf: 
 	worldOutToEliminate: 1
 	displayModel: 1
 	setAngle: -34000
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	setSpeedByCurrentAngle: 70000, 0
 	spriteEnd: 
 	label: s32'Koware'
 	sprite: s32'null', 10
 	resetSpeed: 1
 	linkParticle: s32''
 } endState: 
 
 beginState: s32'bsn_reversalkidan' {
 	upon: (IMMEDIATE) {
 		linkCollisionObject: s32'bsn_reversal_kidan'
 		resetSpeed: 1
 	} endUpon: 
 	sprite: s32'null', 600
 	createObject: s32'cmn_reversal_mazzle', (ZERO)
 } endState: 
 
 beginSubroutine: s32'hakaiShot_AtkInit' {
 	specialAttack: 
 	callSubroutine: s32'cmnAtkLevel_2_AtkInit'
 	callSubroutine: s32'cmnSpecialShot_AtkInit'
 	setProjectileDurability: 1
 	damage: 500
 	chipDamagePercent: 20
 	tensionGainOnConnect: 700
 	atkHoseiTadan: 1
 	guardPushback: 70
 	callSubroutine: s32'cmn_AssistShotHosei'
 	groundKnockbackEffect: (AIR_FACE_UP)
 	airKnockbackEffect: (AIR_FACE_UP)
 	attackXKnockback: 8000
 	attackYKnockback: 25000
 	airHitstun: 20
 	attackDamageEffect: 4, s32'cmn_BomHit'
 	atkIgnoreCaution: 1
 	atkShotIgnoreZReversal: 0
 	atkShakeCamera: s32'ShakeTateYoko', 800, 1, 10, 0
 	checkCurrentStateName: (PLAYER), s32'Hakai_seiseiC'
 	if: var(Result) {
 		callSubroutine: s32'cmnSpecialAttackEx_TimeLagShotI'
 	} endIf: 
 	checkCurrentStateName: (PLAYER), s32'AirHakai_seiseiC'
 	if: var(Result) {
 		callSubroutine: s32'cmnSpecialAttackEx_TimeLagShotI'
 	} endIf: 
 	checkCurrentStateName: (PLAYER), s32'AssistAttack'
 	if: var(Result) {
 		airHitstun: 35
 		attackYKnockback: 20000
 		atkHoseiTadan: 0
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			ifVarOfObjectTrue: (PLAYER), var(obj1) {
 				damage: 300
 			} endIf: 
 			copyVarFromObject: (PLAYER), var(obj1), (SELF), int(1)
 		} endUpon: 
 	} endIf: 
 	landingHeight: 50000
 	autoHitSignalSending: 1
 	worldOutToEliminate: 1
 	setWallCollision: 1
 	linkParticle: s32'bsn233_kidan01'
 	damageSoundEffect: s32'dammy'
 	guardSoundEffect: s32'ARC_BTL_CMN_Guard_Beam'
 	modifyVar: (ADD), var(plr0), int(1)
 	upon: (BEFORE_EXIT) {
 		modifyVar: (ADD), var(plr0), int(-1)
 		ifOperation: (IS_LESS), var(plr0), int(1) {
 			stopBankSE: 19, 6
 		} endIf: 
 	} endUpon: 
 	naguri_Nagurareru: 1
 	copyVar: var(act4), int(1)
 	upon: (SIGNAL0) {
 		ifNot: var(CheckStop) {
 			createParticle: s32'bsn402_kidan01_mazzle1', (ZERO)
 			clearRegisteredUponCode: (TIMER)
 			clearRegisteredUponCode: (IDLING)
 			upon: (IDLING) {
 				modifyVar: (ADD), var(act3), int(1)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(act3), int(2) {
 					clearRegisteredUponCode: (IDLING)
 				} endIf: 
 			} endUpon: 
 			naguri_Nagurareru: 0
 			directionPlayer: 
 			screenCollision: 1
 			screenSideCollisionCheck: 1
 			xSpeedPerFrame: 100
 			ySpeedPerFrame: 100
 			setAngleSpeed: -24000
 			resetSpeed: 1
 			copyVar: var(act0), int(1)
 			copyVar: var(obj3), int(1)
 			copyVar: var(act1), int(1)
 			copyVar: var(act2), int(1)
 			copyVar: var(act4), int(1)
 			copyVar: var(act5), int(0)
 			spriteOverrideFrame: s32'hakaikyu2'
 			damage: 700
 			groundKnockbackEffect: (SPIN)
 			airKnockbackEffect: (SPIN)
 			attackXKnockback: 15000
 			attackYKnockback: 33000
 			airHitstun: 35
 			hitstop: 10
 			enemyHitstopAddition: 5, 0, 0
 			atkShakeCamera: s32'ShakeTateYoko', 800, 1, 10, 0
 			checkCurrentStateName: (PLAYER), s32'AssistAttack'
 			if: var(Result) {
 				airHitstun: 35
 				attackXKnockback: 15000
 				atkHoseiTadan: 0
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'CmnActHomingDash'
 			if: var(Result) {
 				setXSpeed: 20000
 				setYSpeed: 25000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk5A'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk5A', (SIGNAL9)
 				setXSpeed: 25000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk5A2nd'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk5A2nd', (SIGNAL9)
 				setYSpeed: 45000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk5A3rd'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk5A3rd', (SIGNAL9)
 				setXSpeed: 25000
 				setYSpeed: -80000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk5B'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk5B', (SIGNAL9)
 				setXSpeed: 70000
 				setYSpeed: -65000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk5C'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk5C', (SIGNAL9)
 				setXSpeed: 110000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk2A'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk2A', (SIGNAL9)
 				setXSpeed: 25000
 				setYSpeed: 20000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk2B'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk2B', (SIGNAL9)
 				setXSpeed: 60000
 				setYSpeed: -40000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk2C'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk2C', (SIGNAL9)
 				setXSpeed: 30000
 				setYSpeed: 100000
 				setGravity: 7000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtk6B'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtk6B', (SIGNAL9)
 				setXSpeed: 45000
 				setYSpeed: -100000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkIsObjectActive: (STACK0)
 			if: var(Result) {
 				ifOperation: (IS_LESS_OR_EQUAL), var(obj1), var(obj2) {
 					if: var(Direction) {
 						setXSpeed: 20000
 					} endIf: 
 					else:  {
 						setXSpeed: -20000
 					} endElse: 
 				} endIf: 
 				else:  {
 					if: var(Direction) {
 						setXSpeed: -20000
 					} endIf: 
 					else:  {
 						setXSpeed: 20000
 					} endElse: 
 				} endElse: 
 				setYSpeed: 25000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtkAir5A'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtkAir5A', (SIGNAL9)
 				setXSpeed: 25000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtkAir5B'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtkAir5B', (SIGNAL9)
 				setXSpeed: 70000
 				setYSpeed: 45000
 				characterSfx: s32'ARC_BTL_BSN_seisei_PushBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtkAir5C'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtkAir5C', (SIGNAL9)
 				setXSpeed: 80000
 				setYSpeed: -50000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'NmlAtkAir2C'
 			if: var(Result) {
 				triggerUponInObject: s32'NmlAtkAir2C', (SIGNAL9)
 				setXSpeed: 80000
 				setYSpeed: 20000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AssistAttack'
 			if: var(Result) {
 				ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(LoopCount), int(40) {
 					setXSpeed: 40000
 					setYSpeed: -70000
 				} endIf: 
 				else:  {
 					setXSpeed: 80000
 					setYSpeed: -25000
 				} endElse: 
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Hakai_seiseiA'
 			if: var(Result) {
 				setXSpeed: 80000
 				setYSpeed: -25000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirHakai_seiseiA'
 			if: var(Result) {
 				setXSpeed: 80000
 				setYSpeed: -25000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Hakai_seiseiB'
 			if: var(Result) {
 				setXSpeed: 30000
 				setYSpeed: -70000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirHakai_seiseiB'
 			if: var(Result) {
 				setXSpeed: 30000
 				setYSpeed: -70000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Hakai_seiseiC'
 			if: var(Result) {
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 				ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(act0), int(8) {
 					setXSpeed: 30000
 					setYSpeed: -70000
 				} endIf: 
 				ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(act0), int(2) {
 					setXSpeed: 80000
 					setYSpeed: -25000
 				} endIf: 
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirHakai_seiseiC'
 			if: var(Result) {
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 				ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(act0), int(8) {
 					setXSpeed: 30000
 					setYSpeed: -70000
 				} endIf: 
 				ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(act0), int(2) {
 					setXSpeed: 80000
 					setYSpeed: -25000
 				} endIf: 
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'MoukouA'
 			if: var(Result) {
 				triggerUponInObject: s32'MoukouA', (SIGNAL0)
 				moveToPosType: (PLAYER), (EX_POINT_0)
 				setXSpeed: 25000
 				setYSpeed: -100000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirMoukouA'
 			if: var(Result) {
 				triggerUponInObject: s32'AirMoukouA', (SIGNAL0)
 				moveToPosType: (PLAYER), (EX_POINT_0)
 				setXSpeed: 25000
 				setYSpeed: -100000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'MoukouB'
 			if: var(Result) {
 				triggerUponInObject: s32'MoukouB', (SIGNAL0)
 				moveToPosType: (PLAYER), (EX_POINT_0)
 				setXSpeed: 60000
 				setYSpeed: -100000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirMoukouB'
 			if: var(Result) {
 				triggerUponInObject: s32'AirMoukouB', (SIGNAL0)
 				moveToPosType: (PLAYER), (EX_POINT_0)
 				setXSpeed: 60000
 				setYSpeed: -100000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'MoukouC'
 			if: var(Result) {
 				triggerUponInObject: s32'MoukouC', (SIGNAL0)
 				moveToPosType: (PLAYER), (EX_POINT_0)
 				setXSpeed: 60000
 				setYSpeed: -100000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirMoukouC'
 			if: var(Result) {
 				triggerUponInObject: s32'AirMoukouC', (SIGNAL0)
 				moveToPosType: (PLAYER), (EX_POINT_0)
 				setXSpeed: 60000
 				setYSpeed: -100000
 				characterSfx: s32'ARC_BTL_BSN_seisei_FireBall'
 			} endIf: 
 		} endIf: 
 	} endUpon: 
 	upon: (SIGNAL9) {
 		clearRegisteredUponCode: (SIGNAL9)
 		requestDestroy: (SELF)
 	} endUpon: 
 	numberOfHits: 1
 	destroyOnPlayerDamage: 1
 	destroyOnPlayerGuard: 1
 	upon: (DESTROY) {
 		clearRegisteredUponCode: (DESTROY)
 		clearRegisteredUponCode: (SIGNAL0)
 		clearRegisteredUponCode: (TIMER)
 		jumpToLabel: s32'end'
 	} endUpon: 
 	setEventTrigger: (TIMER), 300
 	upon: (TIMER) {
 		clearRegisteredUponCode: (DESTROY)
 		clearRegisteredUponCode: (TIMER)
 		jumpToLabel: s32'end'
 	} endUpon: 
 	upon: (ATTACK_COLLISION) {
 		clearRegisteredUponCode: (ATTACK_COLLISION)
 		clearRegisteredUponCode: (DESTROY)
 		clearRegisteredUponCode: (SIGNAL0)
 		clearRegisteredUponCode: (TIMER)
 		jumpToLabel: s32'end'
 		createObject: s32'cmn_kidan_end', (ZERO)
 		cameraShake: s32'ShakeTateYoko', 400, 1, 10, 0
 		characterSfx: s32'ARC_BTL_BSN_seisei_HitWxpl'
 		finishCameraRequest: s32'FinishCamera_hakaiShot'
 		checkCurrentStateName: (PLAYER), s32'NmlAtk2D'
 		if: var(Result) {
 			copyVar: var(act7), int(1)
 		} endIf: 
 	} endUpon: 
 	upon: (Z_REVERSAL) {
 		clearRegisteredUponCode: (Z_REVERSAL)
 		createObject: s32'bsn_reversalkidan', (ZERO)
 		linkParticle: s32''
 		checkCurrentStateName: (PLAYER), s32'NmlAtk2D'
 		if: var(Result) {
 			jumpToLabel: s32'end'
 			copyVar: var(act7), int(2)
 		} endIf: 
 		else:  {
 			deactivateObject: (SELF)
 		} endElse: 
 	} endUpon: 
 	upon: (LANDING_INFINITY) {
 		if: var(act2) {
 			copyVar: var(act2), int(0)
 			percentageYSpeed: -100
 		} endIf: 
 		else:  {
 			deactivateObject: (SELF)
 			createParticle: s32'bg_groundsmokeS', (PLAYER_LAND)
 			createObject: s32'cmn_kidan_end', (ZERO)
 			effectForLandBreak: 0, 1000, 0, 0
 			characterSfx: s32'ARC_BTL_BSN_seisei_HitWxpl'
 		} endElse: 
 	} endUpon: 
 	upon: (SCREEN_COLLISION) {
 		if: var(Direction) {
 			ifOperation: (IS_LESS_OR_EQUAL), var(obj1), var(obj0) {
 				if: var(act3) {
 					copyVar: var(act3), int(0)
 					screenCollision: 0
 					screenSideCollisionCheck: 0
 					percentageXSpeed: -100
 				} endIf: 
 			} endIf: 
 		} endIf: 
 		else:  {
 			ifOperation: (IS_GREATER_OR_EQUAL), var(obj1), var(obj0) {
 				if: var(act3) {
 					copyVar: var(act3), int(0)
 					screenCollision: 0
 					screenSideCollisionCheck: 0
 					percentageXSpeed: -100
 				} endIf: 
 			} endIf: 
 		} endElse: 
 	} endUpon: 
 	upon: (FRAME_STEP) {
 		putObjectIntoStackEx: (SELF), (STACK0), (PLAYER), (STACK7)
 		copyVarFromObject: (SELF), var(obj0), (SELF_MAIN_PLAYER), var(PosX)
 		copyVarFromObject: (SELF), var(obj1), (SELF), var(PosX)
 		copyVarFromObject: (SELF), var(obj2), (STACK0), var(PosX)
 		modifyVar: (ADD), var(act1), int(1)
 		if: var(act0) {
 			ifOperation: (IS_EQUAL), var(act1), int(9) {
 				percentageAngleSpeed: 60
 			} endIf: 
 			ifOperation: (IS_EQUAL), var(act1), int(17) {
 				percentageAngleSpeed: 60
 			} endIf: 
 			ifOperation: (IS_EQUAL), var(act1), int(21) {
 				percentageAngleSpeed: 60
 			} endIf: 
 			ifOperation: (IS_EQUAL), var(act1), int(25) {
 				percentageAngleSpeed: 60
 			} endIf: 
 			ifOperation: (IS_EQUAL), var(act1), int(29) {
 				percentageAngleSpeed: 60
 			} endIf: 
 			ifOperation: (IS_GREATER_OR_EQUAL), var(act1), int(32) {
 				ifOperation: (IS_LESS_OR_EQUAL), var(Angle), int(8000) {
 					setAngleSpeed: 0
 					setAngle: 0
 				} endIf: 
 			} endIf: 
 			ifOperation: (IS_EQUAL), var(act1), int(7) {
 				naguri_Nagurareru: 1
 			} endIf: 
 			ifOperation: (IS_EQUAL), var(act1), int(11) {
 				xSpeedPerFrame: 90
 				ySpeedPerFrame: 90
 				copyVar: var(act4), int(0)
 			} endIf: 
 			ifOperation: (IS_EQUAL), var(act1), int(31) {
 				copyVar: var(obj3), int(0)
 				copyVar: var(act2), int(0)
 				copyVar: var(act3), int(0)
 				screenCollision: 0
 				screenSideCollisionCheck: 0
 				spriteOverrideFrame: s32'hakaikyu'
 				damage: 500
 				groundKnockbackEffect: (AIR_FACE_UP)
 				airKnockbackEffect: (AIR_FACE_UP)
 				attackXKnockback: 5000
 				attackYKnockback: 25000
 				atkShakeCamera: s32'ShakeTateYoko', 800, 1, 10, 0
 				atkAirStunTimeReset: 
 				airHitstun: 20
 				hitstop: 5
 				enemyHitstopAddition: 0, 0, 0
 			} endIf: 
 		} endIf: 
 		else:  {
 			if: var(act5) {
 				ifOperation: (IS_EQUAL), var(act5), int(1) {
 					ifOperation: (IS_EQUAL), var(act1), int(6) {
 						xSpeedPerFrame: 90
 						ySpeedPerFrame: 90
 						copyVar: var(act4), int(0)
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			else:  {
 				ifOperation: (IS_GREATER), var(ObjectParameter1), int(0) {
 					ifOperation: (IS_EQUAL), var(act1), int(22) {
 						xSpeedPerFrame: 90
 						ySpeedPerFrame: 90
 						copyVar: var(act4), int(0)
 					} endIf: 
 				} endIf: 
 			} endElse: 
 		} endElse: 
 		ifNot: var(act6) {
 			checkCurrentStateName: (PLAYER), s32'Hakai_seiseiC'
 			if: var(Result) {
 				ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(act1), int(1) {
 					ifOperation: (IS_EQUAL), var(ObjectParameter1), int(9) {
 						ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(LoopCount), int(33) {
 							naguri_Nagurareru: 1
 							copyVar: var(act6), int(1)
 						} endIf: 
 						else:  {
 							naguri_Nagurareru: 0
 						} endElse: 
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirHakai_seiseiC'
 			if: var(Result) {
 				ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(act1), int(1) {
 					ifOperation: (IS_EQUAL), var(ObjectParameter1), int(9) {
 						ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(LoopCount), int(29) {
 							naguri_Nagurareru: 1
 							copyVar: var(act6), int(1)
 						} endIf: 
 						else:  {
 							naguri_Nagurareru: 0
 						} endElse: 
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AssistAttack'
 			if: var(Result) {
 				ifOperation: (IS_EQUAL), var(ObjectParameter1), int(11) {
 					ifVarOfObjectOperation: (IS_EQUAL), (PLAYER), var(LoopCount), int(41) {
 						naguri_Nagurareru: 1
 						copyVar: var(act6), int(1)
 					} endIf: 
 					else:  {
 						naguri_Nagurareru: 0
 					} endElse: 
 				} endIf: 
 			} endIf: 
 		} endIf: 
 		ifNot: var(act4) {
 			setGravity: 100
 			ifOperation: (IS_GREATER_OR_EQUAL), var(PosY), int(1200000) {
 				setGravity: 3000
 			} endIf: 
 			ifOperation: (IS_GREATER_OR_EQUAL), var(PosY), int(600000) {
 				setGravity: 2000
 			} endIf: 
 			ifOperation: (IS_GREATER_OR_EQUAL), var(PosY), int(400000) {
 				setGravity: 800
 			} endIf: 
 			if: var(act5) {
 				setGravity: 20
 			} endIf: 
 		} endIf: 
 		checkCurrentStateName: (PARENT), s32'AssistAttack'
 		if: var(Result) {
 			setEventTrigger: (TIMER), 50
 		} endIf: 
 		else:  {
 			ifNot: var(obj3) {
 				callSubroutine: s32'cmnNotMainPlayer_ObjectDelete'
 			} endIf: 
 		} endElse: 
 		ifNot: var(IN_BATTLE) {
 			requestDestroy: (SELF)
 		} endIf: 
 	} endUpon: 
 	privateBankSE: 19, s32'ARC_BTL_BSN_seisei_BallFlaot_LP', 100
 } endSubroutine: 
 
 beginState: s32'hakaiShot' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'hakaiShot_AtkInit'
 		ifNot: var(ObjectParameter1) {
 			upon: (IDLING) {
 				if: var(act7) {
 					modifyVar: (ADD), var(act7), int(1)
 				} endIf: 
 				ifOperation: (IS_GREATER), var(act7), int(6) {
 					copyVar: var(act7), int(0)
 					checkCurrentStateName: (PLAYER), s32'NmlAtk2D'
 					if: var(Result) {
 						copyVarFromObject: (PLAYER), var(act7), (PLAYER), int(1)
 					} endIf: 
 				} endIf: 
 				ifOperation: (IS_GREATER), var(SpeedY), int(7000) {
 					percentageGravity: -100
 				} endIf: 
 				ifOperation: (IS_LESS), var(SpeedY), int(-7000) {
 					percentageGravity: -100
 				} endIf: 
 				ifOperation: (IS_GREATER_OR_EQUAL), var(FrameCount), int(2) {
 					addYPosition: 1500
 				} endIf: 
 			} endUpon: 
 		} endIf: 
 	} endUpon: 
 	gotoLabelIfOperation: s32'Crutch', (IS_EQUAL), var(ObjectParameter1), int(0)
 	gotoLabelIfOperation: s32'hakaiA', (IS_EQUAL), var(ObjectParameter1), int(1)
 	gotoLabelIfOperation: s32'hakaiB', (IS_EQUAL), var(ObjectParameter1), int(2)
 	gotoLabelIfOperation: s32'FUpper', (IS_EQUAL), var(ObjectParameter1), int(3)
 	gotoLabelIfOperation: s32'FCenter', (IS_EQUAL), var(ObjectParameter1), int(4)
 	gotoLabelIfOperation: s32'FDown', (IS_EQUAL), var(ObjectParameter1), int(5)
 	gotoLabelIfOperation: s32'BUpper', (IS_EQUAL), var(ObjectParameter1), int(6)
 	gotoLabelIfOperation: s32'BCenter', (IS_EQUAL), var(ObjectParameter1), int(7)
 	gotoLabelIfOperation: s32'BDown', (IS_EQUAL), var(ObjectParameter1), int(8)
 	gotoLabelIfOperation: s32'hakaiC1', (IS_EQUAL), var(ObjectParameter1), int(9)
 	gotoLabelIfOperation: s32'Assist1', (IS_EQUAL), var(ObjectParameter1), int(11)
 	gotoLabelIfOperation: s32'Assist2', (IS_EQUAL), var(ObjectParameter1), int(12)
 	label: s32'Crutch'
 	sprite: s32'hakaikyu', 1
 	setXSpeed: 3500
 	setGravity: -500
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'hakaiA'
 	sprite: s32'hakaikyu', 15
 	setXSpeed: 1000
 	xSpeedPerFrame: 102
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'hakaiB'
 	sprite: s32'hakaikyu', 15
 	setXSpeed: 1500
 	setYSpeed: 2500
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 108
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'FUpper'
 	sprite: s32'hakaikyu', 15
 	copyVar: var(act5), int(1)
 	setXSpeed: 2500
 	setYSpeed: 7000
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 108
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'FCenter'
 	sprite: s32'hakaikyu', 15
 	copyVar: var(act5), int(1)
 	setXSpeed: 3500
 	setYSpeed: 3000
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 108
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'FDown'
 	sprite: s32'hakaikyu', 15
 	copyVar: var(act5), int(1)
 	setXSpeed: 2500
 	xSpeedPerFrame: 102
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'BUpper'
 	sprite: s32'hakaikyu', 15
 	copyVar: var(act5), int(1)
 	setXSpeed: -2500
 	setYSpeed: 7000
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 108
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'BCenter'
 	sprite: s32'hakaikyu', 15
 	copyVar: var(act5), int(1)
 	setXSpeed: -3500
 	setYSpeed: 3000
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 108
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'BDown'
 	sprite: s32'hakaikyu', 15
 	copyVar: var(act5), int(1)
 	setXSpeed: -2500
 	xSpeedPerFrame: 102
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'hakaiC1'
 	sprite: s32'hakaikyu', 15
 	setXSpeed: 8000
 	xSpeedPerFrame: 102
 	enableHitbox: 
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'Assist1'
 	sprite: s32'hakaikyu', 13
 	setXSpeed: 1000
 	xSpeedPerFrame: 102
 	enableHitbox: 
 	sprite: s32'hakaikyu', 2
 	setXSpeed: 80000
 	xSpeedPerFrame: 70
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'Assist2'
 	sprite: s32'hakaikyu', 13
 	setXSpeed: 1000
 	xSpeedPerFrame: 102
 	enableHitbox: 
 	sprite: s32'hakaikyu', 2
 	setYSpeed: 80000
 	ySpeedPerFrame: 80
 	sprite: s32'hakaikyu', 300
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'end'
 	sprite: s32'null', 10
 	clearRegisteredUponCode: (ATTACK_COLLISION)
 	clearRegisteredUponCode: (DESTROY)
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (TIMER)
 	naguri_Nagurareru: 0
 	copyVar: var(obj3), int(0)
 	linkParticle: s32'bsn233_end01'
 	characterSfx: s32'ARC_BTL_BSN_seisei_BallEnd'
 	resetSpeed: 1
 } endState: 
 
 beginState: s32'bsn402_kick_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'bsn402_eff01'
 } endState: 
 
 beginState: s32'bsn405_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectChangeToEliminate: (PARENT)
 		createObject: s32'bsn405_charge_eff01', (ZERO)
 		upon: (NO_STOP_IDLING) {
 			moveToPosType: (PARENT), (EX_POINT_0)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 30
 } endState: 
 
 beginState: s32'bsn405_charge_eff01' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'bsn405_charge01'
 	} endUpon: 
 	sprite: s32'null', 30
 } endState: 
 
 beginState: s32'bsn405_attack_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		setAngle: -60000
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'bsn405_eff01'
 } endState: 
 
 beginState: s32'bsn405_trail_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		setAngle: -60000
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'bsn405_trail01'
 } endState: 
 
 beginState: s32'bsn405_attack_eff02' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bsn405_eff02'
 } endState: 
 
 beginState: s32'bsn303_charge_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 18
 	linkParticle: s32'bsn303_charge01'
 } endState: 
 
 beginState: s32'bsn303_Aura_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectDamageToEliminate: (PLAYER)
 	} endUpon: 
 	sprite: s32'null', 90
 	linkParticle: s32'bsn303_Aura01'
 } endState: 
 
 beginState: s32'bsn303_Aura_end_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectDamageToEliminate: (PLAYER)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'bsn303_Aura01_end'
 } endState: 
 
 beginState: s32'gekirin' {
 	upon: (IMMEDIATE) {
 		specialAttack: 
 		callSubroutine: s32'cmnAtkLevel_2_AtkInit'
 		callSubroutine: s32'cmnSpecialShot_AtkInit'
 		setProjectileDurability: 1
 		damage: 400
 		chipDamagePercent: 20
 		enemyHitstopAddition: -5, 0, 0
 		callSubroutine: s32'cmn_MutekiHosei'
 		groundKnockbackEffect: (AIR_FACE_UP)
 		airKnockbackEffect: (AIR_FACE_UP)
 		attackYKnockback: 30000
 		guardPushback: 20
 		blockstun: 11
 		attackDamageEffect: 4, s32'cmn_BomHit'
 		atkIgnoreCaution: 1
 		atkDamageToRecovery: 1
 		atkShotIgnoreZReversal: 0
 		naguri_Naguru: 23
 		landingHeight: 0
 		autoHitSignalSending: 1
 		linkObjectDamageToEliminate: (NULL)
 		worldOutToEliminate: 1
 		setActionTimeMax: 300
 		linkParticle: s32'bsn303_kidan01'
 		damageSoundEffect: s32'ARC_BTL_CMN_Hit_Kidan'
 		guardSoundEffect: s32'ARC_BTL_CMN_Guard_Kidan'
 		checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 		if: var(Result) {
 			atkDamageToRecovery: 0
 			damage: 200
 			airHitstun: 40
 			attackYKnockback: 45000
 			noHitstunScaling: 1
 			callSubroutine: s32'cmn_AssistHosei'
 			upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 				clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
 				checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act6), (SELF), int(1)
 				} endIf: 
 			} endUpon: 
 		} endIf: 
 		numberOfHits: 1
 		kowareCountDamageCollision: 1
 		destroyOnPlayerDamage: 1
 		destroyOnEnemyDamage: 1
 		destroyOnEnemyGuard: 1
 		destroyOnHit: 1
 		kowareCountSousai: 1
 		upon: (DESTROY) {
 			disableHitbox: 
 			createObject: s32'cmn_kidan_end', (ZERO)
 			jumpToLabel: s32'end'
 		} endUpon: 
 		upon: (AFTER_EXIT) {
 			clearRegisteredUponCode: (AFTER_EXIT)
 			copyVar: var(act0), int(1)
 		} endUpon: 
 		upon: (IDLING) {
 			ifNot: var(act0) {
 				checkCurrentStateName: (PLAYER), s32'Gekirin'
 				if: var(Result) {
 					ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(act1), int(1) {
 						atkShotIgnoreZReversal: 1
 					} endIf: 
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirGekirin'
 				if: var(Result) {
 					ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(act1), int(1) {
 						atkShotIgnoreZReversal: 1
 					} endIf: 
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (TOUCH_GROUND) {
 			createParticle: s32'bg_groundsmokeS', (PLAYER_LAND)
 			effectForLandBreak: 0, 1000, 0, 0
 			requestDestroy: (SELF)
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_PLAYER_GUARD) {
 			ifNot: var(act0) {
 				checkCurrentStateName: (PLAYER), s32'Gekirin'
 				if: var(Result) {
 					sendSignalName: (PLAYER), s32'gekirin', 0
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirGekirin'
 				if: var(Result) {
 					sendSignalName: (PLAYER), s32'gekirin', 0
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_AFTER_FIX_PARAM) {
 			clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_AFTER_FIX_PARAM)
 			ifNot: var(act0) {
 				checkCurrentStateName: (PLAYER), s32'Gekirin'
 				if: var(Result) {
 					sendSignalName: (PLAYER), s32'gekirin_hit', 0
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirGekirin'
 				if: var(Result) {
 					sendSignalName: (PLAYER), s32'gekirin_hit', 0
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (Z_REVERSAL) {
 			clearRegisteredUponCode: (Z_REVERSAL)
 			deactivateObject: (SELF)
 			linkParticle: s32''
 			createObject: s32'bsn_reversalkidan', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'kidan', 30
 	enableHitbox: 
 	upon: (SIGNAL0) {
 		setAngle: 60000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL1) {
 		setAngle: 30000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL2) {
 		setAngle: 0
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL3) {
 		setAngle: -30000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL4) {
 		setAngle: -60000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL5) {
 		setAngle: -120000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL6) {
 		setAngle: -150000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL7) {
 		setAngle: -180000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL8) {
 		setAngle: -210000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	upon: (SIGNAL9) {
 		setAngle: -240000
 		setSpeedByCurrentAngle: 100000, 0
 	} endUpon: 
 	label: s32'end'
 	sprite: s32'null', 10
 	resetSpeed: 1
 	linkParticle: s32''
 } endState: 
 
 beginState: s32'BillsBall_start' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		addXPosition: -60000
 		addZPosition: -90000
 		linkObjectChangeToEliminate: (PARENT)
 		createObject: s32'BillsBall_start_Eff', (ZERO)
 		upon: (NO_STOP_IDLING) {
 			moveToPosType: (PARENT), (EX_POINT_0)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 10
 	addWindPowerOffset: 0, 0, 0, 1000, 0, 170, 30
 	sprite: s32'null', 40
 	sprite: s32'null', 40
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'BillsBall_charge_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		createObject: s32'BillsBall_charge_b_Eff', (ZERO)
 		addYPosition: 270000
 		addZPosition: 70000
 		addXPosition: 40000
 		linkParticle: s32'bsn430_tama_charge00'
 	} endUpon: 
 	sprite: s32'null', 8
 	sprite: s32'null', 4
 	setScale: 350
 	sprite: s32'null', 4
 	addYPosition: -5000
 	sprite: s32'null', 4
 	setScale: 125
 	addYPosition: -5000
 	addXPosition: -5000
 } endState: 
 
 beginState: s32'BillsBall_charge_b_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		ignoreWorldStopByFriend: 1
 		linkParticle: s32'bsn430_tama_charge01'
 	} endUpon: 
 	sprite: s32'null', 8
 	sprite: s32'null', 4
 	sprite: s32'null', 4
 	sprite: s32'null', 4
 } endState: 
 
 beginState: s32'BillsBall_start_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		addYPosition: 200000
 		addZPosition: 50000
 		linkParticle: s32'bsn432_tama_loop01'
 	} endUpon: 
 	sprite: s32'null', 10
 	createObject: s32'BillsBall_start_EffSub', (ZERO)
 	setScale: 100
 	setYSpeed: 4000
 	sprite: s32'null', 40
 	setScaleSpeed: 22
 	sprite: s32'null', 40
 	setScaleSpeed: 0
 	setYSpeed: 0
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'BillsBall_start_EffSub' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		addZPosition: 50000
 		setAngleY: -45000
 		linkParticle: s32'bsn_430_chargeAura'
 	} endUpon: 
 	sprite: s32'null', 1000
 } endState: 
 
 beginSubroutine: s32'BillsBall_AtkInit' {
 	overdriveAttack: 
 	callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 	callSubroutine: s32'cmnUltimateShot_AtkInit'
 	damage: 100
 	minimalDamagePercent: 30
 	hitstop: 0
 	groundKnockbackEffect: (AIR_FACE_UP)
 	airKnockbackEffect: (AIR_FACE_UP)
 	attackYKnockback: 10000
 	slidingKnockdownDuration: 5
 	attackXSuctionBaseObject: (PREVIOUS), (CENTER), 200000, 20000, 100, 60
 	attackYSuctionBaseObject: (PREVIOUS), (CENTER), -250000, 20000, 100, 0
 	atkFrontDirection: 0
 	atkShakeCamera: s32'ShakeTate', 700, 0, 18, 1
 	atkYogore: 0
+	autoHitSignalSending: 1
 	damageSoundEffect: s32'ARC_BTL_CMN_Hit_Beam'
 	guardSoundEffect: s32'ARC_BTL_CMN_Guard_Beam'
 	setActionTimeMax: 300
 	landingHeight: 0
 	linkParticle: s32'bsn432_tama_loop01'
 	upon: (AFTER_EXIT) {
 		clearRegisteredUponCode: (AFTER_EXIT)
 		copyVar: var(act1), int(1)
 	} endUpon: 
 	upon: (DESTROY) {
 		clearRegisteredUponCode: (DESTROY)
 		clearRegisteredUponCode: (ON_ATTACK_COLLISION_MAIN_PLAYER)
 		clearRegisteredUponCode: (ON_ATTACK_COLLISION_SOUSAI)
 		clearRegisteredUponCode: (SIGNAL0)
 		clearRegisteredUponCode: (SIGNAL1)
 		jumpToLabel: s32'koware'
 	} endUpon: 
 	upon: (SIGNAL0) {
 		requestDestroy: (SELF)
 	} endUpon: 
 	upon: (SIGNAL1) {
 		clearRegisteredUponCode: (ON_ATTACK_COLLISION_MAIN_PLAYER)
 		clearRegisteredUponCode: (ON_ATTACK_COLLISION_SOUSAI)
 		clearRegisteredUponCode: (SIGNAL0)
 		clearRegisteredUponCode: (SIGNAL1)
 		jumpToLabel: s32'laser_end'
 	} endUpon: 
 	upon: (SIGNAL2) {
 		copyVar: var(act5), int(1)
 	} endUpon: 
 	upon: (ON_ATTACK_COLLISION_MAIN_PLAYER) {
 		setSpeedByCurrentAngle: 500, 0
 		xSpeedPerFrame: 110
 		ySpeedPerFrame: 110
 		cameraShake: s32'ShakeTate', 600, 0, 70, 10
 		modifyVar: (ADD), var(act4), int(1)
 		ifNot: var(act1) {
 			copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(act4), int(15) {
 			clearRegisteredUponCode: (ON_ATTACK_COLLISION_MAIN_PLAYER)
 			ifNot: var(act5) {
 				linkParticle: s32''
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'BillsBall'
 			if: var(Result) {
 				triggerUponInObject: s32'BillsBall', (SIGNAL0)
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirBillsBallAB'
 			if: var(Result) {
 				triggerUponInObject: s32'AirBillsBallAB', (SIGNAL0)
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirBillsBallCD'
 			if: var(Result) {
 				triggerUponInObject: s32'AirBillsBallCD', (SIGNAL0)
 			} endIf: 
 			jumpToLabel: s32'end'
 		} endIf: 
 	} endUpon: 
 	upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 		callSubroutine: s32'cmnUltimate_CameraCombo'
 	} endUpon: 
 	upon: (ON_ATTACK_COLLISION_SOUSAI) {
 		ifNot: var(act0) {
 			pushXSpeed: 
 			pushYSpeed: 
 		} endIf: 
 		copyVar: var(act2), int(0)
 		upon: (FRAME_STEP_AFTER_EX) {
 			ifNot: var(CheckStop) {
 				modifyVar: (ADD), var(act2), int(1)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(act2), int(5) {
 					clearRegisteredUponCode: (FRAME_STEP_AFTER_EX)
 					resetSpeed: 1
 					popXSpeed: 
 					popYSpeed: 
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		setXSpeed: 30000
 		addXPosition: -150000
 		xSpeedPerFrame: 110
 		enableHitbox: 
 		modifyVar: (ADD), var(act0), int(1)
 		ifOperation: (IS_GREATER_OR_EQUAL), var(act0), int(8) {
 			requestDestroy: (SELF)
 			ifNot: var(act1) {
 				checkCurrentStateName: (PLAYER), s32'BillsBall'
 				if: var(Result) {
 					triggerUponInObject: s32'BillsBall', (SIGNAL0)
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirBillsBallAB'
 				if: var(Result) {
 					triggerUponInObject: s32'AirBillsBallAB', (SIGNAL0)
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirBillsBallCD'
 				if: var(Result) {
 					triggerUponInObject: s32'AirBillsBallCD', (SIGNAL0)
 				} endIf: 
 			} endIf: 
 		} endIf: 
 		else:  {
 			ifNot: var(act1) {
 				checkCurrentStateName: (PLAYER), s32'BillsBall'
 				if: var(Result) {
 					triggerUponInObject: s32'BillsBall', (SIGNAL1)
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirBillsBallAB'
 				if: var(Result) {
 					triggerUponInObject: s32'AirBillsBallAB', (SIGNAL1)
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirBillsBallCD'
 				if: var(Result) {
 					triggerUponInObject: s32'AirBillsBallCD', (SIGNAL1)
 					addXPosition: 70000
 					addYPosition: 80000
 					setSpeedByCurrentAngle: 30000, 0
 				} endIf: 
 			} endIf: 
 		} endElse: 
 	} endUpon: 
 	upon: (TOUCH_GROUND) {
 		createParticle: s32'bg_groundsmokeL', (ZERO)
 		effectForLandBreak: 1, 1000, 0, 0
 		cameraShake: s32'ShakeTate', 1200, 0, 50, 10
 		checkCurrentStateName: (PLAYER), s32'AirBillsBallAB'
 		if: var(Result) {
 			triggerUponInObject: s32'AirBillsBallAB', (SIGNAL0)
 		} endIf: 
 		checkCurrentStateName: (PLAYER), s32'AirBillsBallCD'
 		if: var(Result) {
 			triggerUponInObject: s32'AirBillsBallCD', (SIGNAL0)
 		} endIf: 
 		requestDestroy: (SELF)
 	} endUpon: 
 	upon: (IDLING) {
 		ifVarOfObjectTrue: (PLAYER), var(Direction) {
 			ifVarOfObjectTrue: (ENEMY_MAIN_PLAYER), var(Direction) {
 				disableAttackClashing: 1
 			} endIf: 
 			else:  {
 				disableAttackClashing: 0
 			} endElse: 
 		} endIf: 
 		ifVarOfObjectFalse: (PLAYER), var(Direction) {
 			ifVarOfObjectFalse: (ENEMY_MAIN_PLAYER), var(Direction) {
 				disableAttackClashing: 1
 			} endIf: 
 			else:  {
 				disableAttackClashing: 0
 			} endElse: 
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(LoopCount), int(120) {
 			clearRegisteredUponCode: (IDLING)
 			jumpToLabel: s32'end'
 		} endIf: 
 	} endUpon: 
 } endSubroutine: 
 
 beginState: s32'BillsBall_Shot' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'BillsBall_AtkInit'
 		checkCurrentStateName: (PLAYER), s32'AirBillsBallCD'
 		if: var(Result) {
 			setAngle: -45000
 		} endIf: 
 		upon: (BEFORE_EXIT) {
 			stopBankSE: 0, 20
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'billsdama', 3
 	setSpeedByCurrentAngle: 60000, 0
 	xSpeedPerFrame: 103
 	ySpeedPerFrame: 103
 	enableHitbox: 
 	privateBankSE: 0, s32'ARC_BTL_BSN_BillsBall_Gone', 100
 	label: s32'loop'
 	sprite: s32'billsdama', 3
 	enableHitbox: 
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS), var(LoopCount), int(300)
 	gotoLabel: s32'koware'
 	label: s32'end'
 	sprite: s32'billsdama_explosion', 3
 	clearRegisteredUponCode: (IDLING)
 	disableAttackClashing: 0
 	cameraShake: s32'ShakeTateYoko', 700, 0, 50, 20
 	createParticle: s32'bsn432_bomb', (ZERO)
 	characterSfx: s32'ARC_BTL_BSN_BillsBall_Expl'
 	damage: 1200
 	hitstop: 6
 	enemyHitstopAddition: 0, 9, 9
 	airHitstun: 26
 	groundKnockbackEffect: (KirimomiLand)
 	airKnockbackEffect: (KirimomiLand)
 	attackXKnockback: 5000
 	attackYKnockback: 50000
 	attackXSuctionBaseObject: (NULL), (CENTER), 0, 0, 0, 0
 	attackYSuctionBaseObject: (NULL), (CENTER), 0, 0, 0, 0
 	guardPushback: 150
 	atkShakeCamera: s32'ShakeTateYoko', 1200, 0, 50, 20
 	atkYogore: 1
 	slidingKnockdownDuration: 0
 	ifOperation: (IS_LESS_OR_EQUAL), var(PosY), int(400000) {
 		effectForLandBreak: 1, 1000, 0, 0
 	} endIf: 
 	destructionCollision: 0, 0, 0, 10000
 	enableHitbox: 
 	exitState: 
 	label: s32'laser_end'
 	sprite: s32'null', 4
 	clearRegisteredUponCode: (IDLING)
 	disableAttackClashing: 0
 	createParticle: s32'bsn432_end01', (ZERO)
 	createParticle: s32'cmn_bomb_sp', (ZERO)
 	linkParticle: s32''
 	resetSpeed: 1
 	characterSfx: s32'ARC_BTL_BSN_BillsBall_Expl'
 	sprite: s32'billsdama_explosion2', 3
 	cameraShake: s32'ShakeTateYoko', 1000, 0, 60, 20
 	damage: 2000
 	minimalDamagePercent: 40
-	hitstop: 6
+	hitstop: 16
 	enemyHitstopAddition: 0, 9, 9
 	airHitstun: 26
 	groundKnockbackEffect: (KirimomiLand)
 	airKnockbackEffect: (KirimomiLand)
 	attackXKnockback: 5000
 	attackYKnockback: 70000
 	attackXSuctionBaseObject: (NULL), (CENTER), 0, 0, 0, 0
 	attackYSuctionBaseObject: (NULL), (CENTER), 0, 0, 0, 0
 	guardPushback: 150
 	atkShakeCamera: s32'ShakeTateYoko', 1500, 0, 60, 20
 	attackFinishExplosion: 1, 0
 	slidingKnockdownDuration: 0
 	atkYogore: 1
 	ifOperation: (IS_LESS_OR_EQUAL), var(PosY), int(400000) {
 		effectForLandBreak: 1, 1000, 0, 0
 	} endIf: 
 	destructionCollision: 0, 0, 0, 10000
 	enableHitbox: 
 	exitState: 
 	label: s32'koware'
 	sprite: s32'null', 1
 	createParticle: s32'bsn432_bomb', (ZERO)
 } endState: 
 
 beginState: s32'Blaser_charge' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		setScale: 550
 	} endUpon: 
 	sprite: s32'null', 51
 	setZPosition: 30000
 	linkParticle: s32'bsn432_kidan01_charge'
 } endState: 
 
 beginState: s32'Blaser_mazzle' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		setScale: 1200
 	} endUpon: 
 	sprite: s32'null', 100
 	linkParticle: s32'bsn432_kidan_mazzle'
 } endState: 
 
 beginSubroutine: s32'Blaser_AtkInit' {
 	linkParticle: s32'bsn432_kidan01'
 	ignoreWorldStopByAll: 1
 	worldOutToEliminate: 1
 	landingHeight: 0
 	setActionTimeMax: 300
 	numberOfHits: 1
 	destroyOnPlayerDamage: 1
 	destroyOnPlayerGuard: 1
 	upon: (DESTROY) {
 		clearRegisteredUponCode: (IDLING)
 		clearRegisteredUponCode: (DESTROY)
 		jumpToLabel: s32'koware'
 	} endUpon: 
 	upon: (IDLING) {
 		if: var(act0) {
 			calcDistance: (SELF), (CENTER), (STACK0), (CENTER)
 			modifyVarIntoResult: (IS_LESS), var(Result), int(500000)
 			if: var(Result) {
 				requestDestroy: (SELF)
 				triggerUponInObject: s32'BillsBall_Shot', (SIGNAL1)
 				checkCurrentStateName: (PLAYER), s32'BillsBall'
 				if: var(Result) {
 					triggerUponInObject: s32'BillsBall', (SIGNAL2)
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirBillsBallAB'
 				if: var(Result) {
 					triggerUponInObject: s32'AirBillsBallAB', (SIGNAL2)
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'AirBillsBallCD'
 				if: var(Result) {
 					triggerUponInObject: s32'AirBillsBallCD', (SIGNAL2)
 				} endIf: 
 			} endIf: 
 		} endIf: 
 		else:  {
 			checkCurrentStateName: (PLAYER), s32'BillsBall'
 			if: var(Result) {
 				copyVar: var(act0), int(1)
 				putObjectIntoStackEx: (SELF), (STACK0), (PLAYER), (STACK0)
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirBillsBallAB'
 			if: var(Result) {
 				copyVar: var(act0), int(1)
 				putObjectIntoStackEx: (SELF), (STACK0), (PLAYER), (STACK0)
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AirBillsBallCD'
 			if: var(Result) {
 				copyVar: var(act0), int(1)
 				putObjectIntoStackEx: (SELF), (STACK0), (PLAYER), (STACK0)
 			} endIf: 
 		} endElse: 
 	} endUpon: 
 } endSubroutine: 
 
 beginState: s32'Blaser' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Blaser_AtkInit'
 	} endUpon: 
 	sprite: s32'billsdama_laser', 60
 	setSpeedByCurrentAngle: 90000, 0
 	xSpeedPerFrame: 105
 	ySpeedPerFrame: 105
 	label: s32'koware'
 	sprite: s32'null', 10
 	resetSpeed: 1
 	createObject: s32'cmn_kidan_end', (ZERO)
 	linkParticle: s32''
 } endState: 
 
 beginState: s32'AirBlaser' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Blaser_AtkInit'
 	} endUpon: 
 	sprite: s32'billsdama_laser', 60
 	setAngle: -45000
 	setSpeedByCurrentAngle: 90000, 0
 	xSpeedPerFrame: 105
 	ySpeedPerFrame: 105
 	label: s32'koware'
 	sprite: s32'null', 10
 	resetSpeed: 1
 	createObject: s32'cmn_kidan_end', (ZERO)
 	linkParticle: s32''
 } endState: 
 
 beginState: s32'bsn_432_hakai_WindEff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		ignoreWorldStopByFriend: 1
 		linkParticle: s32'bsn432_wind'
 	} endUpon: 
 	sprite: s32'null', 120
 } endState: 
 
 beginState: s32'bsn_432_hakai_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		addXPosition: 200000
 		ignoreWorldStopByFriend: 1
 		linkParticle: s32'bsn_432_tame'
 		jumpToStateInObject: (SELF), s32'Finish'
 	} endUpon: 
 	sprite: s32'null', 300
 	exitState: 
 	label: s32'Finish'
 	sprite: s32'null', 240
 	linkObjectChangeToEliminate: (NULL)
 	linkObjectPosition: (NULL)
 	linkParticle: s32'bsn_432_end'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_432_hakai_CS_BodyAura' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		zLine: int(4)
 		linkParticle: s32'bsn_432_csAura'
 		ignoreWorldStopByAll: 1
 	} endUpon: 
 	sprite: s32'null', 300
 	exitState: 
 	label: s32'Finish'
 	sprite: s32'null', 100
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_432_dkpn_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: 180000
 		addYPosition: 320000
 		ignoreWorldStopByAll: 1
 		createParticle: s32'bsn_432_Dkpn01', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 240
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_432_Smoke_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		addXPosition: 180000
 		ignoreWorldStopByFriend: 1
 		createParticleBG: s32'bg_bsn432_smoke00', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 240
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_432_Smoke01_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		addXPosition: 150000
 		ignoreWorldStopByFriend: 1
 		createParticleBG: s32'bg_bsn432_smoke01', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 240
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_432_FadeIn' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByAll: 1
 		jumpToStateInObject: (SELF), s32'End'
 		zLine: var(plr1)
 	} endUpon: 
 	sprite: s32'null', 300
 	linkParticle: s32'bsn_432_FadeIn'
 	exitState: 
 	label: s32'End'
 	sprite: s32'null', 30
 	linkParticle: s32'bsn_432_FadeOut'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_600cs_gradation_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		ignoreWorldStopByFriend: 1
 		createParticleWithArg: s32'bsn_600cs_gradation', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 240
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_600_cut_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		ignoreWorldStopByFriend: 1
 		linkParticleArg: s32'bsn_600cs_Beam'
 	} endUpon: 
 	sprite: s32'null', 120
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_600_Down_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		ignoreWorldStopByFriend: 1
 		linkParticleArg: s32'bsn_600cs_Down'
 	} endUpon: 
 	sprite: s32'null', 240
 	spriteEnd: 
 } endState: 
 
 beginState: s32'bsn_600_Down_Smoke_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		ignoreWorldStopByFriend: 1
 		createParticle: s32'bg_bsn600cs_Smoke', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 240
 	spriteEnd: 
 } endState: 
 
 beginState: s32'BSN809_GKS' {
 	upon: (IMMEDIATE) {
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkObjectPosition: (PARENT)
 		linkModelEnemyTeam: s32'GKS', s32'GKS_BSN809cs'
 		linkObjectChangeToEliminate: (PARENT)
 		zLine: int(5)
 		beginMeshSet: s32'Normal'
 		addToMeshSet: s32'Normal'
 		endMeshSet: 
 		jumpToStateInObject: (SELF), s32'BreakRock'
 		jumpToStateInObject: (ATTACKER), s32'BreakRock2'
 		jumpToStateInObject: (DEFENDER), s32'SSGAura'
 		jumpToStateInObject: 26, s32'AuraDele'
 		jumpToStateInObject: (PLAYER_1), s32'Ray'
 	} endUpon: 
 	sprite: s32'null', 97
 	sprite: s32'null', 10
 	createObject: s32'BSN809_KaijoAura', (HEAD_3D)
 	sprite: s32'null', 2147483647
 	swapMeshSet: s32'Normal'
 	label: s32'BreakRock'
 	sprite: s32'null', 2147483647
 	createObject: s32'BSN809_BreakRock', (CHEST_3D)
 	label: s32'BreakRock2'
 	sprite: s32'null', 2147483647
 	deactivateObjectByName: s32'bsn809_DeathBallFix'
 	deactivateObjectByName: s32'BSN809_BreakRock'
 	createObject: s32'BSN809_BreakRock2', (CHEST_3D)
 	createObject: s32'bsn809_DeathBall_Fix2', (CHEST_3D)
 	label: s32'SSGAura'
 	sprite: s32'null', 2147483647
 	createObject: s32'bsn809_SsgAura', (CHEST_3D)
 	label: s32'AuraDele'
 	sprite: s32'null', 2147483647
 	deactivateObjectByName: s32'bsn809_SsgAura'
 	label: s32'Ray'
 	sprite: s32'null', 2147483647
 	createObject: s32'bsn809_Ray', (HEAD_3D)
 	applyTo: (PREVIOUS) {
 		addZPosition: -150000
 		addYPosition: 550000
 		addXPosition: -600000
 		setScale: 2000
 	} endApply: 
 	createObject: s32'BSN809_Earth_Fall', (ZERO)
 } endState: 
 
 beginState: s32'BSN809_YMN' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkPawnWithAnimation: s32'YMN', s32'YMN_BSN809cs'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	createObject: s32'BSN809_CharaBG', (CHEST_3D)
 } endState: 
 
 beginState: s32'BSN809_VGS' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkPawnWithAnimation: s32'VGS', s32'VGS_BSN809cs'
 		beginMeshSet: s32'Normal'
 		addToMeshSet: s32'Normal'
 		endMeshSet: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	swapMeshSet: s32'Normal'
 } endState: 
 
 beginState: s32'BSN809_GON' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkPawnWithAnimation: s32'GON', s32'GON_BSN809cs'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_KRN' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkPawnWithAnimation: s32'KRN', s32'KRN_BSN809cs'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_TNN' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkPawnWithAnimation: s32'TNN', s32'TNN_BSN809cs'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_GHU' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkPawnWithAnimation: s32'GHU', s32'GHU_BSN809cs'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_TKN' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkPawnWithAnimation: s32'TKN', s32'TKN_BSN809cs'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_PCN' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkPawnWithAnimation: s32'PCN', s32'PCN_BSN809cs'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_Manp01_01' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'MNP', s32'MNP_BSN809cs_01'
 		zLine: int(5)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_Manp01_02' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'MNP', s32'MNP_BSN809cs_02'
 		zLine: int(5)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_CharaBG' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_ZfightersBG'
 		faceLeft: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_KaijoAura' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_ChangeAura'
 		addYPosition: 40000
 		setScale: 500
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'BSN809_Earth' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkParticle: s32'bsn_809_earth'
 		addYPosition: -12000000
 		addXPosition: 24000000
 		setScale: 4400
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_Space' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkParticle: s32'bsn_809_space'
 		setScale: 10000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_Earth2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkParticle: s32'bsn_809_earth2'
 		addYPosition: -160000000
 		addXPosition: -6000000
 		addZPosition: -150000000
 		setScale: 90000
 		addZScale: 80000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_Earth3' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkParticle: s32'bsn_809_earth3'
 		addYPosition: -190000000
 		addZPosition: -300000000
 		setScale: 90000
 		addXScale: 30000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_Earth_Fall' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkParticle: s32'bsn_809_earth2'
 		addYPosition: -5000000
 		addXPosition: 7000000
 		setAngleY: 95000
 		setAngle: 40000
 		setScale: 2000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'BSN809_smoke' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_Smoke'
 		addXPosition: 1200000
 		setScale: 750
 	} endUpon: 
 	sprite: s32'null', 90
 } endState: 
 
 beginState: s32'BSN809_smokeYoin' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_yoinsmoke'
 		addXPosition: 900000
 	} endUpon: 
 	sprite: s32'null', 90
 	deactivateObjectByName: s32'BSN809_smoke'
 } endState: 
 
 beginState: s32'BSN809_BreakRock' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_BreakRock'
 		setScale: 12500
 		setAngleY: 180000
 		addXPosition: 15000
 		addZPosition: -50000
 	} endUpon: 
 	sprite: s32'null', 90
 } endState: 
 
 beginState: s32'BSN809_BreakRock2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_BreakRock2'
 		setScale: 2500
 		setAngle: 40000
 	} endUpon: 
 	sprite: s32'null', 90
 } endState: 
 
 beginState: s32'bsn809_DeathBallCamera' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_DeathBall_cameraMove'
 		addYPosition: 2000000
 		addXPosition: -850000
 		addZPosition: -50000
 		setScale: 2600
 	} endUpon: 
 	sprite: s32'null', 4
 	sprite: s32'null', 4
 	sprite: s32'null', 4
 	sprite: s32'null', 4
 	sprite: s32'null', 4
 	sprite: s32'null', 4
 	sprite: s32'null', 4
 	setScale: 2500
 	addYPosition: -125000
 	addXPosition: 125000
 	addZPosition: -50000
 	sprite: s32'null', 4
 	addYPosition: -125000
 	addXPosition: 125000
 	setScale: 2000
 } endState: 
 
 beginState: s32'bsn809_DeathBallTame' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_DeathBallTame'
 		setScale: 150
 		addZPosition: 66000
 		addXPosition: 35000
 		addYPosition: -16000
 		jumpToStateInObject: (SELF), s32'Move'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	label: s32'Move'
 	sprite: s32'null', 8
 	addYPosition: -20000
 } endState: 
 
 beginState: s32'bsn809_DeathBallFix' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_DeathBall_Fix'
 		jumpToStateInObject: (SELF), s32'Move'
 		jumpToStateInObject: (ATTACKER), s32'Move2'
 		setScale: 500
 		addYPosition: 100000
 		addXPosition: 50000
 		addZPosition: -25000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	label: s32'Move'
 	sprite: s32'null', 4
 	addYPosition: 300000
 	setScale: 2000
 	sprite: s32'null', 4
 	addYPosition: 1800000
 	addXPosition: -1000000
 	setScale: 17000
 	setScaleSpeed: 30
 	label: s32'BikuBiku'
 	sprite: s32'null', 2
 	addScale: 300
 	sprite: s32'null', 2
 	addScale: -300
 	gotoLabel: s32'BikuBiku'
 	label: s32'Move2'
 	sprite: s32'null', 85
 	setScaleSpeed: 0
 	setScale: 15000
 	addYPosition: 250000
 	linkParticle: s32'bsn_809_DeathBall_Fall'
 } endState: 
 
 beginState: s32'bsn809_WhiteOut' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_WhiteOut'
 	} endUpon: 
 	sprite: s32'null', 300
 } endState: 
 
 beginState: s32'bsn809_DeathBall_Fix2' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'bsn_809_DeathBall_Fall'
 		addXPosition: -1000000
 		addYPosition: 1000000
 		setScale: 10000
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'bsn809_SsgAura' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'bsn_809_SsgAura'
 		setAngle: 460000
 		addXPosition: 300000
 	} endUpon: 
 	sprite: s32'null', 60
 	setScaleSpeed: 30
 } endState: 
 
 beginState: s32'bsn809_DeathBall_Crack' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_DeathBall_Crack'
 		setScale: 15000
 		setAngle: 45000
 		addYPosition: 2000000
 		addXPosition: -2000000
 		addZPosition: -1000000
 		setScaleSpeed: 50
 	} endUpon: 
 	sprite: s32'null', 30
 	label: s32'Loop'
 	sprite: s32'null', 2
 	addScale: 350
 	sprite: s32'null', 2
 	addScale: -350
 	gotoLabel: s32'Loop'
 } endState: 
 
 beginState: s32'bsn809_DeathBall_Bomb' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_DeathBall_Bomb'
 		setScale: 102000
 		addZPosition: -30000000
 		addXPosition: 100000000
 		addYPosition: 20000000
 	} endUpon: 
 	sprite: s32'null', 240
 } endState: 
 
 beginState: s32'bsn809_Ray' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'bsn_809_Ray'
 	} endUpon: 
 	sprite: s32'null', 240
 	setYSpeed: 750
 } endState: 
 
 beginState: s32'bsn_610_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	createParticleWithArg: s32'bsn600_Aura01', (ZERO)
 } endState: 
 
 beginState: s32'bsn_610_eff01_Loop' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 30
 	createObject: s32'bsn_610_eff01', (PLAYER_LAND)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'bsn_610_eff02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		addZPosition: -150000
 	} endUpon: 
 	sprite: s32'null', 60
 	createParticleWithArg: s32'bsn600_Aura01', (ZERO)
 } endState: 
 
 beginState: s32'bsn_610_eff01_Loop2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 30
 	createObject: s32'bsn_610_eff02', (PLAYER_LAND)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'FinishCamera_NmlAtk2C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cameraMotion: -131409, 90096, 19062, 5064, 10096, 0, 92, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_Moukou' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cameraMotion: -263385, 223326, 268301, -4940, 9632, 0, 54, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_Gekirin' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	moveCameraEx: (ENEMY_MAIN_PLAYER), -152353, 442351, 109291, 37, 3436, 0, 54, 0, 90, 30, 0, 0, 2
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_hakaiShot' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	moveCameraEx: (ENEMY_MAIN_PLAYER), -152353, 442351, 109291, 37, 3436, 0, 54, 0, 90, 30, 0, 0, 2
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 	beginState: s32'DUMMYDUMMY' {
 	sprite: s32'null', 1
 } endState: 
 
