--- 1.32/BBS_GHUEF.bbs	2023-05-24 20:54:45.841983062 +0200
+++ 1.33/BBS_GHUEF.bbs	2023-08-23 04:23:01.382892797 +0200
@@ -1,3259 +1,3260 @@
 beginState: s32'dmy' {
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'okasi' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'cmn_okasi'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_okasi'
 } endState: 
 
 beginState: s32'ghu_5CEff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'ghu_262EFF10'
 } endState: 
 
 beginState: s32'ghu_2CEff' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		setAngle: 62500
 		addYPosition: 350000
 		addXPosition: 125000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'cmn_strike00'
 } endState: 
 
 beginState: s32'ghu_hajiki' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'ghu_hajiki'
 } endState: 
 
 beginState: s32'kidan_5D' {
 	upon: (IMMEDIATE) {
 		normalAttack: 
 		callSubroutine: s32'cmnAtkLevel_3_AtkInit'
 		callSubroutine: s32'cmn5DKidan_AtkInit'
 		callSubroutine: s32'cmnKidan_Init'
 		linkObjectPosition: (PLAYER)
 		damage: 800
 		hitstop: 12
 		atkAirStunTimeReset: 
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 60000
 		attackYKnockback: 20000
 		guardPushback: 120
 		blockstun: 14
 		atkShotIgnoreZReversal: 1
 		wallBounceCount: 0
 		attackFinishBlowback: 1
 		atkHitCommonSEReset: s32'ARC_BTL_CMN_Hit_Large-A'
 		guardSoundEffect: s32'ARC_BTL_CMN_Guard_Beam'
 		upon: (ATTACK_COLLISION) {
 			copyVar: var(act7), int(1)
 		} endUpon: 
 		upon: (Z_REVERSAL) {
 			clearRegisteredUponCode: (Z_REVERSAL)
 			copyVar: var(act7), int(1)
 			createObject: s32'cmn_reversal_mazzle', (ATTACK)
 			requestDestroy: (SELF)
 		} endUpon: 
 		upon: (IDLING) {
 			if: var(act7) {
 				modifyVar: (ADD), var(act7), int(1)
 			} endIf: 
 			ifOperation: (IS_GREATER), var(act7), int(12) {
 				copyVar: var(act7), int(0)
 				checkCurrentStateName: (PLAYER), s32'NmlAtk5D'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act7), (PLAYER), int(1)
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'kiaihou0', 1
 	enableHitbox: 
 	sprite: s32'kiaihou', 3
 	attackWhiffCrouch: 1
 	spriteEnd: 
 	label: s32'Koware'
 	sprite: s32'null', 15
 	resetSpeed: 1
 	linkParticle: s32''
 } endState: 
 
 beginState: s32'kidan_smoke_gorup' {
 	upon: (IMMEDIATE) {
 	} endUpon: 
 	sprite: s32'null', 1
 	createObject: s32'kidan_smoke', (ZERO)
 	createObject: s32'kidan_smoke2', (ZERO)
 } endState: 
 
 beginState: s32'kidan_smoke' {
 	upon: (IMMEDIATE) {
 		addXPosition: -200000
 	} endUpon: 
 	sprite: s32'null', 1
 	createParticle: s32'bg_undersmoke00', (ZERO)
 } endState: 
 
 beginState: s32'kidan_smoke2' {
 	upon: (IMMEDIATE) {
 		addXPosition: -100000
 		addZPosition: 100000
 		setScale: 800
 	} endUpon: 
 	sprite: s32'null', 1
 	createParticle: s32'bg_undersmoke00', (ZERO)
 } endState: 
 
 beginState: s32'kidan_smoke3' {
 	upon: (IMMEDIATE) {
 		addXPosition: 200000
 		addZPosition: 100000
 	} endUpon: 
 	sprite: s32'null', 1
 	createParticle: s32'bg_undersmoke00', (ZERO)
 } endState: 
 
 beginState: s32'crouchkidan_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'crouchkidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 3
 	cmnSfx: s32'ARC_BTL_CMN_kidan_Chrg'
 	setZPosition: -90000
 	sprite: s32'null', 3
 	sprite: s32'null', 2
 	spriteEnd: 
 } endState: 
 
 beginState: s32'crouchkidan_start_aura' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 	} endUpon: 
 	sprite: s32'null', 5
 	setScale: 100
 	setScaleSpeed: 120
 	sprite: s32'null', 600
 	setScale: 700
 	setScaleSpeed: 0
 } endState: 
 
 beginState: s32'crouchkidan' {
 	upon: (IMMEDIATE) {
 		normalAttack: 
 		callSubroutine: s32'cmnAtkLevel_2_AtkInit'
 		callSubroutine: s32'cmnAir5DKidan_AtkInit'
 		callSubroutine: s32'cmnKidan_Init'
 		damage: 600
 		groundKnockbackEffect: (AIR_FACE_UP)
 		airKnockbackEffect: (AIR_FACE_UP)
 		attackXKnockback: 8000
 		attackYKnockback: 30000
 		airHitstun: 28
 		attackDamageEffect: 4, s32'cmn_BomHit'
 		setAngle: -25000
 		xSpeedPerFrame: 102
 		ySpeedPerFrame: 102
 		setZPosition: 80000
 		landingHeight: 0
 		linkParticle: s32'cmn_kidan'
 	} endUpon: 
 	sprite: s32'kidan_nanamesita', 1
 	enableHitbox: 
 	setColorAlpha: 0
 	sprite: s32'kidan_nanamesita', 2147483647
 	worldOutToEliminate: 1
 	setColorAlpha: 255
 	setSpeedByCurrentAngle: 70000, 0
 	spriteEnd: 
 	label: s32'Koware'
 	sprite: s32'null', 10
 	resetSpeed: 1
 	linkParticle: s32''
 } endState: 
 
 beginState: s32'jumpkidan_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'crouchkidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 1
 	setZPosition: -90000
 	sprite: s32'null', 2
 	cmnSfx: s32'ARC_BTL_CMN_kidan_Chrg'
 	sprite: s32'null', 1
 	sprite: s32'null', 1
 	spriteEnd: 
 } endState: 
 
 beginState: s32'jumpkidan_start_aura' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 	} endUpon: 
 	sprite: s32'null', 8
 	setScale: 100
 	setScaleSpeed: 75
 	sprite: s32'null', 600
 	setScale: 700
 	setScaleSpeed: 0
 } endState: 
 
 beginState: s32'jumpkidan' {
 	upon: (IMMEDIATE) {
 		normalAttack: 
 		callSubroutine: s32'cmnAtkLevel_2_AtkInit'
 		callSubroutine: s32'cmnAir5DKidan_AtkInit'
 		callSubroutine: s32'cmnKidan_Init'
 		damage: 600
 		groundKnockbackEffect: (AIR_FACE_UP)
 		airKnockbackEffect: (AIR_FACE_UP)
 		attackXKnockback: 8000
 		attackYKnockback: 30000
 		airHitstun: 24
 		attackDamageEffect: 4, s32'cmn_BomHit'
 		setAngle: -25000
 		xSpeedPerFrame: 102
 		ySpeedPerFrame: 102
 		setZPosition: 80000
 		landingHeight: 0
 		linkParticle: s32'cmn_kidan'
 	} endUpon: 
 	sprite: s32'kidan_nanamesita', 1
 	enableHitbox: 
 	setColorAlpha: 0
 	sprite: s32'kidan_nanamesita', 2147483647
 	worldOutToEliminate: 1
 	setColorAlpha: 255
 	setSpeedByCurrentAngle: 70000, 0
 	spriteEnd: 
 	label: s32'Koware'
 	sprite: s32'null', 10
 	resetSpeed: 1
 	linkParticle: s32''
 } endState: 
 
 beginState: s32'Masenko_Tame' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_402_tame00'
 		upon: (NO_STOP_IDLING) {
 			moveToPosType: (PARENT), (EX_POINT_0)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 3
 	characterSfx: s32'ARC_BTL_GHU_Masenko_Chrg'
 	setZPosition: 90000
 	setScale: 250
 	addZPosition: -40000
 	sprite: s32'null', 3
 	setScale: 350
 	addZPosition: -10000
 	sprite: s32'null', 3
 	setScale: 500
 	addZPosition: -10000
 	sprite: s32'null', 2
 	addZPosition: 40000
 } endState: 
 
 beginState: s32'Masenko_Tame_Assist' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_402_tame00'
 		upon: (NO_STOP_IDLING) {
 			moveToPosType: (PARENT), (EX_POINT_0)
 			copyVarFromParent: var(PosXRaw), var(PosXRaw)
 			copyVarFromParent: var(PosZ), var(PosZ)
 			copyVarFromObject: (SELF), var(act0), (PLAYER), var(PlayerArmorHit)
 			ifOperation: (IS_GREATER_OR_EQUAL), var(PlayerArmorHit), int(0) {
 				modifyVar: (DIV), var(act0), int(700000)
 				modifyVar: (ADD), var(act0), int(-1)
 				ifOperation: (IS_LESS_OR_EQUAL), var(PlayerArmorHit), int(700000) {
 					modifyVar: (MUL), var(act0), int(30000)
 					modifyVar: (SUB), var(PosXRaw), var(act0)
 				} endIf: 
 				else:  {
 					modifyVar: (MUL), var(act0), int(30000)
 					modifyVar: (SUB), var(PosXRaw), var(act0)
 				} endElse: 
 			} endIf: 
 			else:  {
 				addXPosition: 70000
 			} endElse: 
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 3
 	characterSfx: s32'ARC_BTL_GHU_Masenko_Chrg'
 	setZPosition: 90000
 	setScale: 250
 	addZPosition: -40000
 	sprite: s32'null', 3
 	setScale: 350
 	addZPosition: -10000
 	sprite: s32'null', 3
 	setScale: 500
 	addZPosition: -10000
 	sprite: s32'null', 2
 	addZPosition: 40000
 } endState: 
 
 beginState: s32'Masenko_Core' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PLAYER)
 		linkObjectPosition: (PLAYER)
 		setActionTimeMax: 300
 		linkParticle: s32'ghu_402_core00'
 		privateBankSE: 0, s32'ARC_BTL_GHU_Masenko_Fire', 100
 		jumpToStateInObject: (SELF), s32'end'
 		jumpToStateInObject: 47, s32'end'
 		upon: (PLAYER_DAMAGE) {
 			linkObjectPosition: (NULL)
 			linkObjectStop: (NULL)
 			ignoreWorldStopByAll: 1
 		} endUpon: 
 		upon: (SIGNAL1) {
 			clearRegisteredUponCode: (SIGNAL1)
 			copyVar: var(act0), int(0)
 			triggerUponInObject: s32'Masenko_Atk', (SIGNAL6)
 			checkCurrentStateName: (PLAYER), s32'Masenko'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				sendSignal: (PLAYER), (SIGNAL2)
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				sendSignal: (PLAYER), (SIGNAL2)
 			} endIf: 
 			copyVar: var(act2), int(1)
 			copyVar: var(act6), int(1)
 			upon: (SIGNAL1) {
 				modifyVar: (ADD), var(act2), int(1)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(act2), int(4) {
 					checkCurrentStateName: (PLAYER), s32'Masenko'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL1)
 					} endIf: 
 					checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL1)
 					} endIf: 
 				} endIf: 
 				else:  {
 					checkCurrentStateName: (PLAYER), s32'Masenko'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL2)
 					} endIf: 
 				} endElse: 
 			} endUpon: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifNot: var(act1) {
 				sendSignalActionNameOldest: s32'Masenko_Atk', (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	modifyVar: (ADD), var(act3), int(1)
 	setArg_objectParameters: var(act3), var(act6)
 	createObjectWithArg: s32'Masenko_Atk', (ZERO)
 	ifOperation: (IS_GREATER_OR_EQUAL), var(act0), int(3) {
 		copyVarFromObject: (PREVIOUS), var(act0), (SELF), int(1)
 	} endIf: 
 	sprite: s32'null', 2
 	spriteEnd: 
 	modifyVar: (ADD), var(act0), int(1)
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(act0), int(5)
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	clearRegisteredUponCode: (AFTER_EXIT)
 	clearRegisteredUponCode: (IDLING)
 	copyVar: var(act4), int(1)
 	createObject: s32'Masenko_Atk', (ZERO)
 	triggerUponInObject: s32'Masenko_Atk', (SIGNAL7)
 	checkCurrentStateName: (PLAYER), s32'Masenko'
 	if: var(Result) {
 		sendSignal: (PLAYER), (SIGNAL1)
 	} endIf: 
 	checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 	if: var(Result) {
 		sendSignal: (PLAYER), (SIGNAL1)
 	} endIf: 
 	linkObjectPosition: (NULL)
 	linkObjectStop: (NULL)
 	ignoreWorldStopByFriend: 1
 	sprite: s32'null', 13
 	deactivateObjectByName: s32'Masenko_Atk'
 	linkParticle: s32'ghu_402_core01'
 	sprite: s32'null', 35
 	stopBankSE: 0, 35
 } endState: 
 
 beginState: s32'Masenko_Atk' {
 	upon: (IMMEDIATE) {
 		specialAttack: 
 		callSubroutine: s32'cmnAtkLevel_3_AtkInit'
 		callSubroutine: s32'cmnSpecialShot_AtkInit'
 		damage: 262
 		tensionGainOnConnect: 240
 		hitstop: 2
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 60000
 		attackYKnockback: 13000
 		attackYSuction: 1, 15000, 300000, 100, 0
 		wallBounceCount: 0
 		airHitstun: 35
 		atkRange: 50000000, -385000, 0, 0
 		atkSousaiSevereCheck: 1
 		atkEffectBeamSousai: 1
 		atkIgnoreSousaiSameLevel: 1
 		atkHitCommonBankSE: 16, s32'ARC_BTL_CMN_Hit_BeamLoop_LP', 0
 		atkGuardCommonBankSE: 17, s32'ARC_BTL_CMN_Grd_BeamLoop_LP', 0
 		atkShakeCamera: s32'ShakeYoko', 600, 0, 20, 10
 		upon: (SIGNAL7) {
 			attackXKnockback: 120000
 			attackYSuction: 0, 0, 0, 0, 0
 			wallBounceCount: 1
 			wallBounceXVelocityPercent: 0
 			checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 			if: var(Result) {
 				attackXKnockback: 20000
 				attackYKnockback: 45000
 				airHitstun: 35
 				groundKnockbackEffect: (SPIN)
 				airKnockbackEffect: (SPIN)
 				wallBounceCount: 0
 			} endIf: 
 			ifVarOfObjectFalse: (PLAYER), var(PlayerDamage) {
 				enableHitbox: 
 			} endIf: 
 			else:  {
 				recoveryState: 
 			} endElse: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PARENT), var(act2), int(4) {
 				recoveryState: 
 			} endIf: 
 		} endUpon: 
 		checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 		if: var(Result) {
 			damage: 200
 			noHitstunScaling: 1
 			callSubroutine: s32'cmn_AssistHosei'
 			upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_AFTER_FIX_PARAM) {
 				clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_AFTER_FIX_PARAM)
 				checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act6), (SELF), int(1)
 				} endIf: 
 			} endUpon: 
 		} endIf: 
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectAngle: 2
 		setActionTimeMax: 60
 		autoHitSignalSending: 1
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu_402_naka00'
 		upon: (SIGNAL0) {
 			clearRegisteredUponCode: (SIGNAL0)
 			createObject: s32'Masenko_End', (ZERO)
 			copyVar: var(act1), int(1)
 			copyVarFromObject: (PARENT), var(act1), (SELF), int(1)
 		} endUpon: 
 		upon: (ON_ATTACK_COLLISION_SOUSAI) {
 			copyVarFromObject: (PARENT), var(ObjectParameter1), (SELF), var(ObjectParameter1)
 			deactivateObject: (SELF)
 			sendSignal: (PARENT), (SIGNAL1)
 		} endUpon: 
 		upon: (SIGNAL6) {
 			copyVar: var(act0), int(0)
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			if: var(ObjectParameter2) {
 				checkCurrentStateName: (PLAYER), s32'Masenko'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (ATTACK_COLLISION) {
 			if: var(act0) {
 				sendSignal: (PARENT), (SIGNAL0)
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Masenko'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act1), (SELF), int(1)
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifVarOfObjectTrue: (PARENT), var(act2) {
 				ifVarOfObjectFalse: (PARENT), var(act4) {
 					copyVarFromObject: (SELF), var(Result), (PARENT), var(ObjectParameter1)
 					ifOperation: (IS_LESS_OR_EQUAL), var(ObjectParameter1), var(Result) {
 						deactivateObject: (SELF)
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			ifOperation: (IS_LESS_OR_EQUAL), var(PosY), int(400000) {
 				effectForLandBreak: 2, 1000, 0, 0
 			} endIf: 
 			copyVarFromObject: (SELF), var(Result), (ENEMY), var(PlayerDamageOrGuard)
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Masenko'
 			ifNot: var(Result) {
 				checkCurrentStateName: (PLAYER), s32'AssistAttack3'
 				ifNot: var(Result) {
 					stopBankSE: 16, 20
 					stopBankSE: 17, 20
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			if: var(act1) {
 				copyVar: var(act1), int(0)
 				copyVarFromObject: (PARENT), var(act1), (SELF), int(0)
 			} endIf: 
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu_402_naka01', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'masenko', 1
 	enableHitbox: 
 	setSpeedByCurrentAngle: 100000, 0
 	sprite: s32'masenko', 1
 	atkIgnoreSousaiSameLevel: 0
 	label: s32'loop'
 	sprite: s32'masenko', 2
 	scaleForLinkY: 1200
 	sprite: s32'masenko', 2
 	scaleForLinkY: 1000
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'Masenko_End' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		setActionTimeMax: 60
 		moveToPosType: (PARENT), (EX_POINT_0)
 		copyVarFromParent: var(PosZ), var(PosZ)
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu_402_end00'
 		upon: (IDLING) {
 			checkIsObjectActive: (PARENT)
 			if: var(Result) {
 				copyVarFromParent: var(PosZ), var(PosZ)
 				moveToPosType: (PARENT), (EX_POINT_0)
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu_402_end01', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	scaleForLinkY: 1000
 	sprite: s32'null', 2
 	scaleForLinkY: 1200
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'R_Masenko_Core' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PLAYER)
 		linkObjectPosition: (PLAYER)
 		setActionTimeMax: 300
 		linkParticle: s32'ghu_402_core00'
 		privateBankSE: 0, s32'ARC_BTL_GHU_Masenko_Fire', 100
 		jumpToStateInObject: (SELF), s32'end'
 		jumpToStateInObject: 47, s32'end'
 		upon: (PLAYER_DAMAGE) {
 			linkObjectPosition: (NULL)
 			linkObjectStop: (NULL)
 			ignoreWorldStopByAll: 1
 		} endUpon: 
 		upon: (SIGNAL1) {
 			clearRegisteredUponCode: (SIGNAL1)
 			copyVar: var(act0), int(0)
 			triggerUponInObject: s32'R_Masenko_Atk', (SIGNAL6)
 			checkCurrentStateName: (PLAYER), s32'Masenko'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				sendSignal: (PLAYER), (SIGNAL2)
 			} endIf: 
 			copyVar: var(act2), int(1)
 			copyVar: var(act6), int(1)
 			upon: (SIGNAL1) {
 				modifyVar: (ADD), var(act2), int(1)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(act2), int(4) {
 					checkCurrentStateName: (PLAYER), s32'Masenko'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL1)
 					} endIf: 
 				} endIf: 
 				else:  {
 					checkCurrentStateName: (PLAYER), s32'Masenko'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL2)
 					} endIf: 
 				} endElse: 
 			} endUpon: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifNot: var(act1) {
 				sendSignalActionNameOldest: s32'R_Masenko_Atk', (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	modifyVar: (ADD), var(act3), int(1)
 	setArg_objectParameters: var(act3), var(act6)
 	createObjectWithArg: s32'R_Masenko_Atk', (ZERO)
 	ifOperation: (IS_GREATER_OR_EQUAL), var(act0), int(5) {
 		copyVarFromObject: (PREVIOUS), var(act0), (SELF), int(1)
 	} endIf: 
 	sprite: s32'null', 2
 	spriteEnd: 
 	modifyVar: (ADD), var(act0), int(1)
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(act0), int(7)
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	clearRegisteredUponCode: (AFTER_EXIT)
 	clearRegisteredUponCode: (IDLING)
 	copyVar: var(act4), int(1)
 	createObject: s32'R_Masenko_Atk', (ZERO)
 	triggerUponInObject: s32'R_Masenko_Atk', (SIGNAL7)
 	checkCurrentStateName: (PLAYER), s32'Masenko'
 	if: var(Result) {
 		sendSignal: (PLAYER), (SIGNAL1)
 	} endIf: 
 	linkObjectPosition: (NULL)
 	linkObjectStop: (NULL)
 	ignoreWorldStopByFriend: 1
 	sprite: s32'null', 13
 	deactivateObjectByName: s32'R_Masenko_Atk'
 	linkParticle: s32'ghu_402_core01'
 	sprite: s32'null', 35
 	stopBankSE: 0, 35
 } endState: 
 
 beginState: s32'R_Masenko_Atk' {
 	upon: (IMMEDIATE) {
 		specialAttack: 
 		callSubroutine: s32'cmnAtkLevel_3_AtkInit'
 		callSubroutine: s32'cmnSpecialShot_AtkInit'
 		damage: 262
 		tensionGainOnConnect: 240
 		hitstop: 2
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 60000
 		attackYKnockback: 10000
 		attackYSuction: 1, 15000, 300000, 100, 0
 		wallBounceCount: 0
 		airHitstun: 35
 		atkRange: 50000000, -385000, 0, 0
 		atkSousaiSevereCheck: 1
 		atkEffectBeamSousai: 1
 		atkIgnoreSousaiSameLevel: 1
 		atkHitCommonBankSE: 16, s32'ARC_BTL_CMN_Hit_BeamLoop_LP', 0
 		atkGuardCommonBankSE: 17, s32'ARC_BTL_CMN_Grd_BeamLoop_LP', 0
 		atkShakeCamera: s32'ShakeYoko', 1000, 1, 20, 0
 		upon: (SIGNAL7) {
 			attackXKnockback: 120000
 			attackYSuction: 0, 0, 0, 0, 0
 			wallBounceCount: 1
 			wallBounceXVelocityPercent: 0
 			ifVarOfObjectFalse: (PLAYER), var(PlayerDamage) {
 				enableHitbox: 
 			} endIf: 
 			else:  {
 				recoveryState: 
 			} endElse: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PARENT), var(act2), int(4) {
 				recoveryState: 
 			} endIf: 
 		} endUpon: 
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectAngle: 2
 		setActionTimeMax: 60
 		autoHitSignalSending: 1
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu_402_naka00'
 		upon: (SIGNAL0) {
 			clearRegisteredUponCode: (SIGNAL0)
 			createObject: s32'R_Masenko_End', (ZERO)
 			copyVar: var(act1), int(1)
 			copyVarFromObject: (PARENT), var(act1), (SELF), int(1)
 		} endUpon: 
 		upon: (ON_ATTACK_COLLISION_SOUSAI) {
 			copyVarFromObject: (PARENT), var(ObjectParameter1), (SELF), var(ObjectParameter1)
 			deactivateObject: (SELF)
 			sendSignal: (PARENT), (SIGNAL1)
 		} endUpon: 
 		upon: (SIGNAL6) {
 			copyVar: var(act0), int(0)
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			if: var(ObjectParameter2) {
 				checkCurrentStateName: (PLAYER), s32'Masenko'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (ATTACK_COLLISION) {
 			if: var(act0) {
 				sendSignal: (PARENT), (SIGNAL0)
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Masenko'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act1), (SELF), int(1)
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifVarOfObjectTrue: (PARENT), var(act2) {
 				ifVarOfObjectFalse: (PARENT), var(act4) {
 					copyVarFromObject: (SELF), var(Result), (PARENT), var(ObjectParameter1)
 					ifOperation: (IS_LESS_OR_EQUAL), var(ObjectParameter1), var(Result) {
 						deactivateObject: (SELF)
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			ifOperation: (IS_LESS_OR_EQUAL), var(PosY), int(400000) {
 				effectForLandBreak: 2, 1000, 0, 0
 			} endIf: 
 			copyVarFromObject: (SELF), var(Result), (ENEMY), var(PlayerDamageOrGuard)
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Masenko'
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			if: var(act1) {
 				copyVar: var(act1), int(0)
 				copyVarFromObject: (PARENT), var(act1), (SELF), int(0)
 			} endIf: 
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu_402_naka01', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'masenko', 1
 	enableHitbox: 
 	setSpeedByCurrentAngle: 100000, 0
 	sprite: s32'masenko', 1
 	atkIgnoreSousaiSameLevel: 0
 	label: s32'loop'
 	sprite: s32'masenko', 2
 	scaleForLinkY: 1200
 	sprite: s32'masenko', 2
 	scaleForLinkY: 1000
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'R_Masenko_End' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		setActionTimeMax: 60
 		moveToPosType: (PARENT), (EX_POINT_0)
 		copyVarFromParent: var(PosZ), var(PosZ)
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu_402_end00'
 		upon: (IDLING) {
 			checkIsObjectActive: (PARENT)
 			if: var(Result) {
 				copyVarFromParent: var(PosZ), var(PosZ)
 				moveToPosType: (PARENT), (EX_POINT_0)
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu_402_end01', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	scaleForLinkY: 1000
 	sprite: s32'null', 2
 	scaleForLinkY: 1200
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'Release' {
 	upon: (IMMEDIATE) {
 		overdriveAttack: 
 		callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 		callSubroutine: s32'cmnUltimateShot_AtkInit'
 		damage: 1645
 		minimalDamagePercent: 40
 		hitstop: 9
 		enemyHitstopAddition: 0, 6, 6
 		groundKnockbackEffect: (SPIN)
 		airKnockbackEffect: (SPIN)
 		attackXKnockback: 60000
 		attackYKnockback: 38000
-		airHitstun: 34
+		airHitstun: 40
 		autoHitStopSending: 1
 		atkYogore: 0
+		autoHitSignalSending: 1
 		ifOperation: (IS_EQUAL), var(ObjectParameter1), int(1) {
 			damage: 2043
 			minimalDamagePercent: 42
 		} endIf: 
 		ifOperation: (IS_EQUAL), var(ObjectParameter1), int(2) {
 			damage: 2300
 			minimalDamagePercent: 46
 		} endIf: 
 		ifOperation: (IS_EQUAL), var(ObjectParameter1), int(3) {
 			damage: 2568
 			minimalDamagePercent: 49
 		} endIf: 
 		ifOperation: (IS_EQUAL), var(ObjectParameter1), int(4) {
 			damage: 2804
 			minimalDamagePercent: 52
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(5) {
 			damage: 3015
 			minimalDamagePercent: 55
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(6) {
 			damage: 3204
 			minimalDamagePercent: 58
 		} endIf: 
 		damageSoundEffect: s32'ARC_BTL_CMN_Hit_Large-A'
 		guardSoundEffect: s32'ARC_BTL_CMN_Guard_Large'
 		linkObjectDamageToEliminate: (PLAYER)
 		linkObjectPosition: (PLAYER)
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
 			callSubroutine: s32'cmnUltimate_CameraCombo'
 		} endUpon: 
 		setActionTimeMax: 60
 	} endUpon: 
 	sprite: s32'release_shock', 3
 	enableHitbox: 
 	sprite: s32'release_shock', 1
 	addXScale: 200
 	addYScale: 100
 	sprite: s32'release_shock', 1
 	addXScale: 200
 	addYScale: 100
 	sprite: s32'release_shock', 1
 	addXScale: 150
 	addYScale: 100
 	sprite: s32'release_shock', 1
 	addXScale: 150
 	addYScale: 100
 	sprite: s32'release_shock', 1
 	addXScale: 150
 	addYScale: 100
 	sprite: s32'release_shock', 1
 	addXScale: 150
 	addYScale: 100
 	sprite: s32'release_shock', 1
 	addXScale: 150
 	addYScale: 200
 } endState: 
 
 beginSubroutine: s32'U_Kamehameha_AtkInit' {
 	damage: 190
 	minimalDamagePercent: 37
 	damageScalingOverride: 500
 	guardPushback: 10
 } endSubroutine: 
 
 beginState: s32'U_Kamehameha_Tame' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'ghu_kmhm_tame'
 		ignoreWorldStopByFriend: 1
 		addXPosition: 50000
 	} endUpon: 
 	sprite: s32'null', 3
 	setScale: 250
 	sprite: s32'null', 15
 	setScale: 300
 	addXPosition: -15000
 	addYPosition: -5000
 	spriteEnd: 
 } endState: 
 
 beginState: s32'U_Kamehameha_Core' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PLAYER)
 		linkObjectPosition: (PLAYER)
 		setActionTimeMax: 300
 		addXPosition: 50000
 		linkParticle: s32'ghu431c_core01'
 		setScale: 700
 		jumpToStateInObject: (SELF), s32'end'
 		jumpToStateInObject: 47, s32'end'
 		upon: (PLAYER_DAMAGE) {
 			linkObjectPosition: (NULL)
 			linkObjectStop: (NULL)
 			ignoreWorldStopByAll: 1
 		} endUpon: 
 		upon: (SIGNAL1) {
 			clearRegisteredUponCode: (SIGNAL1)
 			copyVar: var(act0), int(0)
 			triggerUponInObject: s32'U_Kamehameha_Atk', (SIGNAL6)
 			checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				sendSignal: (PLAYER), (SIGNAL2)
 			} endIf: 
 			copyVar: var(act2), int(1)
 			copyVar: var(act6), int(1)
 			upon: (SIGNAL1) {
 				modifyVar: (ADD), var(act2), int(1)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(act2), int(16) {
 					checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL1)
 					} endIf: 
 				} endIf: 
 				else:  {
 					checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL2)
 					} endIf: 
 				} endElse: 
 			} endUpon: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifNot: var(act1) {
 				sendSignalActionNameOldest: s32'U_Kamehameha_Atk', (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	modifyVar: (ADD), var(act3), int(1)
 	setArg_objectParameters: var(act3), var(act6)
 	createObjectWithArg: s32'U_Kamehameha_Atk', (ZERO)
 	ifOperation: (IS_LESS_OR_EQUAL), var(act0), int(12) {
 		applyTo: (PREVIOUS) {
 			attackFinishBeam: 1, 0
 		} endApply: 
 	} endIf: 
 	ifOperation: (IS_GREATER_OR_EQUAL), var(act0), int(22) {
 		copyVarFromObject: (PREVIOUS), var(act0), (SELF), int(1)
 	} endIf: 
 	sprite: s32'null', 2
 	spriteEnd: 
 	modifyVar: (ADD), var(act0), int(1)
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(act0), int(24)
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	clearRegisteredUponCode: (AFTER_EXIT)
 	clearRegisteredUponCode: (IDLING)
 	copyVar: var(act4), int(1)
 	createObject: s32'U_Kamehameha_Atk', (ZERO)
 	triggerUponInObject: s32'U_Kamehameha_Atk', (SIGNAL7)
 	checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 	if: var(Result) {
 		sendSignal: (PLAYER), (SIGNAL1)
 	} endIf: 
 	addXPosition: 150000
 	linkObjectPosition: (NULL)
 	linkObjectStop: (NULL)
 	ignoreWorldStopByFriend: 1
 	sprite: s32'null', 1
 	deactivateObjectByName: s32'U_Kamehameha_Atk'
 	sprite: s32'null', 13
 	linkParticle: s32'ghu431c_core02'
 	addXPosition: -50000
 	setScale: 700
 	sprite: s32'null', 35
 } endState: 
 
 beginState: s32'U_Kamehameha_Atk' {
 	upon: (IMMEDIATE) {
 		overdriveAttack: 
 		callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 		callSubroutine: s32'cmnUltimateShot_AtkInit'
 		callSubroutine: s32'U_Kamehameha_AtkInit'
 		hitstop: 1
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 80000
 		attackYKnockback: 10000
 		attackYSuction: 1, 30000, 250000, 100, 0
 		wallBounceCount: 0
 		airHitstun: 60
 		atkRange: 50000000, -460000, 0, 0
 		disableOpponentRecovery: 0
 		atkDamageToRecovery: 1
 		atkSousaiSevereCheck: 1
 		atkEffectBeamSousai: 1
 		atkIgnoreSousaiSameLevel: 1
 		atkHitCommonBankSE: 16, s32'ARC_BTL_CMN_Hit_BeamLoop_LP', 0
 		atkGuardCommonBankSE: 17, s32'ARC_BTL_CMN_Grd_BeamLoop_LP', 0
 		unknown719: 1
 		upon: (SIGNAL7) {
 			groundKnockbackEffect: (SPIN)
 			airKnockbackEffect: (SPIN)
 			attackXKnockback: 20000
 			attackYKnockback: 35000
 			attackYSuction: 0, 0, 0, 0, 0
 			applyAttackLockWithStateName: s32'CmnNeutral'
 			ifVarOfObjectFalse: (PLAYER), var(PlayerDamage) {
 				enableHitbox: 
 			} endIf: 
 			else:  {
 				recoveryState: 
 			} endElse: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PARENT), var(act2), int(16) {
 				recoveryState: 
 			} endIf: 
 		} endUpon: 
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectAngle: 2
 		setActionTimeMax: 60
 		autoHitSignalSending: 1
 		worldOutToEliminate: 1
 		addXPosition: 130000
 		linkParticle: s32'ghu431c_naka01'
 		setYScale: 700
 		setZScale: 700
 		upon: (SIGNAL0) {
 			clearRegisteredUponCode: (SIGNAL0)
 			createObject: s32'U_Kamehameha_End', (ZERO)
 			copyVar: var(act1), int(1)
 			copyVarFromObject: (PARENT), var(act1), (SELF), int(1)
 		} endUpon: 
 		upon: (ON_ATTACK_COLLISION_SOUSAI) {
 			copyVarFromObject: (PARENT), var(ObjectParameter1), (SELF), var(ObjectParameter1)
 			deactivateObject: (SELF)
 			sendSignal: (PARENT), (SIGNAL1)
 		} endUpon: 
 		upon: (SIGNAL6) {
 			copyVar: var(act0), int(0)
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			callSubroutine: s32'cmnUltimate_CameraCombo'
 			if: var(ObjectParameter2) {
 				checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (ATTACK_COLLISION) {
 			if: var(act0) {
 				sendSignal: (PARENT), (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifVarOfObjectTrue: (PARENT), var(act2) {
 				ifVarOfObjectFalse: (PARENT), var(act4) {
 					copyVarFromObject: (SELF), var(Result), (PARENT), var(ObjectParameter1)
 					ifOperation: (IS_LESS_OR_EQUAL), var(ObjectParameter1), var(Result) {
 						deactivateObject: (SELF)
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			ifOperation: (IS_LESS_OR_EQUAL), var(PosY), int(500000) {
 				effectForLandBreak: 3, 1000, 0, 0
 			} endIf: 
 			copyVarFromObject: (SELF), var(Result), (ENEMY), var(PlayerDamageOrGuard)
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			if: var(act1) {
 				copyVar: var(act1), int(0)
 				copyVarFromObject: (PARENT), var(act1), (SELF), int(0)
 			} endIf: 
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			addXPosition: 130000
 			createParticleWithArg: s32'ghu431c_naka02', (ZERO)
 			setYScale: 700
 			setZScale: 700
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'U_Kamehameha', 1
 	enableHitbox: 
 	setSpeedByCurrentAngle: 90000, 0
 	label: s32'loop'
 	sprite: s32'U_Kamehameha', 1
 	atkIgnoreSousaiSameLevel: 0
 	sprite: s32'U_Kamehameha', 2
 	scaleForLinkY: 1200
 	sprite: s32'U_Kamehameha', 1
 	scaleForLinkY: 1000
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'U_Kamehameha_End' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		setActionTimeMax: 60
 		moveToPosType: (PARENT), (EX_POINT_0)
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu431b_end01'
 		addXPosition: 50000
 		setScale: 900
 		upon: (IDLING) {
 			checkIsObjectActive: (PARENT)
 			if: var(Result) {
 				copyVarFromParent: var(PosZ), var(PosZ)
 				moveToPosType: (PARENT), (EX_POINT_0)
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			setArg_multiplyScale: int(150)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu431b_end02', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	scaleForLinkY: 1000
 	sprite: s32'null', 2
 	scaleForLinkY: 1200
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'B_Kamehameha_Tame' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'ghu_kmhm_tame'
 		ignoreWorldStopByFriend: 1
 		addXPosition: 50000
 	} endUpon: 
 	sprite: s32'null', 3
 	setScale: 250
 	sprite: s32'null', 15
 	setScale: 300
 	addXPosition: -15000
 	addYPosition: -5000
 	spriteEnd: 
 } endState: 
 
 beginState: s32'B_Kamehameha_Core' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PLAYER)
 		linkObjectPosition: (PLAYER)
 		setActionTimeMax: 300
 		addXPosition: 130000
 		linkParticle: s32'ghu431c_core01'
 		jumpToStateInObject: (SELF), s32'end'
 		jumpToStateInObject: 47, s32'end'
 		upon: (PLAYER_DAMAGE) {
 			linkObjectPosition: (NULL)
 			linkObjectStop: (NULL)
 			ignoreWorldStopByAll: 1
 		} endUpon: 
 		upon: (SIGNAL1) {
 			clearRegisteredUponCode: (SIGNAL1)
 			copyVar: var(act0), int(0)
 			triggerUponInObject: s32'B_Kamehameha_Atk', (SIGNAL6)
 			checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				sendSignal: (PLAYER), (SIGNAL2)
 			} endIf: 
 			copyVar: var(act2), int(1)
 			copyVar: var(act6), int(1)
 			upon: (SIGNAL1) {
 				modifyVar: (ADD), var(act2), int(1)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(act2), int(16) {
 					checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL1)
 					} endIf: 
 				} endIf: 
 				else:  {
 					checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL2)
 					} endIf: 
 				} endElse: 
 			} endUpon: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifNot: var(act1) {
 				sendSignalActionNameOldest: s32'B_Kamehameha_Atk', (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	modifyVar: (ADD), var(act3), int(1)
 	setArg_objectParameters: var(act3), var(act6)
 	createObjectWithArg: s32'B_Kamehameha_Atk', (ZERO)
 	ifOperation: (IS_LESS_OR_EQUAL), var(act0), int(12) {
 		applyTo: (PREVIOUS) {
 			attackFinishBeam: 1, 0
 		} endApply: 
 	} endIf: 
 	ifOperation: (IS_GREATER_OR_EQUAL), var(act0), int(22) {
 		copyVarFromObject: (PREVIOUS), var(act0), (SELF), int(1)
 	} endIf: 
 	sprite: s32'null', 2
 	spriteEnd: 
 	modifyVar: (ADD), var(act0), int(1)
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(act0), int(24)
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	clearRegisteredUponCode: (AFTER_EXIT)
 	clearRegisteredUponCode: (IDLING)
 	copyVar: var(act4), int(1)
 	createObject: s32'B_Kamehameha_Atk', (ZERO)
 	triggerUponInObject: s32'B_Kamehameha_Atk', (SIGNAL7)
 	checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 	if: var(Result) {
 		sendSignal: (PLAYER), (SIGNAL1)
 	} endIf: 
 	addXPosition: 150000
 	linkObjectPosition: (NULL)
 	linkObjectStop: (NULL)
 	ignoreWorldStopByFriend: 1
 	sprite: s32'null', 1
 	deactivateObjectByName: s32'B_Kamehameha_Atk'
 	sprite: s32'null', 13
 	linkParticle: s32'ghu431c_core02'
 	addXPosition: -120000
 	setScale: 500
 	sprite: s32'null', 35
 } endState: 
 
 beginState: s32'B_Kamehameha_Atk' {
 	upon: (IMMEDIATE) {
 		overdriveAttack: 
 		callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 		callSubroutine: s32'cmnUltimateShot_AtkInit'
 		callSubroutine: s32'U_Kamehameha_AtkInit'
 		hitstop: 1
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 80000
 		attackYKnockback: 10000
 		attackYSuction: 1, 30000, 250000, 100, 0
 		wallBounceCount: 0
 		airHitstun: 60
 		atkRange: 50000000, -510000, 0, 0
 		disableOpponentRecovery: 0
 		atkDamageToRecovery: 1
 		atkSousaiSevereCheck: 1
 		atkEffectBeamSousai: 1
 		atkIgnoreSousaiSameLevel: 1
 		atkHitCommonBankSE: 16, s32'ARC_BTL_CMN_Hit_BeamLoop_LP', 0
 		atkGuardCommonBankSE: 17, s32'ARC_BTL_CMN_Grd_BeamLoop_LP', 0
 		atkAdditionalHitCount: 1
 		unknown719: 1
 		upon: (SIGNAL7) {
 			groundKnockbackEffect: (SPIN)
 			airKnockbackEffect: (SPIN)
 			attackXKnockback: 20000
 			attackYKnockback: 35000
 			attackYSuction: 0, 0, 0, 0, 0
 			applyAttackLockWithStateName: s32'CmnNeutral'
 			ifVarOfObjectFalse: (PLAYER), var(PlayerDamage) {
 				enableHitbox: 
 			} endIf: 
 			else:  {
 				recoveryState: 
 			} endElse: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PARENT), var(act2), int(16) {
 				recoveryState: 
 			} endIf: 
 		} endUpon: 
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectAngle: 2
 		setActionTimeMax: 60
 		autoHitSignalSending: 1
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu431c_naka01'
 		addXPosition: 100000
 		addZPosition: -10000
 		upon: (SIGNAL0) {
 			clearRegisteredUponCode: (SIGNAL0)
 			createObject: s32'B_Kamehameha_End', (ZERO)
 			copyVar: var(act1), int(1)
 			copyVarFromObject: (PARENT), var(act1), (SELF), int(1)
 		} endUpon: 
 		upon: (ON_ATTACK_COLLISION_SOUSAI) {
 			copyVarFromObject: (PARENT), var(ObjectParameter1), (SELF), var(ObjectParameter1)
 			deactivateObject: (SELF)
 			sendSignal: (PARENT), (SIGNAL1)
 		} endUpon: 
 		upon: (SIGNAL6) {
 			copyVar: var(act0), int(0)
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			callSubroutine: s32'cmnUltimate_CameraCombo'
 			if: var(ObjectParameter2) {
 				checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (ATTACK_COLLISION) {
 			if: var(act0) {
 				sendSignal: (PARENT), (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifVarOfObjectTrue: (PARENT), var(act2) {
 				ifVarOfObjectFalse: (PARENT), var(act4) {
 					copyVarFromObject: (SELF), var(Result), (PARENT), var(ObjectParameter1)
 					ifOperation: (IS_LESS_OR_EQUAL), var(ObjectParameter1), var(Result) {
 						deactivateObject: (SELF)
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			ifOperation: (IS_LESS_OR_EQUAL), var(PosY), int(500000) {
 				effectForLandBreak: 3, 1000, 0, 0
 			} endIf: 
 			copyVarFromObject: (SELF), var(Result), (ENEMY), var(PlayerDamageOrGuard)
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			if: var(act1) {
 				copyVar: var(act1), int(0)
 				copyVarFromObject: (PARENT), var(act1), (SELF), int(0)
 			} endIf: 
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu431c_naka02', (ZERO)
 			addXPosition: 100000
 			addZPosition: 1000
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'B_Kamehameha', 1
 	enableHitbox: 
 	setSpeedByCurrentAngle: 90000, 0
 	label: s32'loop'
 	sprite: s32'B_Kamehameha', 1
 	atkIgnoreSousaiSameLevel: 0
 	sprite: s32'B_Kamehameha', 2
 	scaleForLinkY: 1200
 	sprite: s32'B_Kamehameha', 1
 	scaleForLinkY: 1000
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'B_Kamehameha_End' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		setActionTimeMax: 60
 		moveToPosType: (PARENT), (EX_POINT_0)
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu431b_end01'
 		upon: (IDLING) {
 			checkIsObjectActive: (PARENT)
 			if: var(Result) {
 				copyVarFromParent: var(PosZ), var(PosZ)
 				moveToPosType: (PARENT), (EX_POINT_0)
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			setArg_multiplyScale: int(300)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu431b_end02', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	scaleForLinkY: 1000
 	sprite: s32'null', 2
 	scaleForLinkY: 1200
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'PC_Kamehameha_Tame' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'ghu_kmhm_tame'
 		ignoreWorldStopByFriend: 1
 		addXPosition: 50000
 	} endUpon: 
 	sprite: s32'null', 3
 	setScale: 250
 	sprite: s32'null', 15
 	setScale: 300
 	addXPosition: -15000
 	addYPosition: -5000
 	spriteEnd: 
 } endState: 
 
 beginState: s32'PC_Kamehameha_Core' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PLAYER)
 		linkObjectPosition: (PLAYER)
 		setActionTimeMax: 300
 		addXPosition: 190000
 		linkParticle: s32'ghu431c_core01'
 		setScale: 1500
 		jumpToStateInObject: (SELF), s32'end'
 		jumpToStateInObject: 47, s32'end'
 		upon: (PLAYER_DAMAGE) {
 			linkObjectPosition: (NULL)
 			linkObjectStop: (NULL)
 			ignoreWorldStopByAll: 1
 		} endUpon: 
 		upon: (SIGNAL1) {
 			clearRegisteredUponCode: (SIGNAL1)
 			copyVar: var(act0), int(0)
 			triggerUponInObject: s32'PC_Kamehameha_Atk', (SIGNAL6)
 			checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				sendSignal: (PLAYER), (SIGNAL2)
 			} endIf: 
 			copyVar: var(act2), int(1)
 			copyVar: var(act6), int(1)
 			upon: (SIGNAL1) {
 				modifyVar: (ADD), var(act2), int(1)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(act2), int(16) {
 					checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL1)
 					} endIf: 
 				} endIf: 
 				else:  {
 					checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL2)
 					} endIf: 
 				} endElse: 
 			} endUpon: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifNot: var(act1) {
 				sendSignalActionNameOldest: s32'PC_Kamehameha_Atk', (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	modifyVar: (ADD), var(act3), int(1)
 	setArg_objectParameters: var(act3), var(act6)
 	createObjectWithArg: s32'PC_Kamehameha_Atk', (ZERO)
 	ifOperation: (IS_LESS_OR_EQUAL), var(act0), int(12) {
 		applyTo: (PREVIOUS) {
 			attackFinishBeam: 1, 0
 		} endApply: 
 	} endIf: 
 	ifOperation: (IS_GREATER_OR_EQUAL), var(act0), int(22) {
 		copyVarFromObject: (PREVIOUS), var(act0), (SELF), int(1)
 	} endIf: 
 	sprite: s32'null', 2
 	spriteEnd: 
 	modifyVar: (ADD), var(act0), int(1)
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(act0), int(24)
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	clearRegisteredUponCode: (AFTER_EXIT)
 	clearRegisteredUponCode: (IDLING)
 	copyVar: var(act4), int(1)
 	createObject: s32'PC_Kamehameha_Atk', (ZERO)
 	triggerUponInObject: s32'PC_Kamehameha_Atk', (SIGNAL7)
 	checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 	if: var(Result) {
 		sendSignal: (PLAYER), (SIGNAL1)
 	} endIf: 
 	addXPosition: 150000
 	linkObjectPosition: (NULL)
 	linkObjectStop: (NULL)
 	ignoreWorldStopByFriend: 1
 	sprite: s32'null', 1
 	deactivateObjectByName: s32'PC_Kamehameha_Atk'
 	sprite: s32'null', 13
 	linkParticle: s32'ghu431_core01'
 	setScale: 500
 	addXPosition: -80000
 	linkParticle: s32'ghu431c_core02'
 	setScale: 700
 	addXPosition: -80000
 	sprite: s32'null', 35
 } endState: 
 
 beginState: s32'PC_Kamehameha_Atk' {
 	upon: (IMMEDIATE) {
 		overdriveAttack: 
 		callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 		callSubroutine: s32'cmnUltimateShot_AtkInit'
 		callSubroutine: s32'U_Kamehameha_AtkInit'
 		hitstop: 1
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 80000
 		attackYKnockback: 10000
 		attackYSuction: 1, 30000, 250000, 100, 0
 		wallBounceCount: 0
 		airHitstun: 60
 		atkRange: 50000000, -460000, 0, 0
 		disableOpponentRecovery: 0
 		atkDamageToRecovery: 1
 		atkSousaiSevereCheck: 1
 		atkEffectBeamSousai: 1
 		atkIgnoreSousaiSameLevel: 1
 		atkHitCommonBankSE: 16, s32'ARC_BTL_CMN_Hit_BeamLoop_LP', 0
 		atkGuardCommonBankSE: 17, s32'ARC_BTL_CMN_Grd_BeamLoop_LP', 0
 		atkAdditionalHitCount: 2
 		unknown719: 1
 		upon: (SIGNAL7) {
 			groundKnockbackEffect: (SPIN)
 			airKnockbackEffect: (SPIN)
 			attackXKnockback: 20000
 			attackYKnockback: 35000
 			attackYSuction: 0, 0, 0, 0, 0
 			applyAttackLockWithStateName: s32'CmnNeutral'
 			ifVarOfObjectFalse: (PLAYER), var(PlayerDamage) {
 				enableHitbox: 
 			} endIf: 
 			else:  {
 				recoveryState: 
 			} endElse: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PARENT), var(act2), int(16) {
 				recoveryState: 
 			} endIf: 
 		} endUpon: 
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectAngle: 2
 		setActionTimeMax: 60
 		autoHitSignalSending: 1
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu431c_naka01'
 		setYScale: 1500
 		setZScale: 1500
 		upon: (SIGNAL0) {
 			clearRegisteredUponCode: (SIGNAL0)
 			createObject: s32'PC_Kamehameha_End', (ZERO)
 			copyVar: var(act1), int(1)
 			copyVarFromObject: (PARENT), var(act1), (SELF), int(1)
 		} endUpon: 
 		upon: (ON_ATTACK_COLLISION_SOUSAI) {
 			copyVarFromObject: (PARENT), var(ObjectParameter1), (SELF), var(ObjectParameter1)
 			deactivateObject: (SELF)
 			sendSignal: (PARENT), (SIGNAL1)
 		} endUpon: 
 		upon: (SIGNAL6) {
 			copyVar: var(act0), int(0)
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			callSubroutine: s32'cmnUltimate_CameraCombo'
 			if: var(ObjectParameter2) {
 				checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (ATTACK_COLLISION) {
 			if: var(act0) {
 				sendSignal: (PARENT), (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifVarOfObjectTrue: (PARENT), var(act2) {
 				ifVarOfObjectFalse: (PARENT), var(act4) {
 					copyVarFromObject: (SELF), var(Result), (PARENT), var(ObjectParameter1)
 					ifOperation: (IS_LESS_OR_EQUAL), var(ObjectParameter1), var(Result) {
 						deactivateObject: (SELF)
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			ifOperation: (IS_LESS_OR_EQUAL), var(PosY), int(500000) {
 				effectForLandBreak: 3, 1000, 0, 0
 			} endIf: 
 			copyVarFromObject: (SELF), var(Result), (ENEMY), var(PlayerDamageOrGuard)
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'UltimateKamehameha'
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			if: var(act1) {
 				copyVar: var(act1), int(0)
 				copyVarFromObject: (PARENT), var(act1), (SELF), int(0)
 			} endIf: 
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu431c_naka02', (ZERO)
 			setYScale: 1500
 			setZScale: 1500
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'PC_Kamehameha', 1
 	enableHitbox: 
 	setSpeedByCurrentAngle: 90000, 0
 	label: s32'loop'
 	sprite: s32'PC_Kamehameha', 1
 	atkIgnoreSousaiSameLevel: 0
 	sprite: s32'PC_Kamehameha', 2
 	scaleForLinkY: 1200
 	sprite: s32'PC_Kamehameha', 1
 	scaleForLinkY: 1000
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'PC_Kamehameha_End' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		setActionTimeMax: 60
 		moveToPosType: (PARENT), (EX_POINT_0)
 		worldOutToEliminate: 1
 		linkParticle: s32'ghu431b_end01'
 		setScale: 1500
 		upon: (IDLING) {
 			checkIsObjectActive: (PARENT)
 			if: var(Result) {
 				copyVarFromParent: var(PosZ), var(PosZ)
 				moveToPosType: (PARENT), (EX_POINT_0)
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			setArg_multiplyScale: int(300)
 			setArg_addAngle: int(0)
 			createParticleWithArg: s32'ghu431b_end02', (ZERO)
 			setScale: 1500
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	scaleForLinkY: 1000
 	sprite: s32'null', 2
 	scaleForLinkY: 1200
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'ghu_431_aura' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PLAYER)
 		ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(act2), int(1) {
 			linkParticle: s32'ghu431_aura02'
 		} endIf: 
 		else:  {
 			linkParticle: s32'ghu431_aura02white'
 		} endElse: 
 		linkObjectPosition: (PARENT)
 		jumpToStateInObject: (SELF), s32'End'
 	} endUpon: 
 	sprite: s32'null', 600
 	label: s32'End'
 	sprite: s32'null', 30
 	ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(act2), int(1) {
 		linkParticle: s32'ghu431_aura02_End'
 	} endIf: 
 	else:  {
 		linkParticle: s32'ghu431_aura02white_End'
 	} endElse: 
 	linkObjectStop: (NULL)
 	linkObjectChangeToEliminate: (NULL)
 	linkObjectPosition: (NULL)
 } endState: 
 
 beginState: s32'431Eff' {
 	upon: (IMMEDIATE) {
 		addXPosition: 20000
 		linkParticle: s32'ghu_431_kaijo01'
 		setScale: 800
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	characterSfx: s32'ARC_BTL_GHU_UKame_OrlaEnd'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'gkh_groundSmokeMatome' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByAll: 1
 		addXPosition: 400000
 		addZPosition: 400000
 		jumpToStateInObject: (SELF), s32'EXloop'
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(1500)
 	setArg_setYAngle: int(-180000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 	label: s32'EXloop'
 	sprite: s32'null', 10
 	setArg_scale: int(2000)
 	setArg_setYAngle: int(-180000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'EXloop'
 } endState: 
 
 beginState: s32'groundSmoke' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: 150000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	createParticle: s32'bg_undersmoke00', (ZERO)
 	spriteEnd: 
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'kamehameha_shadow' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PLAYER)
 		linkParticle: s32'ghu_shadowLine_loop'
 		linkObjectPosition: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'ghu_431_tame' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu431a_kmhm_tame_loop'
 		ignoreWorldStopByFriend: 1
 		linkObjectPosition: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'ghu_431_Aura01' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkParticle: s32'ghu431_aura01'
 		linkObjectPosition: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'ghu_431_tameCS_GHU' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		jumpToStateInObject: (SELF), s32'End'
 	} endUpon: 
 	sprite: s32'null', 45
 	sprite: s32'null', 300
 	linkParticle: s32'ghu431_csTameLoop'
 	label: s32'End'
 	sprite: s32'null', 5
 	sprite: s32'null', 4
 	addXPosition: 100000
 	addYPosition: -15000
 	sprite: s32'null', 9
 } endState: 
 
 beginState: s32'ghu_431_tameCS_GON_Family' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu431_csTameLoop'
 		ignoreWorldStopByFriend: 1
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		jumpToStateInObject: (SELF), s32'End'
 	} endUpon: 
 	sprite: s32'null', 300
 	label: s32'End'
 	sprite: s32'null', 5
 	sprite: s32'null', 4
 	addXPosition: 40000
 	addYPosition: -50000
 	addZPosition: -70000
 	sprite: s32'null', 4
 	addXPosition: -45000
 	sprite: s32'null', 5
 	addXPosition: -20000
 } endState: 
 
 beginState: s32'ghu_431_tameCS_GKS' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		jumpToStateInObject: (SELF), s32'End'
 	} endUpon: 
 	sprite: s32'null', 300
 	linkParticle: s32'ghu431_csTameLoop'
 	label: s32'End'
 	sprite: s32'null', 5
 	sprite: s32'null', 4
 	addXPosition: 70000
 	addYPosition: -40000
 	sprite: s32'null', 9
 	addXPosition: -30000
 	addYPosition: 15000
 } endState: 
 
 beginState: s32'ghu_431_tameCS_GON_Brother' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu431_csTameLoop'
 		ignoreWorldStopByFriend: 1
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		jumpToStateInObject: (SELF), s32'End'
 	} endUpon: 
 	sprite: s32'null', 300
 	label: s32'End'
 	sprite: s32'null', 5
 	sprite: s32'null', 4
 	addYPosition: -50000
 	addZPosition: -70000
 	sprite: s32'null', 4
 	addXPosition: 10000
 	addYPosition: -20000
 	sprite: s32'null', 5
 	addXPosition: 20000
 } endState: 
 
 beginState: s32'gon431cs_02' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		zLine: var(plr0)
 		linkPawnWithAnimation: s32'GON', s32'GON_GHU431cs_02'
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'gon431cs_03' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		zLine: var(plr0)
 		linkPawnWithAnimation: s32'GON', s32'GON_GHU431cs_03'
 	} endUpon: 
 	sprite: s32'null', 45
 	spriteEnd: 
 } endState: 
 
 beginState: s32'gon431cs_04' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		zLine: var(plr0)
 		linkPawnWithAnimation: s32'GON', s32'GON_GHU431cs_05'
 	} endUpon: 
 	sprite: s32'null', 45
 	sprite: s32'null', 41
 	createObject: s32'ghu_431_tameCS_GON_Brother', (HAND_L_3D)
 	applyTo: (PREVIOUS) {
 		addXPosition: 25000
 		addYPosition: 12500
 		setScale: 700
 	} endApply: 
 	spriteEnd: 
 	spriteEnd: 
 } endState: 
 
 beginState: s32'gon431cs_05' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		zLine: var(plr0)
 		linkPawnWithAnimation: s32'GON', s32'GON_GHU431cs_04'
 	} endUpon: 
 	sprite: s32'null', 45
 	sprite: s32'null', 41
 	createObject: s32'ghu_431_tameCS_GON_Family', (HAND_L_3D)
 	applyTo: (PREVIOUS) {
 		addXPosition: 25000
 		addYPosition: 12500
 		setScale: 700
 	} endApply: 
 	spriteEnd: 
 } endState: 
 
 beginState: s32'gks431cs_03' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		zLine: var(CommonActionMark)
 		linkModelSelfTeam: s32'GKS', s32'GKS_GHU431cs_03'
 	} endUpon: 
 	sprite: s32'null', 45
 	spriteEnd: 
 } endState: 
 
 beginState: s32'gks431cs_04' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 		zLine: var(CommonActionMark)
 		linkModelSelfTeam: s32'GKS', s32'GKS_GHU431cs_04'
 	} endUpon: 
 	sprite: s32'null', 45
 	sprite: s32'null', 41
 	createObject: s32'ghu_431_tameCS_GKS', (HAND_L_3D)
 	applyTo: (PREVIOUS) {
 		addXPosition: 40000
 		addYPosition: 12500
 	} endApply: 
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_gon400' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		ignoreWorldStopByFriend: 1
 		setXPosition: -40000
 		setZPosition: 150000
 		linkPawnWithAnimation: s32'gon', s32''
 		upon: (AFTER_EXIT) {
 			clearRegisteredUponCode: (AFTER_EXIT)
 			linkObjectPosition: (NULL)
 			jumpToLabel: s32'end'
 		} endUpon: 
 		jumpToStateInObject: (SELF), s32'end'
 		upon: (SIGNAL1) {
 			setXPosition: -90000
 			setZPosition: 250000
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'gon400_12', 2
 	label: s32'loop'
 	sprite: s32'gon400_13', 2
 	sprite: s32'gon400_14', 2
 	sprite: s32'gon400_15', 2
 	gotoLabel: s32'loop'
 	label: s32'end'
 	sprite: s32'gon400_16', 2
 	sprite: s32'gon400_17', 2
 	sprite: s32'gon400_18', 2
 	sprite: s32'gon400_19', 2
 	createParticle: s32'cmn_Warp', (CENTER)
 } endState: 
 
 beginState: s32'ghu_gks400' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		ignoreWorldStopByFriend: 1
 		linkModelSelfTeam: s32'gks', s32''
 		setXPosition: -130000
 		setZPosition: 100000
 		upon: (AFTER_EXIT) {
 			clearRegisteredUponCode: (AFTER_EXIT)
 			linkObjectPosition: (NULL)
 			jumpToLabel: s32'end'
 		} endUpon: 
 		jumpToStateInObject: (SELF), s32'end'
 	} endUpon: 
 	sprite: s32'gks400_12', 2
 	label: s32'loop'
 	sprite: s32'gks400_13', 2
 	sprite: s32'gks400_14', 2
 	sprite: s32'gks400_15', 2
 	gotoLabel: s32'loop'
 	label: s32'end'
 	sprite: s32'gks400_16', 2
 	sprite: s32'gks400_17', 2
 	sprite: s32'gks400_18', 2
 	sprite: s32'gks400_19', 2
 	createParticle: s32'cmn_Warp', (CENTER)
 } endState: 
 
 beginState: s32'201Eff' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu_201_punch01'
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'201Eff2' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu_201_punch02'
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'201Eff3' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu_201_punch03'
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_267_move1' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu_267_move'
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_267_Smoke1' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -70000
 		addYPosition: -5000
 		addZPosition: 10000
 	} endUpon: 
 	sprite: s32'null', 30
 	setArg_scale: int(600)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_267_Smoke2' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: 70000
 		addYPosition: -5000
 		addZPosition: 10000
 	} endUpon: 
 	sprite: s32'null', 30
 	setArg_scale: int(600)
 	setArg_setYAngle: int(180000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_267_Smoke' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	createObject: s32'ghu_267_Smoke1', (ZERO)
 	createObject: s32'ghu_267_Smoke2', (ZERO)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu400_eff' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'ghu400_02'
 } endState: 
 
 beginState: s32'ghu400_eff2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		linkParticle: s32'ghu400_00'
 		addXPosition: -30000
 		addAngle: -7000
 	} endUpon: 
 	sprite: s32'null', 300
 } endState: 
 
 beginState: s32'ghu400_GroundEff' {
 	upon: (IMMEDIATE) {
 		setScale: 700
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 7
 	linkParticle: s32'ghu400_ground_end'
 	ignoreFinishStop: 1
 	ignoreWorldStopByAll: 1
 	sprite: s32'null', 10
 	ignoreFinishStop: 0
 	ignoreWorldStopByAll: 0
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu400_GroundEff_Assist' {
 	upon: (IMMEDIATE) {
 		setScale: 625
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 7
 	linkParticle: s32'ghu400_ground_end'
 	ignoreFinishStop: 1
 	ignoreWorldStopByAll: 1
 	sprite: s32'null', 10
 	ignoreFinishStop: 0
 	ignoreWorldStopByAll: 0
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu400_Eff3' {
 	upon: (IMMEDIATE) {
 		setScale: 700
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 2
 	linkParticle: s32'ghu400_01'
 	ignoreFinishStop: 1
 	ignoreWorldStopByAll: 1
 } endState: 
 
 beginState: s32'b_kick_eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		setScale: 650
 		addXPosition: 300000
 		addYPosition: 230000
 		setXSpeed: 15000
 	} endUpon: 
 	sprite: s32'null', 4
 	linkParticle: s32'ghu401_loop'
 	sprite: s32'null', 15
 	linkParticle: s32'ghu401_end'
 	linkObjectStop: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'b_kick_eff2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		setScale: 650
 		addXPosition: 260000
 		addYPosition: 310000
 		addAngle: 25000
 		setXSpeed: 15000
 		setYSpeed: 7500
 	} endUpon: 
 	sprite: s32'null', 4
 	linkParticle: s32'ghu401_loop'
 	sprite: s32'null', 15
 	linkParticle: s32'ghu401_end'
 	linkObjectStop: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'b_kick_eff3' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		setScale: 650
 		addXPosition: 260000
 		addYPosition: 150000
 		addAngle: -25000
 		setXSpeed: 15000
 		setYSpeed: -7500
 	} endUpon: 
 	sprite: s32'null', 4
 	linkParticle: s32'ghu401_loop'
 	sprite: s32'null', 15
 	linkParticle: s32'ghu401_end'
 	linkObjectStop: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'b_kick_inp_eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		addXPosition: 70000
 		addYPosition: 250000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu401_inpact'
 } endState: 
 
 beginState: s32'b_kick_fin_eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectDirection: (PARENT)
 		jumpToStateInObject: (SELF), s32'End'
 	} endUpon: 
 	sprite: s32'null', 10
 	linkParticle: s32'ghu401_finish_01'
 	sprite: s32'null', 15
 	linkParticle: s32'ghu401_finish_02'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_403_eff' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu403_Aura'
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_403_eff4' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu403_Aura02'
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByFriend: 1
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_403_swing' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu_403_swing01'
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_403_eff2' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu403_Shock01'
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	addAngle: -7000
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_403_eff3' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu403_Aura_warp'
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_430_eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'ghu430_spark01'
 } endState: 
 
 beginState: s32'ghu_430_eff2' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (PARENT)
 		linkParticle: s32'ghu430_inpact01'
 		ignoreWorldStopByFriend: 1
 	} endUpon: 
 	sprite: s32'null', 60
 	characterSfx: s32'ARC_BTL_GHU_Potential_Fin'
 } endState: 
 
 beginState: s32'groundSmoke_EXtame' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectPosition: (PARENT)
 		linkObjectDamageToEliminate: (PARENT)
 		setActionTimeMax: 30
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(1200)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	spriteEnd: 
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ghu_430_eff3' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'ghu430_tameAura01'
 		upon: (BEFORE_EXIT) {
 			stopBankSE: 0, 36
 			stopBankSE: 1, 50
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 25
 	privateBankSE: 0, s32'ARC_BTL_GHU_Potential_Start', 100
 	sprite: s32'null', 120
 	privateBankSE: 1, s32'ARC_BTL_GHU_Potential_Orla', 100
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_430_eff4' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (PARENT)
 		linkParticle: s32'ghu430_spark02'
 		ignoreWorldStopByFriend: 1
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'ghu_430_eff5' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (PARENT)
 		linkParticle: s32'ghu430_inpact02'
 		ignoreWorldStopByFriend: 1
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'FinishCamera_NmlAtk2C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cameraMotion: -91780, 241632, 14345, 4113, 4780, 0, 72, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_NmlAtkAir2C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cameraMotion: -91780, 241632, 14345, 4113, 4780, 0, 72, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_JetUpper' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	ifOperation: (IS_GREATER_OR_EQUAL), var(EnemyDistanceY), int(230000) {
 		cameraMotion: -111326, 563827, 38916, 2805, 2675, 0, 54, 0, 90, 30, 1
 	} endIf: 
 	else:  {
 		cameraMotion: -118510, 400981, 74135, 915, 3398, 0, 54, 0, 90, 30, 1
 	} endElse: 
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_NmlAtk3C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	ifOperation: (IS_GREATER_OR_EQUAL), var(EnemyDistanceY), int(250000) {
 		cameraMotion: -36640, 514892, 37540, 2731, 2099, 0, 54, 0, 90, 30, 1
 	} endIf: 
 	else:  {
 		cameraMotion: -161089, 419201, 62970, 1417, 5120, 0, 54, 0, 90, 30, 1
 	} endElse: 
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_Potential' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cameraMotion: 7659, 535726, 72974, 1535, -311, 0, 54, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'ght_600cutin_wind' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 300
 	setArg_setYScale: int(-1000)
 	setArg_setYAngle: int(180000)
 	createParticleWithArg: s32'bg_ghu600_wind', (ZERO)
 } endState: 
 
 beginState: s32'ght_600cutin_wind2' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 150
 	setArg_setYScale: int(-1000)
 	setArg_setYAngle: int(180000)
 	createParticleWithArg: s32'bg_ghu600_wind2', (ZERO)
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeA' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -75000
 		addZPosition: -450000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(1800)
 	setArg_setYAngle: int(90000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeB' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -75000
 		addZPosition: 450000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(1800)
 	setArg_setYAngle: int(-90000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeC' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -75000
 		addZPosition: 300000
 		addXPosition: -300000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(1500)
 	setArg_setYAngle: int(-40000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeD' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -75000
 		addZPosition: -300000
 		addXPosition: -300000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(1500)
 	setArg_setYAngle: int(38000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeE' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -75000
 		addZPosition: 200000
 		addXPosition: 365000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(1800)
 	setArg_setYAngle: int(1300000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeF' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -30000
 		addZPosition: -350000
 		addXPosition: 400000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(2500)
 	setArg_setYAngle: int(-1300000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeG' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -675000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(2500)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeBurstA' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -75000
 		addYPosition: -100000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 30
 	setArg_scale: int(2500)
 	setArg_setYAngle: int(900000)
 	createParticleWithArg: s32'bg_restart_dash', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ght_600cutin_SmokeBurstB' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -200000
 		addYPosition: -150000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 30
 	setArg_scale: int(2500)
 	createParticleWithArg: s32'bg_restart_dash', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'ghu_600aura' {
 	upon: (IMMEDIATE) {
 		linkParticle: s32'ghu600_Aura01'
 		addZPosition: 15000
 		ignoreWorldStopByAll: 1
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'ghu_431_Flash1' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'ghu431_scrFlash01'
 	} endUpon: 
 	sprite: s32'null', 9
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_431_Flash2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'ghu431_scrFlash02'
 	} endUpon: 
 	sprite: s32'null', 18
 	spriteEnd: 
 } endState: 
 
 beginState: s32'ghu_431_Flash' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	createObject: s32'ghu_431_Flash1', (ZERO)
 	createObject: s32'ghu_431_Flash2', (ZERO)
 } endState: 
 
 beginState: s32'GKS815_brs' {
 	upon: (IMMEDIATE) {
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkObjectPosition: (PARENT)
 		linkModelEnemyTeam: s32'brs', s32'brs_GHU815cs'
 		linkObjectChangeToEliminate: (PARENT)
 		zLine: int(5)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	upon: (IDLING) {
 		ifOnFrame: int(143) {
 			createObject: s32'GHU815_EnemyKindanBall', (ZERO)
 		} endIf: 
 		ifOnFrame: int(211) {
 			deactivateObjectByName: s32'GHU815_EnemyKindanBall'
 		} endIf: 
 		ifOnFrame: int(279) {
 			createObject: s32'GHU815_EnemyKindanTwinkle', (HAND_L_3D)
 		} endIf: 
 	} endUpon: 
 } endState: 
 
 beginState: s32'GHU815_Manp01_01' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'MNP', s32'MNP_GHU815cs_01'
 		zLine: int(5)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_Manp01_02' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'MNP', s32'MNP_GHU815cs_02'
 		zLine: int(5)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_Manp01_03' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'MNP', s32'MNP_GHU815cs_03'
 		zLine: int(5)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_Manp01_04' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'MNP', s32'MNP_GHU815cs_04'
 		zLine: int(5)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_GON' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'VS_brs_GON', s32'GON_GHU815cs'
 		zLine: int(5)
 		jumpToStateInObject: (SELF), s32'Super'
 		jumpToStateInObject: (ATTACKER), s32'Normal'
 		jumpToStateInObject: (DEFENDER), s32'Teri'
 		beginMeshSet: s32'Super'
 		addToMeshSet: s32'Super'
 		endMeshSet: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	label: s32'Super'
 	sprite: s32'null', 2147483647
 	swapMeshSet: s32'Super'
 	label: s32'Teri'
 	sprite: s32'null', 2147483647
 	createObject: s32'GHU815_GONRay', (HEAD_3D)
 	label: s32'Normal'
 	sprite: s32'null', 2147483647
 	changeDefaultMeshSet: 
 } endState: 
 
 beginState: s32'GHU815_GKS' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'VS_brs_GKS', s32'GKS_GHU815cs'
 		zLine: int(5)
 		jumpToStateInObject: (ATTACKER), s32'1'
 		jumpToStateInObject: (SELF), s32'0'
 		beginMeshSet: s32'Normal'
 		addToMeshSet: s32'Normal'
 		endMeshSet: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	label: s32'1'
 	sprite: s32'null', 2147483647
 	createObject: s32'GHU815_kmhmTame_GKS', (HAND_L_3D)
 	label: s32'0'
 	sprite: s32'null', 2147483647
 	swapMeshSet: s32'Normal'
 } endState: 
 
 beginState: s32'GHU815_EFFDUMMY' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkPawnWithAnimation: s32'EFF', s32'EFF_GHU815cs_01'
 		zLine: int(5)
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_ScrAnimRay' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_ScrAnimRay'
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 40
 } endState: 
 
 beginState: s32'GHU815_Space' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		setXPosition: 0
 		setYPosition: 0
 		setZPosition: 0
 		linkParticle: s32'ghu_815_space'
 		setScale: 10000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_EnemyKindanBall' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_KidanPower'
 		addXPosition: 4650000
 		addYPosition: 4450000
 		addZPosition: 480000
 		setScale: 250
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_EnemyKindanTwinkle' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_KindanTwinkle'
 		setScale: 100
 		addXPosition: -180000
 		addYPosition: 150000
 		addZPosition: -50000
 	} endUpon: 
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_KindanTwinkle'
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_KindanTwinkle2'
 	setScale: 300
 	setXSpeed: -3750
 	setYSpeed: -1875
 } endState: 
 
 beginState: s32'GHU815_EnemyKindanTwinkle2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_KindanTwinkle'
 		setScale: 100
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_SaiyaFlash' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_SaiyaFlash'
 		setScale: 1000
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'GHU815_kmhmTame' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhmTame'
 		setScale: 500
 		addXPosition: 45000
 		addYPosition: -45000
 		jumpToStateInObject: (SELF), s32'Move'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	label: s32'Move'
 	sprite: s32'null', 4
 	addXPosition: 30000
 	sprite: s32'null', 4
 } endState: 
 
 beginState: s32'GHU815_kmhmTame_GON' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhmTame'
 		setScale: 500
 		addXPosition: -200000
 		addYPosition: -100000
 		jumpToStateInObject: (SELF), s32'Move'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	label: s32'Move'
 	sprite: s32'null', 4
 	addXPosition: 30000
 	sprite: s32'null', 4
 } endState: 
 
 beginState: s32'GHU815_kmhm1A' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhm1'
 		setAngle: 43000
 		addZPosition: 200000
 		addYPosition: 49000
 		setAngleY: -5000
 	} endUpon: 
 	sprite: s32'null', 8
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_kmhm2'
 	setXScaleSpeed: 10
 } endState: 
 
 beginState: s32'GHU815_kmhm1B' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhm1'
 		setAngle: 43000
 		addZPosition: -400000
 		addYPosition: 210000
 		setAngleY: 5000
 	} endUpon: 
 	sprite: s32'null', 8
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_kmhm2'
 	setXScaleSpeed: 10
 } endState: 
 
 beginState: s32'GHU815_GHURay' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: 100000
 		addYPosition: -18000
 		setScale: 400
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_TameRay'
 } endState: 
 
 beginState: s32'GHU815_GONRay' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: 100000
 		addYPosition: -25000
 		setScale: 500
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_TameRay'
 } endState: 
 
 beginState: s32'GHU815_kmhmTame_GKS' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhmTame'
 		addXPosition: 100000
 		addYPosition: -7500
 		addZPosition: 50000
 		setScale: 400
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_EnemyKindanTwinkle3' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_KindanTwinkle'
 		addXPosition: 3000000
 		addYPosition: 3250000
 		jumpToStateInObject: (SELF), s32'Bomb'
 		jumpToStateInObject: (ATTACKER), s32'Bomb2'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	setScaleSpeed: 25
 	setScale: 500
 	addZPosition: 150000
 	label: s32'Bomb'
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_EnemyKindanBomb'
 	setScaleSpeed: 0
 	setScale: 1000
 	setAngleY: -10000
 	addZPosition: -150000
 	label: s32'Bomb2'
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_EnemyKindanBomb2'
 	setScale: 1000
 	setScaleSpeed: 0
 	setAngleY: -10000
 	addZPosition: -150000
 } endState: 
 
 beginState: s32'GHU815_kmhm2A' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhm3'
 		setAngle: 43000
 		addYPosition: 425000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_kmhm2B' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhm3'
 		setAngle: 43000
 		addYPosition: 250000
 		addZPosition: 300000
 		addXPosition: -150000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_Wind' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_GroundDust1'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_WhiteBG' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_WhiteBG'
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_kmhm3A' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhm4'
 		setScale: 350
 		addYPosition: 190000
 		addXPosition: -170000
 		setAngle: 10000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_kmhm3B' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhm4'
 		setScale: 550
 		addYPosition: 340000
 		addXPosition: -100000
 		setAngle: 10000
 		addZPosition: -250000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_kmhm3C' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhm4'
 		addYPosition: 340000
 		addXPosition: -10000
 		setAngle: 10000
 		addZPosition: -150000
 	} endUpon: 
 	sprite: s32'null', 4
 	setScale: 500
 	sprite: s32'null', 4
 	setScale: 1000
 	sprite: s32'null', 2147483647
 	setScale: 1100
 } endState: 
 
 beginState: s32'GHU815_kmhmChangeFlash' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhmChangeFlash'
 	} endUpon: 
 	sprite: s32'null', 30
 } endState: 
 
 beginState: s32'GHU815_kmhmRyuhai' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhmRyuhai'
 		setAngle: 10000
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_BigAura' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_Aura'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_kmhm4' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_kmhm5'
 		addYPosition: 250000
 		addXPosition: 150000
 		setAngle: -20000
 		setAngleY: 180000
 	} endUpon: 
 	sprite: s32'null', 6
 	setScale: 2000
 	sprite: s32'null', 4
 	setScale: 4000
 	sprite: s32'null', 4
 	setScale: 8000
 	sprite: s32'null', 4
 	setScale: 7750
 	sprite: s32'null', 4
 	setScale: 7500
 	sprite: s32'null', 4
 	setScale: 7250
 	sprite: s32'null', 2147483647
 	setScale: 7000
 	setScaleSpeed: 25
 } endState: 
 
 beginState: s32'GHU815_WhiteOut' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_WhiteOut'
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'GHU815_shoutotuKidan' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_shoutotuKidan'
 		flipDirection: 
 		jumpToStateInObject: (SELF), s32'Brake'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	label: s32'Brake'
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_shoutotuKidan2'
 } endState: 
 
 beginState: s32'GHU815_Earth' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_Earth'
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_beam' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		flipDirection: 
 		jumpToStateInObject: (SELF), s32'MoveToSun'
 	} endUpon: 
 	sprite: s32'null', 2
 	linkParticle: s32'ghu_815_beam01'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam02'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam03'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam04'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam05'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam06'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam07'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam08'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam09'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam10'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam11'
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_beam12'
 	label: s32'MoveToSun'
 	sprite: s32'null', 4
 	linkParticle: s32'ghu_815_beam13'
 	sprite: s32'null', 2147483647
 	linkParticle: s32'ghu_815_beam14'
 } endState: 
 
 beginState: s32'GHU815_Sun' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_sun1'
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_RyuhaiSun' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_RyuhaiSun'
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_Sun2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_sun2'
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 } endState: 
 
 beginState: s32'GHU815_FinalWind' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkParticle: s32'ghu_815_wind'
 		addXPosition: 1500000
 		setScale: 2200
 	} endUpon: 
 	sprite: s32'null', 240
 } endState: 
 
 beginState: s32'DUMMYDUMMY' {
 	sprite: s32'null', 1
 } endState: 
 
