--- 1.32/BBS_PCNEF.bbs	2023-05-24 20:54:50.362057615 +0200
+++ 1.33/BBS_PCNEF.bbs	2023-08-23 04:23:05.463962339 +0200
@@ -1,2288 +1,2291 @@
 beginState: s32'dmy' {
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'okasi' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'cmn_okasi'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'pcn_okasi'
 } endState: 
 
 beginState: s32'pcn_200Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addYPosition: 40000
 		addXPosition: 40000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'pcn200_slash'
 } endState: 
 
 beginState: s32'pcn_203Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'pcn203_slash'
 } endState: 
 
 beginState: s32'pcn_260Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'pcn260_slash'
 } endState: 
 
 beginState: s32'pcn_262Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'pcn262_slash'
 } endState: 
 
 beginState: s32'pcn_2CEff' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		setAngle: 92500
 		addYPosition: 350000
 		addXPosition: 125000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'cmn_strike00'
 } endState: 
 
 beginState: s32'pcn_5DEff' {
 	upon: (IMMEDIATE) {
 		addYPosition: 256000
 		addXPosition: 256000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'pcn205_arm'
 	createObject: s32'pcn_5D_smoke', (ZERO)
 } endState: 
 
 beginState: s32'pcn_5D_smoke' {
 	upon: (IMMEDIATE) {
 		addXPosition: 50000
 		setYPosition: 0
 	} endUpon: 
 	sprite: s32'null', 1
 	createParticle: s32'bg_undersmoke00', (ZERO)
 } endState: 
 
 beginState: s32'pcn_2DEff' {
 	upon: (IMMEDIATE) {
 		addYPosition: 480000
 		addXPosition: 288000
 	} endUpon: 
 	sprite: s32'null', 60
 	setAngle: 45000
 	linkParticle: s32'pcn233_arm'
 	createObject: s32'pcn_2D_smoke', (ZERO)
 } endState: 
 
 beginState: s32'pcn_2D_smoke' {
 	upon: (IMMEDIATE) {
 		addXPosition: -150000
 		setYPosition: 4000
 	} endUpon: 
 	sprite: s32'null', 1
 	createParticle: s32'bg_undersmoke00', (ZERO)
 } endState: 
 
 beginState: s32'pcn_hajiki' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'pcn_hajiki'
 } endState: 
 
 beginState: s32'pcn_air5DEff' {
 	upon: (IMMEDIATE) {
 		addYPosition: 128000
 		addXPosition: 256000
 	} endUpon: 
 	sprite: s32'null', 60
 	setAngle: -45000
 	linkParticle: s32'pcn265_arm'
 } endState: 
 
 beginSubroutine: s32'Dengeki_default' {
 	normalAttack: 
 	callSubroutine: s32'cmnAtkLevel_2_AtkInit'
 	callSubroutine: s32'cmnNormalShot_AtkInit'
 	damage: 350
 	attackXKnockback: 19000
 	attackYKnockback: 18000
 	atkHitBackForHit: 40
 	groundHitstun: 15
 	atkFrontDirection: 0
 	atkShotIgnoreZReversal: 1
 	attackDamageEffect: 1, s32''
 	hitEffectElectric: 1
 	checkCurrentStateName: (PLAYER), s32'AssistAttack2'
 	if: var(Result) {
 		damage: 300
 		attackXKnockback: 15000
 		attackYKnockback: 25100
 		groundHitstun: 22
 		airHitstun: 27
-		enemyHitstopAddition: 0, 10, 10
+		enemyHitstopAddition: 4, 14, 14
 		atkHoseiTadan: 0
 		callSubroutine: s32'cmn_AssistHosei'
 	} endIf: 
 	worldOutToEliminate: 1
 	linkObjectDamageToEliminate: (PLAYER)
 	atkHitCommonBankSE: 19, s32'ARC_BTL_CMN_Hit_Kidan', 0
 	atkGuardCommonBankSE: 20, s32'ARC_BTL_CMN_Guard_Beam', 0
 	jumpToStateInObject: (SELF), s32'end'
 	upon: (AFTER_EXIT) {
 		clearRegisteredUponCode: (AFTER_EXIT)
 		copyVar: var(act1), int(1)
 	} endUpon: 
 	upon: (SIGNAL1) {
 		copyVar: var(act0), int(1)
 		autoHitSignalSending: 1
 		upon: (ATTACK_COLLISION_PLAYER) {
 			ifNot: var(act1) {
 				checkCurrentStateName: (PLAYER), s32'NmlAtk6D'
 				if: var(Result) {
 					sendSignalName: (PLAYER), s32'PCN_Dengeki_Hit', 0
 				} endIf: 
 				checkCurrentStateName: (PLAYER), s32'NmlAtk3D'
 				if: var(Result) {
 					sendSignalName: (PLAYER), s32'PCN_Dengeki_Hit', 0
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	upon: (ATTACK_HIT_PLAYER_DAMAGE) {
 		clearRegisteredUponCode: (ATTACK_HIT_PLAYER_DAMAGE)
 		createParticleWithArg: s32'cmn_hit_sp2', (ATTACK)
 	} endUpon: 
 } endSubroutine: 
 
 beginState: s32'Dengeki' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectDamageToEliminate: (PLAYER)
 		linkParticle: s32'pcn402_core00'
 		scaleForLinkY: 0
 		atkEffectLimit: 0
 		destroyOnPlayerDamage: 1
 		jumpToStateInObject: 45, s32'end'
 		upon: (BEFORE_EXIT) {
 			stopBankSE: 19, 10
 			stopBankSE: 20, 10
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 3
 	setEX_POINT_F: (ZERO), 0, -30000
 	createObject: s32'Dengeki_end', (EX_POINT_F)
 	sprite: s32'null', 5
 	setEX_POINT_F: (ZERO), -108000, -30000
 	createObject: s32'Dengeki_naka', (EX_POINT_F)
 	sprite: s32'null', 1
 	setEX_POINT_F: (ZERO), -80000, -30000
 	createObject: s32'Dengeki_naka', (EX_POINT_F)
 	sendSignal: (PREVIOUS), (SIGNAL1)
 	label: s32'end'
 	sprite: s32'null', 30
 	clearRegisteredUponCode: (DESTROY)
 	clearRegisteredUponCode: (SIGNAL0)
 	linkParticle: s32'pcn402_core01'
 	triggerUponInObject: s32'Dengeki_end', (SIGNAL0)
 	triggerUponInObject: s32'Dengeki_naka', (SIGNAL0)
 	scaleForLinkY: 0
 	spriteEnd: 
 } endState: 
 
 beginState: s32'Dengeki_naka' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Dengeki_default'
 		linkParticle: s32'pcn402_naka00'
 	} endUpon: 
 	sprite: s32'Madengeki', 8
 	enableHitbox: 
 	setXSpeed: 70000
 	setAngleBySpeed: 0
 	worldOutToEliminate: 1
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	linkParticle: s32'pcn402_naka01'
 	if: var(act0) {
 		spriteOverrideFrame: s32'Madengeki_Last'
 	} endIf: 
 	resetSpeed: 1
 	sprite: s32'null', 30
 	recoveryState: 
 } endState: 
 
 beginState: s32'Dengeki_end' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Dengeki_default'
 		linkParticle: s32'pcn402_end00'
 	} endUpon: 
 	sprite: s32'Madengeki_end', 10
 	enableHitbox: 
 	setXSpeed: 70000
 	setAngleBySpeed: 0
 	label: s32'end'
 	sprite: s32'null', 30
 	clearRegisteredUponCode: (SIGNAL0)
 	linkParticle: s32'pcn402_end01'
 	recoveryState: 
 	resetSpeed: 1
 	spriteEnd: 
 } endState: 
 
 beginState: s32'Dengeki2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectDamageToEliminate: (PLAYER)
 		linkParticle: s32'pcn402_core00'
 		destroyOnPlayerDamage: 1
 		jumpToStateInObject: 45, s32'end'
 		upon: (BEFORE_EXIT) {
 			stopBankSE: 19, 10
 			stopBankSE: 20, 10
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 3
 	setEX_POINT_F: (ZERO), 40000, 10000
 	createObject: s32'Dengeki2_end', (EX_POINT_F)
 	sprite: s32'null', 5
 	setEX_POINT_F: (ZERO), -138000, -78000
 	createObject: s32'Dengeki2_naka', (EX_POINT_F)
 	sprite: s32'null', 1
 	createObject: s32'Dengeki2_naka', (EX_POINT_F)
 	sendSignal: (PREVIOUS), (SIGNAL1)
 	label: s32'end'
 	sprite: s32'null', 30
 	clearRegisteredUponCode: (DESTROY)
 	clearRegisteredUponCode: (SIGNAL0)
 	linkParticle: s32'pcn402_core01'
 	triggerUponInObject: s32'Dengeki2_end', (SIGNAL0)
 	triggerUponInObject: s32'Dengeki2_naka', (SIGNAL0)
 	recoveryState: 
 	resetSpeed: 1
 	spriteEnd: 
 } endState: 
 
 beginState: s32'Dengeki2_naka' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Dengeki_default'
 		attackYKnockback: 25000
 		linkParticle: s32'pcn402_naka00'
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_PRE_CTRL) {
 			clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_PRE_CTRL)
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(EnemyFrontDistance), int(300000) {
 				attackXKnockback: 17000
 			} endIf: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(EnemyFrontDistance), int(350000) {
 				attackXKnockback: 15000
 			} endIf: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(EnemyFrontDistance), int(400000) {
 				attackXKnockback: 12000
 			} endIf: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(EnemyFrontDistance), int(450000) {
 				attackXKnockback: 10000
 			} endIf: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(EnemyFrontDistance), int(500000) {
 				attackXKnockback: 8000
 			} endIf: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PLAYER), var(EnemyFrontDistance), int(550000) {
 				attackXKnockback: 5000
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'Madengeki2', 8
 	enableHitbox: 
 	setXSpeed: 70000
 	setYSpeed: 40000
 	setAngleBySpeed: 0
 	worldOutToEliminate: 1
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	linkParticle: s32'pcn402_naka01'
 	resetSpeed: 1
 	if: var(act0) {
 		spriteOverrideFrame: s32'Madengeki2_Last'
 	} endIf: 
 	sprite: s32'null', 30
 	recoveryState: 
 	spriteEnd: 
 } endState: 
 
 beginState: s32'Dengeki2_end' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Dengeki_default'
 		attackXKnockback: 8000
 		attackYKnockback: 25000
 		linkParticle: s32'pcn402_end00'
 	} endUpon: 
 	sprite: s32'Madengeki_end2', 10
 	enableHitbox: 
 	setXSpeed: 70000
 	setYSpeed: 40000
 	setAngleBySpeed: 0
 	label: s32'end'
 	sprite: s32'null', 30
 	clearRegisteredUponCode: (SIGNAL0)
 	linkParticle: s32'pcn402_end01'
 	resetSpeed: 1
 	recoveryState: 
 	spriteEnd: 
 } endState: 
 
 beginState: s32'HomingEdan_eff' {
 	upon: (IMMEDIATE) {
 		specialAttack: 
 		callSubroutine: s32'cmnAtkLevel_2_AtkInit'
 		callSubroutine: s32'cmnSpecialShot_AtkInit'
 		setProjectileDurability: 1
 		damage: 700
 		callSubroutine: s32'cmn_AssistShotHosei'
 		groundKnockbackEffect: (AIR_FACE_UP)
 		attackXKnockback: 3000
 		attackYKnockback: 25000
 		airHitstun: 26
 		hitstop: 10
 		atkFrontDirection: 0
 		atkShotIgnoreZReversal: 0
 		autoHitSignalSending: 1
 		atkIgnoreCaution: 1
 		atkShakeCamera: s32'ShakeTate', 600, 0, 25, 10
 		atkEasyGuard: 1
 		damageSoundEffect: s32'ARC_BTL_CMN_HmngSht_Expl'
 		guardSoundEffect: s32'ARC_BTL_CMN_Guard_Beam'
 		linkObjectDamageToEliminate: (PLAYER)
 		worldOutToEliminate: 1
 		landingHeight: 0
 		setXSpeed: 9000
 		upon: (SIGNAL0) {
 			requestDestroy: (SELF)
 		} endUpon: 
 		upon: (SIGNAL1) {
 			copyVar: var(act0), int(1)
 			setXSpeed: 6000
 			setScale: 1400
 		} endUpon: 
 		setEventTrigger: (TIMER), 300
 		upon: (TIMER) {
 			requestDestroy: (SELF)
 		} endUpon: 
 		checkCurrentStateName: (PLAYER), s32'AssistAttack'
 		if: var(Result) {
 			damage: 450
 			setEventTrigger: (TIMER), 180
 			copyVar: var(act1), int(1)
 			setXSpeed: 18000
 			enemyHitstopAddition: 0, 4, 4
+			hitstop: 12
+			attackXKnockback: 1000
 			if: var(ObjectParameter1) {
 				setEventTrigger: (TIMER), 240
 				copyVar: var(act0), int(1)
 				setXSpeed: 6000
 				setScale: 1400
 			} endIf: 
 		} endIf: 
 		numberOfHits: 1
 		destroyOnPlayerDamage: 1
 		upon: (DESTROY) {
 			deactivateObject: (SELF)
 			if: var(act0) {
 				setArg_scale: int(1400)
 			} endIf: 
 			createParticleWithArg: s32'pcn403_end', (ZERO)
 			characterSfx: s32'ARC_BTL_PCN_HmngSht_End'
 		} endUpon: 
 		upon: (ATTACK_COLLISION) {
 			deactivateObject: (SELF)
 			createObject: s32'cmn_kidan_end_M', (ZERO)
 			cmnSfx: s32'ARC_BTL_CMN_HmngSht_Expl'
 		} endUpon: 
 		upon: (Z_REVERSAL) {
 			clearRegisteredUponCode: (Z_REVERSAL)
 			clearRegisteredUponCode: (DESTROY)
 			deactivateObject: (SELF)
 			createObject: s32'cmn_reversalkidan', (ZERO)
 			linkParticle: s32''
 		} endUpon: 
 		upon: (TOUCH_GROUND) {
 			deactivateObject: (SELF)
 			effectForLandBreak: 0, 1000, 0, 0
 			createParticle: s32'bg_groundsmokeM', (ZERO)
 			cmnSfx: s32'ARC_BTL_CMN_HmngSht_Expl'
 		} endUpon: 
 		upon: (AFTER_EXIT) {
 			clearRegisteredUponCode: (AFTER_EXIT)
 			copyVar: var(act3), int(1)
 		} endUpon: 
 		upon: (ON_ATTACK_COLLISION_MAIN_PLAYER) {
 			ifNot: var(act3) {
 				checkCurrentStateName: (PLAYER), s32'HomingEdan'
 				if: var(Result) {
 					sendSignalName: (PLAYER), s32'PCN_HomEdanShot', 0
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifOnFrame: int(30) {
 				privateBankSE: 0, s32'ARC_BTL_PCN_HmngSht_Sphere', 100
 			} endIf: 
 			if: var(act2) {
 				faceOpponent: 
 				homing: 2, 32, 116, 0, 0, 0, 0, 0, 11000, 0, 10000
 				if: var(act1) {
 					homing: 2, 32, 116, 0, 0, 0, 0, 0, 18000, 0, 10000
 				} endIf: 
 				if: var(act0) {
 					homing: 2, 32, 116, 0, 0, 0, 0, 0, 8000, 0, 10000
 				} endIf: 
 			} endIf: 
 			if: var(act1) {
 				if: var(act3) {
 					ifVarOfObjectOperation: (IS_EQUAL), (SELF_MAIN_PLAYER), var(PlayerDamageOrGuard), int(1) {
 						requestDestroy: (SELF)
 					} endIf: 
 					ifVarOfObjectOperation: (IS_EQUAL), (ENEMY_MAIN_PLAYER), var(PlayerDamageOrGuard), int(1) {
 						ifVarOfObjectOperation: (IS_EQUAL), (ENEMY_MAIN_PLAYER), var(IsLastDamageKakusei), int(1) {
 							ifVarOfObjectTrue: (SELF_MAIN_PLAYER), var(Gravity) {
 								requestDestroy: (SELF)
 							} endIf: 
 						} endIf: 
 					} endIf: 
 					checkCurrentStateName: (SELF_MAIN_PLAYER), s32'CmnActRushRush'
 					if: var(Result) {
 						requestDestroy: (SELF)
 					} endIf: 
 					checkCurrentStateName: (SELF_MAIN_PLAYER), s32'CmnActRushSousai'
 					if: var(Result) {
 						requestDestroy: (SELF)
 					} endIf: 
 				} endIf: 
 				else:  {
 					checkCurrentStateName: (SELF_MAIN_PLAYER), s32'CmnActRushRush'
 					if: var(Result) {
 						requestDestroy: (SELF)
 					} endIf: 
 					checkCurrentStateName: (SELF_MAIN_PLAYER), s32'CmnActRushSousai'
 					if: var(Result) {
 						requestDestroy: (SELF)
 					} endIf: 
 				} endElse: 
 			} endIf: 
 			else:  {
 				callSubroutine: s32'cmnNotMainPlayer_ObjectDelete'
 			} endElse: 
 			ifNot: var(IN_BATTLE) {
 				requestDestroy: (SELF)
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			stopBankSE: 0, 20
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'HomingEdan', 2
 	linkParticleArg: s32'pcn403_shot'
 	createObject: s32'HomingEdan_add', (ZERO)
 	enableHitbox: 
 	sprite: s32'keep', 13
 	setColorAlpha: 255
 	sprite: s32'keep', 3
 	copyVar: var(act2), int(1)
 	label: s32'loop'
 	sprite: s32'keep', 3
 	createObject: s32'HomingEdan_add', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'HomingEdan_add' {
 	sprite: s32'null', 1
 	setYPosition: 0
 	addXPosition: 30000
 	createParticleWithArg: s32'pcn403_light', (ZERO)
 } endState: 
 
 beginState: s32'pcn403_smoke' {
 	upon: (IMMEDIATE) {
 		addXPosition: -100000
 		setYPosition: 0
 	} endUpon: 
 	sprite: s32'null', 1
 	createParticle: s32'bg_undersmoke00', (ZERO)
 } endState: 
 
 beginState: s32'HomingEdan_start' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'HomingEdan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 2
 	characterSfx: s32'ARC_BTL_PCN_HmngSht_Fire'
 	setZPosition: -50000
 	sprite: s32'null', 2
 	setZPosition: -140000
 	addXPosition: -110000
 	addYPosition: 10000
 	sprite: s32'null', 2
 	setZPosition: -150000
 	addXPosition: -20000
 	addYPosition: 10000
 } endState: 
 
 beginState: s32'HomingEdan_start_aura' {
 	upon: (IMMEDIATE) {
 		worldOutToEliminate: 1
 		linkObjectDamageToEliminate: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 		checkCurrentStateName: (PLAYER), s32'AssistAttack'
 		if: var(Result) {
 			addZPosition: -350000
 		} endIf: 
 	} endUpon: 
 	sprite: s32'null', 4
 	setScale: 400
 	setScaleSpeed: 162
 	sprite: s32'null', 2147483647
 	setScale: 750
 	setScaleSpeed: 0
 } endState: 
 
 beginState: s32'HomingEdan_aura' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		upon: (BEFORE_EXIT) {
 			stopBankSE: 0, 20
 		} endUpon: 
 	} endUpon: 
 	privateBankSE: 0, s32'ARC_BTL_PCN_HmngSht_Chrg', 100
 	label: s32'0'
 	sprite: s32'null', 3
 	createParticle: s32'pcn403_spark', (ZERO)
 	gotoLabel: s32'0'
 } endState: 
 
 beginState: s32'HomingEdan_aura2' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PLAYER)
 		setYPosition: 0
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'pcn403_aura'
 } endState: 
 
 beginState: s32'PCN400_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreFinishStop: 1
 		ignoreWorldStopByAll: 1
 		jumpToStateInObject: (SELF), s32'end'
 		addXPosition: 25000
 		addYPosition: 40000
 	} endUpon: 
 	sprite: s32'null', 3
 	linkParticle: s32'pcn400_00'
 	sprite: s32'null', 3
 	linkParticle: s32'pcn400_01'
 	sprite: s32'null', 2147483647
 	linkParticle: s32'pcn400_02'
 	linkObjectStop: (PARENT)
 	ignoreFinishStop: 0
 	ignoreWorldStopByAll: 0
 	spriteEnd: 
 	label: s32'end'
 	sprite: s32'null', 20
 	linkParticle: s32'pcn400_03'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'PCN400B_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreFinishStop: 1
 		ignoreWorldStopByAll: 1
 		addXPosition: 100000
 		addYPosition: 25000
 		addZPosition: 280000
 		jumpToStateInObject: (SELF), s32'end'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'pcn400b_00'
 	linkObjectStop: (PARENT)
 	ignoreFinishStop: 0
 	ignoreWorldStopByAll: 0
 	spriteEnd: 
 	label: s32'end'
 	sprite: s32'null', 20
 	linkParticle: s32'pcn400b_01'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'PCN401_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectStop: (PARENT)
 		jumpToStateInObject: (SELF), s32'end'
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'pcn401_00'
 	setArg_scale: int(1250)
 	setArg_setYAngle: int(180000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	createParticle: s32'pcn401_nokosi', (ZERO)
 	spriteEnd: 
 	label: s32'end'
 	sprite: s32'null', 20
 	clearRegisteredUponCode: (SIGNAL0)
 	linkParticle: s32'pcn401_01'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'PCN401_AirEff' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectStop: (PARENT)
 		jumpToStateInObject: (SELF), s32'end'
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'pcn401_00'
 	setArg_scale: int(1250)
 	setArg_setYAngle: int(180000)
 	createParticle: s32'pcn401_nokosi', (ZERO)
 	spriteEnd: 
 	label: s32'end'
 	sprite: s32'null', 20
 	linkParticle: s32'pcn401_01'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'MakuuL_Ue' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 2
 	setZPosition: -180000
 } endState: 
 
 beginState: s32'MakuuL_Ue2' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 2
 	setZPosition: -180000
 } endState: 
 
 beginState: s32'MakuuL_Ue3' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 1
 	setZPosition: -180000
 } endState: 
 
 beginState: s32'MakuuL_Naka' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 3
 	setZPosition: -170000
 	sprite: s32'keep', 2
 	addXPosition: 70000
 	addYPosition: 40000
 } endState: 
 
 beginState: s32'MakuuL_Shita' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 2
 	setZPosition: -180000
 } endState: 
 
 beginState: s32'MakuuR_Ue' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 1
 	setZPosition: 150000
 	sprite: s32'keep', 2
 	addXPosition: 30000
 	addYPosition: 20000
 	addZPosition: 30000
 	sprite: s32'keep', 2
 	addXPosition: 155000
 	addYPosition: 55000
 } endState: 
 
 beginState: s32'MakuuR_Naka' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 1
 	setZPosition: 150000
 	sprite: s32'keep', 2
 	addXPosition: 35000
 	addYPosition: 20000
 	addZPosition: 30000
 	sprite: s32'keep', 2
 	addXPosition: 155000
 	addYPosition: 55000
 	addZPosition: 20000
 } endState: 
 
 beginState: s32'MakuuR_Shita' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 1
 	setZPosition: 130000
 	sprite: s32'keep', 2
 	addXPosition: 40000
 	addYPosition: 20000
 	addZPosition: 50000
 	sprite: s32'keep', 2
 	addXPosition: 140000
 	addYPosition: 55000
 	addZPosition: -10000
 } endState: 
 
 beginState: s32'MakuuR_Shita2' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 1
 	setZPosition: 130000
 	sprite: s32'keep', 2
 	addXPosition: 40000
 	addYPosition: 20000
 	addZPosition: 50000
 	sprite: s32'keep', 2
 	addXPosition: 140000
 	addYPosition: 55000
 	addZPosition: -10000
 } endState: 
 
 beginState: s32'MakuuR_Shita3' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		createObject: s32'kidan_start_aura', (ZERO)
 	} endUpon: 
 	sprite: s32'null', 1
 	setZPosition: 130000
 	sprite: s32'keep', 2
 	addXPosition: 40000
 	addYPosition: 20000
 	addZPosition: 50000
 	sprite: s32'keep', 2
 	addXPosition: 140000
 	addYPosition: 55000
 	addZPosition: -10000
 } endState: 
 
 beginState: s32'kidan_start_aura' {
 	upon: (IMMEDIATE) {
 		linkObjectDamageToEliminate: (NULL)
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_kidan_throw'
 	} endUpon: 
 	sprite: s32'keep', 3
 	setScale: 600
 	sprite: s32'keep', 7
 	setScale: 800
 	sprite: s32'keep', 2147483647
 } endState: 
 
 beginSubroutine: s32'Makuu_dafault' {
 	addSkillForceDisableFlag: 1
+	autoHitSignalSending: 1
 	overdriveAttack: 
 	callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 	callSubroutine: s32'cmnUltimateShot_AtkInit'
 	setProjectileDurability: 2
 	damage: 260
 	minimalDamagePercent: 20
 	damageScalingOverride: 500
 	atkHoseiTadan: 1
 	hitstop: 2
 	guardPushback: 30
 	groundKnockbackEffect: (AIR_FACE_UP)
 	airKnockbackEffect: (AIR_FACE_UP)
 	attackXKnockback: 5000
 	attackYKnockback: 24000
 	atkHitTensionPenalty: 1
 	atkIgnoreCaution: 1
 	attackDamageEffect: 4, s32'cmn_BomHit'
 	atkYogore: 0
 	disableDoubleSuperScaling: 1
 	ignoreWorldStopByFriend: 1
 	atkShakeCamera: s32'ShakeTate', 600, 1, 15, 0
 	linkParticle: s32'cmn_kidan_homing'
 	upon: (SIGNAL0) {
 		ignoreWorldStopByFriend: 0
 		jumpToLabel: s32'attack'
 	} endUpon: 
 	landingHeight: 0
 	numberOfHits: 1
 	destroyOnPlayerDamage: 1
 	destroyOnEnemyDamage: 1
 	destroyOnHit: 1
 	kowareCountSousai: 1
 	upon: (DESTROY) {
 		clearRegisteredUponCode: (DESTROY)
 		linkParticle: s32''
 		createObject: s32'cmn_kidan_end', (ZERO)
 		jumpToLabel: s32'end'
 	} endUpon: 
 	upon: (ON_ATTACK_HIT_PLAYER_GUARD) {
 		clearRegisteredUponCode: (ON_ATTACK_HIT_PLAYER_GUARD)
 		linkParticle: s32''
 		if: var(act0) {
 			createObject: s32'cmn_kidan_end_L', (ZERO)
 			cmnSfx: s32'ARC_BTL_CMN_KidanLast_Expl'
 			jumpToLabel: s32'end'
 			addWindPowerOffset: 0, 0, 0, 1000, 0, 60, 30
 			destructionCollision: 0, 0, 0, 10000
 			cameraShake: s32'ShakeTate', 1500, 1, 30, 10
 		} endIf: 
 		else:  {
 			createObject: s32'cmn_kidan_end', (ZERO)
 		} endElse: 
 		jumpToLabel: s32'end'
 	} endUpon: 
 	upon: (ATTACK_HIT_PLAYER_ARMOR) {
 		clearRegisteredUponCode: (ATTACK_HIT_PLAYER_ARMOR)
 		linkParticle: s32''
 		createObject: s32'cmn_kidan_end', (ZERO)
 		jumpToLabel: s32'end'
 	} endUpon: 
 	upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 		clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
 		callSubroutine: s32'cmnUltimate_CameraCombo'
 		if: var(act0) {
 			createObject: s32'cmn_kidan_end_L', (ZERO)
 			cmnSfx: s32'ARC_BTL_CMN_KidanLast_Expl'
 			jumpToLabel: s32'end'
 			addWindPowerOffset: 0, 0, 0, 1000, 0, 60, 30
 			destructionCollision: 0, 0, 0, 10000
 			atkShakeCamera: s32'ShakeTate', 1500, 1, 30, 10
 		} endIf: 
 		jumpToLabel: s32'end'
 		finishCameraRequest: s32'FinishCamera_MakuuHouidan'
 	} endUpon: 
 	upon: (SHOT_HIT_Z_REVERSAL_ACT) {
 		clearRegisteredUponCode: (SHOT_HIT_Z_REVERSAL_ACT)
 		clearRegisteredUponCode: (DESTROY)
 		deactivateObject: (SELF)
 		clearRegisteredUponCode: (ATTACK_HIT_PLAYER_ARMOR)
 		deactivateObjectByName: s32'cmn_kidan_end'
 		createObject: s32'reversalkidan', (ZERO)
 	} endUpon: 
 	upon: (IDLING) {
 		callSubroutine: s32'cmnNotMainPlayer_ObjectDelete'
 	} endUpon: 
 	upon: (TOUCH_GROUND) {
 		createParticle: s32'bg_groundsmokeM', (ZERO)
 		effectForLandBreak: 0, 1000, 0, 0
 		jumpToLabel: s32'end'
 	} endUpon: 
 	damageSoundEffect: s32'ARC_BTL_CMN_Hit_Kidan'
 	guardSoundEffect: s32'ARC_BTL_CMN_Guard_Kidan'
 } endSubroutine: 
 
 beginState: s32'Makuu_kidan1' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 120000
 	setYSpeed: 5000
 	setZSpeed: 40000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	characterSfx: s32'ARC_BTL_PCN_Mkhdn_KidanAir'
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 13
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	characterSfx: s32'ARC_BTL_PCN_Mkhdn_KidanAttk'
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan2' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 60000
 	setYSpeed: 65000
 	setZSpeed: 20000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 18
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan3' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 20000
 	setYSpeed: 35000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 23
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan4' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 80000
 	setYSpeed: 15000
 	setZSpeed: 30000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 28
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan5' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 60000
 	setYSpeed: 35000
 	setZSpeed: -60000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 33
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan6' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 70000
 	setYSpeed: 60000
 	setZSpeed: -20000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 38
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan7' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 80000
 	setYSpeed: 25000
 	setZSpeed: -30000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 43
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan8' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 20000
 	setYSpeed: 35000
 	setZSpeed: -40000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 48
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan9' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 105000
 	setYSpeed: 50000
 	setZSpeed: -30000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 53
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan10' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 35000
 	setYSpeed: 70000
 	setZSpeed: 10000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 58
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan11' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 92000
 	setYSpeed: 45000
 	setZSpeed: 50000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 63
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'Makuu_kidan_Last' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'Makuu_dafault'
 		atkYogore: 1
 		attackXKnockback: 8000
 		attackYKnockback: 23000
 		airHitstun: 28
 		hitstop: 5
 		guardPushback: 100
 		copyVar: var(act0), int(1)
 		atkHitCommonSEReset: s32'dammy'
 	} endUpon: 
 	sprite: s32'null', 1
 	setColorAlpha: 0
 	setXSpeed: 109000
 	setYSpeed: 50000
 	setZSpeed: -10000
 	sprite: s32'null', 7
 	setColorAlpha: 255
 	setAngleBySpeed: 0
 	sprite: s32'keep', 300
 	linkParticle: s32'cmn_kidan_hold'
 	setXSpeed: 0
 	setYSpeed: 0
 	setZSpeed: 0
 	setZAcceleration: 0
 	label: s32'attack'
 	sprite: s32'Makuuhouidan', 68
 	clearRegisteredUponCode: (SIGNAL0)
 	setZPosition: 0
 	sprite: s32'Makuuhouidan', 1
 	linkParticle: s32'cmn_kidan_homing'
 	faceOpponent: 
 	enableHitbox: 
 	homing: 2, 22, 116, 0, 0, 0, 0, 0, 65000, 0, 65000
 	upon: (IDLING) {
 		ifOperation: (IS_LESS_OR_EQUAL), var(SpeedY), int(0) {
 			setAngleBySpeed: 20000
 		} endIf: 
 		else:  {
 			setAngleBySpeed: -20000
 		} endElse: 
 	} endUpon: 
 	xSpeedPerFrame: 102
 	ySpeedPerFrame: 102
 	sprite: s32'Makuuhouidan', 60
 	homingClear: 
 	label: s32'end'
 	sprite: s32'null', 70
 	clearRegisteredUponCode: (IDLING)
 	clearRegisteredUponCode: (DESTROY)
 	linkParticle: s32''
 	homingClear: 
 	resetSpeed: 1
 } endState: 
 
 beginState: s32'reversalkidan' {
 	upon: (IMMEDIATE) {
 		linkCollisionObject: s32'cmn_reversal_kidan'
 		resetSpeed: 1
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'Makankou_tame' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'pcn431_tame00'
 		ignoreWorldStopByFriend: 1
 		jumpToStateInObject: (SELF), s32'normal'
 		jumpToStateInObject: (ATTACKER), s32'max'
 		ignoreDispOnlyMe: 1
 	} endUpon: 
 	sprite: s32'keep', 2147483647
 	addZPosition: 45000
 	label: s32'normal'
 	sprite: s32'keep', 4
 	addXPosition: -190000
 	addYPosition: 90000
 	addZPosition: 150000
 	sprite: s32'keep', 4
 	addXPosition: 40000
 	addYPosition: 55000
 	addZPosition: -230000
 	sprite: s32'keep', 4
 	addXPosition: 30000
 	addYPosition: 10000
 	addZPosition: -40000
 	sprite: s32'keep', 12
 	addXPosition: 20000
 	addYPosition: -5000
 	addZPosition: -25000
 	sprite: s32'keep', 3
 	addXPosition: -90000
 	addYPosition: -75000
 	addZPosition: 140000
 	spriteEnd: 
 	exitState: 
 	label: s32'max'
 	sprite: s32'keep', 3
 	addXPosition: 20000
 	addYPosition: -30000
 	addZPosition: 40000
 	sprite: s32'keep', 3
 	addXPosition: 90000
 	addYPosition: -110000
 	addZPosition: -60000
 	sprite: s32'keep', 3
 	addXPosition: 35000
 	addYPosition: -30000
 	addZPosition: -120000
 	sprite: s32'keep', 3
 	addXPosition: -40000
 	addYPosition: -20000
 	addZPosition: -80000
 	sprite: s32'keep', 3
 	addXPosition: -35000
 	addZPosition: -30000
 	sprite: s32'keep', 3
 	addXPosition: -5000
 	addYPosition: 50000
 	addZPosition: -10000
 	sprite: s32'keep', 3
 	addXPosition: 100000
 	addZPosition: 110000
 	sprite: s32'keep', 3
 	addXPosition: -70000
 	addYPosition: 45000
 	addZPosition: 315000
 	sprite: s32'keep', 4
 	addXPosition: -250000
 	addYPosition: 170000
 	addZPosition: 10000
 	sprite: s32'keep', 4
 	addXPosition: -15000
 	addYPosition: 75000
 	addZPosition: -290000
 	sprite: s32'keep', 4
 	addXPosition: 30000
 	addYPosition: -5000
 	addZPosition: -30000
 	sprite: s32'keep', 12
 	addXPosition: 20000
 	addYPosition: -5000
 	addZPosition: -15000
 	sprite: s32'keep', 3
 	addXPosition: -50000
 	addYPosition: -85000
 	addZPosition: 220000
 } endState: 
 
 beginState: s32'Makankou_core' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		setActionTimeMax: 300
 		linkParticle: s32'pcn431_core00'
 		jumpToStateInObject: (SELF), s32'end'
 		jumpToStateInObject: 47, s32'end'
 		upon: (PLAYER_DAMAGE) {
 			linkObjectPosition: (NULL)
 			linkObjectStop: (NULL)
 			ignoreWorldStopByAll: 1
 		} endUpon: 
 		upon: (SIGNAL1) {
 			clearRegisteredUponCode: (SIGNAL1)
 			copyVar: var(act0), int(0)
 			triggerUponInObject: s32'Makankou_Atk', (SIGNAL6)
 			checkCurrentStateName: (PLAYER), s32'Makankou'
 			if: var(Result) {
 				copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				sendSignal: (PLAYER), (SIGNAL2)
 			} endIf: 
 			copyVar: var(act2), int(1)
 			copyVar: var(act6), int(1)
 			upon: (SIGNAL1) {
 				modifyVar: (ADD), var(act2), int(1)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(act2), int(8) {
 					checkCurrentStateName: (PLAYER), s32'Makankou'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL1)
 					} endIf: 
 				} endIf: 
 				else:  {
 					checkCurrentStateName: (PLAYER), s32'Makankou'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL2)
 					} endIf: 
 				} endElse: 
 			} endUpon: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifNot: var(act1) {
 				sendSignalActionNameOldest: s32'Makankou_Atk', (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	modifyVar: (ADD), var(act3), int(1)
 	setArg_objectParameters: var(act3), var(act6)
 	createObjectWithArg: s32'Makankou_Atk', (ZERO)
 	ifOperation: (IS_GREATER_OR_EQUAL), var(act0), int(8) {
 		copyVarFromObject: (PREVIOUS), var(act0), (SELF), int(1)
 	} endIf: 
 	sprite: s32'null', 2
 	spriteEnd: 
 	modifyVar: (ADD), var(act0), int(1)
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(act0), int(10)
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	clearRegisteredUponCode: (AFTER_EXIT)
 	clearRegisteredUponCode: (IDLING)
 	copyVar: var(act4), int(1)
 	createObject: s32'Makankou_Atk', (ZERO)
 	triggerUponInObject: s32'Makankou_Atk', (SIGNAL7)
 	checkCurrentStateName: (PLAYER), s32'Makankou'
 	if: var(Result) {
 		sendSignal: (PLAYER), (SIGNAL1)
 	} endIf: 
 	linkObjectPosition: (NULL)
 	linkObjectStop: (NULL)
 	ignoreWorldStopByFriend: 1
 	sprite: s32'null', 1
 	deactivateObjectByName: s32'Makankou_Atk'
 	sprite: s32'null', 13
 	linkParticle: s32'pcn431_core01'
 	sprite: s32'null', 35
 } endState: 
 
 beginState: s32'Makankou_Atk' {
 	upon: (IMMEDIATE) {
 		overdriveAttack: 
 		callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 		callSubroutine: s32'cmnUltimateShot_AtkInit'
 		damage: 250
 		minimalDamagePercent: 30
 		hitstop: 1
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 60000
 		attackYKnockback: 13000
 		attackYSuction: 1, 10000, 3000000, 100, 0
 		airHitstun: 30
 		wallBounceCount: 0
 		atkRange: 50000000, -550000, 0, 0
 		atkSousaiSevereCheck: 1
 		atkEffectBeamSousai: 1
 		atkIgnoreSousaiSameLevel: 1
 		attackDamageEffect: 4, s32'cmn_hit_sp2'
 		upon: (SIGNAL7) {
 			wallBounceCount: 1
 			wallBounceXVelocityPercent: 0
 			attackYSuction: 0, 0, 0, 0, 0
 			airHitstun: 21
 			enemyHitstopAddition: 0, 9, 9
 			ifVarOfObjectFalse: (PLAYER), var(PlayerDamage) {
 				enableHitbox: 
 			} endIf: 
 			else:  {
 				recoveryState: 
 			} endElse: 
 			ifVarOfObjectOperation: (IS_GREATER_OR_EQUAL), (PARENT), var(act2), int(8) {
 				recoveryState: 
 			} endIf: 
 			copyVarFromObject: (SELF), var(Result), (PLAYER), var(act2)
 			if: var(Result) {
 				enemyHitstopAddition: 9, 9, 9
 			} endIf: 
 		} endUpon: 
 		copyVarFromObject: (SELF), var(Result), (PLAYER), var(act2)
 		if: var(Result) {
 			damage: 380
 			chipDamagePercent: 50
 		} endIf: 
 		atkHitCommonBankSE: 16, s32'ARC_BTL_CMN_Hit_BeamLoop_LP', 0
 		atkGuardCommonBankSE: 17, s32'ARC_BTL_CMN_Grd_BeamLoop_LP', 0
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectAngle: 2
 		setActionTimeMax: 60
 		autoHitSignalSending: 1
 		worldOutToEliminate: 1
 		linkParticle: s32'pcn431_naka00'
 		upon: (SIGNAL0) {
 			clearRegisteredUponCode: (SIGNAL0)
 			createObject: s32'Makankou_end', (ZERO)
 			copyVar: var(act1), int(1)
 			copyVarFromObject: (PARENT), var(act1), (SELF), int(1)
 		} endUpon: 
 		upon: (ON_ATTACK_COLLISION_SOUSAI) {
 			copyVarFromObject: (PARENT), var(ObjectParameter1), (SELF), var(ObjectParameter1)
 			deactivateObject: (SELF)
 			sendSignal: (PARENT), (SIGNAL1)
 		} endUpon: 
 		upon: (SIGNAL6) {
 			copyVar: var(act0), int(0)
 		} endUpon: 
 		upon: (ATTACK_COLLISION) {
 			if: var(act0) {
 				sendSignal: (PARENT), (SIGNAL0)
 			} endIf: 
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			callSubroutine: s32'cmnUltimate_CameraCombo'
 			if: var(ObjectParameter2) {
 				checkCurrentStateName: (PLAYER), s32'Makankou'
 				if: var(Result) {
 					copyVarFromObject: (PLAYER), var(act0), (SELF), int(0)
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			ifVarOfObjectTrue: (PARENT), var(act2) {
 				ifVarOfObjectFalse: (PARENT), var(act4) {
 					copyVarFromObject: (SELF), var(Result), (PARENT), var(ObjectParameter1)
 					ifOperation: (IS_LESS_OR_EQUAL), var(ObjectParameter1), var(Result) {
 						deactivateObject: (SELF)
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			effectForLandBreak: 2, 1200, 0, 0
 			copyVarFromObject: (SELF), var(Result), (ENEMY), var(PlayerDamageOrGuard)
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Makankou'
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			if: var(act1) {
 				copyVar: var(act1), int(0)
 				copyVarFromObject: (PARENT), var(act1), (SELF), int(0)
 			} endIf: 
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			createParticle: s32'pcn431_naka01', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'Makankou', 1
 	enableHitbox: 
 	setSpeedByCurrentAngle: 100000, 0
 	label: s32'loop'
 	sprite: s32'Makankou', 1
 	atkIgnoreSousaiSameLevel: 0
 	sprite: s32'Makankou', 2
 	scaleForLinkY: 1200
 	sprite: s32'Makankou', 1
 	scaleForLinkY: 1000
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'Makankou_end' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		setActionTimeMax: 60
 		moveToPosType: (PARENT), (EX_POINT_0)
 		worldOutToEliminate: 1
 		linkParticle: s32'pcn431_end00'
 		upon: (IDLING) {
 			checkIsObjectActive: (PARENT)
 			if: var(Result) {
 				copyVarFromParent: var(PosZ), var(PosZ)
 				moveToPosType: (PARENT), (EX_POINT_0)
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			setArg_multiplyScale: int(1000)
 			setArg_addAngle: int(0)
 			createParticle: s32'pcn431_end01', (ZERO)
 		} endUpon: 
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 2
 	scaleForLinkY: 1000
 	sprite: s32'null', 2
 	scaleForLinkY: 1200
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(120)
 } endState: 
 
 beginState: s32'groundSmokeMakan' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: 150000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_scale: int(1200)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	spriteEnd: 
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'GekiretukoudanAtk' {
 	upon: (IMMEDIATE) {
 		overdriveAttack: 
 		callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 		callSubroutine: s32'cmnUltimateShot_AtkInit'
 		damage: 450
 		damageScalingOverride: 500
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 10000
 		attackYKnockback: 6000
 		wallBounceCount: 0
 		minimalDamagePercent: 37
 		hitstop: 1
 		attackXSuctionBaseObject: (PREVIOUS), (PLAYER_LAND), 250000, 100000, 100, 0
 		attackYSuctionBaseObject: (PREVIOUS), (PLAYER_LAND), 50000, 100000, 100, 0
 		airHitstun: 60
 		atkFrontDirection: 0
 		disableOpponentRecovery: 0
 		atkDamageToRecovery: 1
 		attackDamageEffect: 4, s32'cmn_hit_sp2'
 		atkAdditionalHitCount: 2
 		atkShakeCamera: s32'ShakeTate', 1800, 0, 40, 10
 		atkHitCommonBankSE: 16, s32'ARC_BTL_CMN_Hit_BeamLoop_LP', 0
 		atkGuardCommonBankSE: 17, s32'ARC_BTL_CMN_Grd_BeamLoop_LP', 0
 		setActionTimeMax: 120
 		linkParticle: s32'pcn432_jizoku'
 		percentageYScale: 150
 		percentageXScale: 150
 		createObject: s32'gekiretuLoop_groundsmoke', (ZERO)
 		upon: (AFTER_EXIT) {
 			clearRegisteredUponCode: (AFTER_EXIT)
 			copyVar: var(act1), int(1)
 		} endUpon: 
 		upon: (DESTROY) {
 			deactivateObject: (SELF)
 			deactivateObjectByName: s32'gekiretuLoop_groundsmoke'
 			effectForLandBreak: 3, 1000, 0, 0
 			setEX_POINT_F: (ZERO), 200000, 0
 			createParticle: s32'pcn432_end', (EX_POINT_F)
 			addWindPowerOffset: 0, 0, 0, 1000, 0, 100, 30
 			destructionCollision: 0, 0, 0, 10000
 		} endUpon: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			callSubroutine: s32'cmnUltimate_CameraCombo'
 		} endUpon: 
 		upon: (ON_ATTACK_COLLISION_SOUSAI) {
 			ifNot: var(act0) {
 				pushXSpeed: 
 				pushYSpeed: 
 			} endIf: 
 			copyVar: var(act2), int(0)
 			upon: (FRAME_STEP_AFTER_EX) {
 				ifNot: var(CheckStop) {
 					modifyVar: (ADD), var(act2), int(1)
 					ifOperation: (IS_GREATER_OR_EQUAL), var(act2), int(5) {
 						clearRegisteredUponCode: (FRAME_STEP_AFTER_EX)
 						resetSpeed: 1
 						popXSpeed: 
 						popYSpeed: 
 						xSpeedPerFrame: 110
 					} endIf: 
 				} endIf: 
 			} endUpon: 
 			damage: 225
 			setXSpeed: 30000
 			addXPosition: -180000
 			xSpeedPerFrame: 110
 			enableHitbox: 
 			modifyVar: (ADD), var(act0), int(1)
 			ifOperation: (IS_GREATER_OR_EQUAL), var(act0), int(16) {
 				requestDestroy: (SELF)
 				ifNot: var(act1) {
 					checkCurrentStateName: (PLAYER), s32'Gekiretukoudan'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL0)
 					} endIf: 
 				} endIf: 
 			} endIf: 
 			else:  {
 				ifNot: var(act1) {
 					checkCurrentStateName: (PLAYER), s32'Gekiretukoudan'
 					if: var(Result) {
 						sendSignal: (PLAYER), (SIGNAL1)
 					} endIf: 
 				} endIf: 
 			} endElse: 
 		} endUpon: 
 		upon: (ATTACK_COLLISION_PLAYER) {
 			createObject: s32'gekiretuHit_groundsmoke', (ZERO)
 			if: var(IsMainPlayerAttackedEnemy) {
 				percentageXSpeed: 10
 			} endIf: 
 			cameraShake: s32'ShakeTate', 1800, 0, 40, 10
 			if: var(IsMainPlayerAttackedEnemy) {
 				modifyVar: (ADD), var(act4), int(1)
 				ifNot: var(act5) {
 					copyVar: var(act5), int(1)
 					copyVar: var(act4), int(1)
 				} endIf: 
 				ifNot: var(act0) {
 					damage: 450
 				} endIf: 
 			} endIf: 
 			else:  {
 				ifNot: var(act5) {
 					modifyVar: (ADD), var(act4), int(1)
 				} endIf: 
 				ifVarOfObjectTrue: (DEFENDER), var(IsSubMember1Self) {
 					ifNot: var(act6) {
 						ifNot: var(act5) {
 							copyVar: var(act4), int(1)
 						} endIf: 
 					} endIf: 
 					ifNot: var(act0) {
 						damage: 450
 					} endIf: 
 					modifyVar: (ADD), var(act6), int(1)
 					ifOperation: (IS_GREATER), var(act6), int(10) {
 						damage: 0
 					} endIf: 
 				} endIf: 
 				ifVarOfObjectTrue: (DEFENDER), var(IsSubMember2Self) {
 					ifNot: var(act7) {
 						ifNot: var(act5) {
 							copyVar: var(act4), int(1)
 						} endIf: 
 					} endIf: 
 					ifNot: var(act0) {
 						damage: 450
 					} endIf: 
 					modifyVar: (ADD), var(act7), int(1)
 					ifOperation: (IS_GREATER), var(act7), int(10) {
 						damage: 0
 					} endIf: 
 				} endIf: 
 			} endElse: 
 			ifOperation: (IS_GREATER_OR_EQUAL), var(act4), int(10) {
 				requestDestroy: (SELF)
 				groundKnockbackEffect: (SPIN)
 				airKnockbackEffect: (SPIN)
 				attackXKnockback: 50000
 				attackYKnockback: 35000
 				attackXSuctionBaseObject: (NULL), (ZERO), 0, 0, 100, 0
 				attackYSuctionBaseObject: (NULL), (ZERO), 0, 0, 100, 0
 				atkAdditionalHitCount: 3
 				applyAttackLockWithStateName: s32'CmnNeutral'
 				checkCurrentStateName: (PLAYER), s32'Gekiretukoudan'
 				if: var(Result) {
 					sendSignal: (PLAYER), (SIGNAL0)
 				} endIf: 
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			effectForLandBreak: 3, 1000, 0, 0
 			copyVarFromObject: (SELF), var(Result), (ENEMY), var(PlayerDamageOrGuard)
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Gekiretukoudan'
 			ifNot: var(Result) {
 				stopBankSE: 16, 20
 				stopBankSE: 17, 20
 			} endIf: 
 			ifVarOfObjectTrue: (PLAYER), var(Direction) {
 				ifVarOfObjectTrue: (ENEMY_MAIN_PLAYER), var(Direction) {
 					disableAttackClashing: 1
 				} endIf: 
 				else:  {
 					disableAttackClashing: 0
 				} endElse: 
 			} endIf: 
 			ifVarOfObjectFalse: (PLAYER), var(Direction) {
 				ifVarOfObjectFalse: (ENEMY_MAIN_PLAYER), var(Direction) {
 					disableAttackClashing: 1
 				} endIf: 
 				else:  {
 					disableAttackClashing: 0
 				} endElse: 
 			} endIf: 
 		} endUpon: 
 		upon: (BEFORE_EXIT) {
 			stopBankSE: 16, 20
 			stopBankSE: 17, 20
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'Gekiretukoudan', 1
 	setXSpeed: 30000
 	setAcceleration: 5000
 	enableHitbox: 
 	sprite: s32'Gekiretukoudan', 2
 	setWallCollision: 0
 	label: s32'loop'
 	sprite: s32'Gekiretukoudan', 3
 	enableHitbox: 
 	spriteEnd: 
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'gekiretu_sphereEff' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'pcn432_tame00'
 		addYPosition: 200000
 		addXPosition: 100000
 		addZPosition: 40000
 		setScale: 600
 		jumpToStateInObject: (SELF), s32'1'
 		jumpToStateInObject: (ATTACKER), s32'2'
 	} endUpon: 
 	sprite: s32'keep', 37
 	sprite: s32'keep', 2147483647
 	setScale: 10
 	spriteEnd: 
 	label: s32'1'
 	sprite: s32'keep', 4
 	setScale: 500
 	addXPosition: 40000
 	addYPosition: 55000
 	addZPosition: -25000
 	sprite: s32'keep', 4
 	addXPosition: 80000
 	sprite: s32'keep', 2
 	addXPosition: 80000
 	addZPosition: -15000
 	sprite: s32'keep', 2
 	addScale: 20
 	sprite: s32'keep', 2147483647
 	addScale: 20
 	spriteEnd: 
 	label: s32'2'
 	sprite: s32'null', 60
 	linkParticle: s32'pcn432_tame01'
 	addXPosition: -80000
 	addScale: 300
 	spriteEnd: 
 } endState: 
 
 beginState: s32'gekiretu_AuraEff' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		jumpToStateInObject: (SELF), s32'fire'
 		jumpToStateInObject: (ATTACKER), s32'End'
 		addXPosition: -100000
 		linkParticleArg: s32'pcn_432_BodyAura_loop'
 		setScale: 500
 	} endUpon: 
 	sprite: s32'keep', 40
 	createObject: s32'Gekiretu_auraGround', (ZERO)
 	label: s32'loop'
 	sprite: s32'keep', 20
 	createParticle: s32'cmn_Saura_nokosi', (ZERO)
 	setScale: 800
 	gotoLabel: s32'loop'
 	label: s32'fire'
 	sprite: s32'keep', 1
 	setAngleY: 0
 	addXPosition: 100000
 	label: s32'fireloop'
 	sprite: s32'keep', 20
 	createParticle: s32'cmn_Saura_nokosi', (ZERO)
 	gotoLabel: s32'fireloop'
 	label: s32'End'
 	sprite: s32'keep', 30
 	linkObjectChangeToEliminate: (NULL)
 	linkObjectPosition: (NULL)
 	linkParticleArg: s32'pcn_432_BodyAura_End'
 	triggerUponInObject: s32'Gekiretu_auraGround', (SIGNAL0)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'Gekiretu_auraGround' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		addYPosition: 40000
 		linkParticleArg: s32'cmn_Saura_ground'
 		jumpToStateInObject: (SELF), s32'Del'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	setColorAlpha: 255
 	setScale: 1200
 	spriteEnd: 
 	label: s32'Del'
 	sprite: s32'null', 20
 	setScaleSpeed: -50
 	spriteEnd: 
 } endState: 
 
 beginState: s32'gekiretucutscene_groundsmoke' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -150000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_setYAngle: int(75000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'gekiretucutscene_groundsmoke2' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -150000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 10
 	setArg_setYAngle: int(-75000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'gekiretuLoop_groundsmoke' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -75000
 		addYPosition: -300000
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 5
 	setArg_scale: int(1600)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'gekiretuHit_groundsmoke' {
 	upon: (IMMEDIATE) {
 		setYPosition: 0
 		linkObjectStop: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		addXPosition: -75000
 		flipDirection: 
 	} endUpon: 
 	sprite: s32'null', 20
 	setArg_scale: int(1300)
 	setArg_setYAngle: int(-30000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 	sprite: s32'null', 20
 	setArg_scale: int(1300)
 	setArg_setYAngle: int(-30000)
 	createParticleWithArg: s32'bg_undersmoke00', (ZERO)
 } endState: 
 
 beginState: s32'FinishCamera_NmlAtk2C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraMotion: -121129, 231088, 62781, 2787, 6168, 0, 72, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_NmlAtkAir2C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraMotion: 1076, 413344, 71645, 1136, 1667, 0, 54, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_MakuuHouidan' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 3
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	moveCameraEx: (LOSER), -308384, 481512, 187997, -1600, 6261, 0, 54, 0, 180, 0, 0, 0, 0
 	sprite: s32'null', 3
 	moveCameraEx: (LOSER), -287293, 450706, 182224, -1600, 6261, 0, 54, 1, 180, 0, 0, 0, 0
 	moveCameraEx: (LOSER), -308384, 481512, 187997, -1600, 6261, 0, 54, 1, 180, 0, 0, 0, 0
 	sprite: s32'null', 3
 	moveCameraEx: (LOSER), -287293, 450706, 182224, -1600, 6261, 0, 54, 1, 180, 0, 0, 0, 0
 	sprite: s32'null', 40
 	moveCameraEx: (LOSER), -251313, 398154, 172337, -1600, 6261, 0, 54, 1, 180, 0, 0, 0, 0
 	sprite: s32'null', 42
 	targetCamera: (WINNER), (LOSER), 0, 150000, 250000, 0, 150000, 0, -200000, 50, 72, 33, 40, 10, 0, 2, 2
 	deactivateObjectByName: s32'cmn_syutyusen'
 	spriteEnd: 
 	exitState: 
 	beginState: s32'pcn_EntryEff_Debris' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'bg_pcn600_debris'
 	} endUpon: 
 	sprite: s32'null', 151
 	sprite: s32'null', 1
 	createObject: s32'pcn_EntryEff_SpeedLine', (ZERO)
 } endState: 
 
 beginState: s32'pcn_EntryEff_SpeedLine' {
 	upon: (IMMEDIATE) {
 		flipDirection: 
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'pcn600_SpeedLine'
 		setZPosition: 500000
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'pcn_WinEff' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'cmn_warp'
 		setScale: 1500
 		addXPosition: -75000
 	} endUpon: 
 	sprite: s32'null', 30
 } endState: 
 
 beginState: s32'DUMMYDUMMY' {
 	sprite: s32'null', 1
 } endState: 
 
