--- 1.32/BBS_HTNEF.bbs	2023-05-24 20:54:47.457009700 +0200
+++ 1.33/BBS_HTNEF.bbs	2023-08-23 04:23:02.922919039 +0200
@@ -1,1606 +1,1605 @@
 beginState: s32'dmy' {
 	sprite: s32'null', 1
 } endState: 
 
 beginState: s32'okasi' {
 	upon: (IMMEDIATE) {
 		callSubroutine: s32'cmn_okasi'
 	} endUpon: 
 	sprite: s32'null', 2147483647
 	linkParticle: s32'htn_okasi'
 } endState: 
 
 beginState: s32'htn_600_hibi01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn600_hibi_eff01'
 } endState: 
 
 beginState: s32'htn_600_hibi02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn600_tokibg01'
 } endState: 
 
 beginState: s32'bg_htn600_smoke01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bg_htn600_smoke01'
 } endState: 
 
 beginState: s32'htn600_warp01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn600_warp01'
 } endState: 
 
 beginState: s32'htn600_smoke02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bg_dash_default'
 } endState: 
 
 beginState: s32'htn600_smoke_temae01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 120
 	linkParticle: s32'bg_htn600_smoke02'
 } endState: 
 
 beginState: s32'htn600_smoke_temae021' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		setAngleY: -45000
 		addXPosition: 200000
 		addZPosition: 200000
 		addYPosition: 10000
 		setScale: 300
 		setXSpeed: 700
 		setZSpeed: -700
 		setYSpeed: -65
 		linkParticle: s32'bg_smokeLoop00'
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 5
 	addXSpeed: -30
 	addZSpeed: 30
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'htn600_smoke_temae022' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		setAngleY: -45000
 		addXPosition: -150000
 		addZPosition: -150000
 		addYPosition: 10000
 		setScale: 1000
 		setXSpeed: 700
 		setZSpeed: -700
 		setYSpeed: -300
 		linkParticle: s32'bg_smokeLoop00'
 	} endUpon: 
 	label: s32'loop'
 	sprite: s32'null', 5
 	addXSpeed: -30
 	addZSpeed: 30
 	gotoLabel: s32'loop'
 } endState: 
 
 beginState: s32'htn_toki_hit01' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectPosition: (ENEMY)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn_toki_hit01'
 		addXPosition: 50000
 		ignoreFinishStop: 1
 	} endUpon: 
 	sprite: s32'null', 1
 	spriteEnd: 
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_toki_hit01_60' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectPosition: (ENEMY)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn_toki_hit01'
 		setAngle: 60000
 		addXPosition: 50000
 		ignoreFinishStop: 1
 	} endUpon: 
 	sprite: s32'null', 1
 	spriteEnd: 
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_toki_hit01_5AA3' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectPosition: (ENEMY)
 		linkObjectStop: (NULL)
 		ignoreFinishStop: 1
 		linkParticle: s32'htn_toki_hit02'
 	} endUpon: 
 	sprite: s32'null', 1
 	spriteEnd: 
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_toki_hit01_5C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectPosition: (ENEMY)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn_toki_hit02'
 		setAngle: 7000
 		ignoreFinishStop: 1
 	} endUpon: 
 	sprite: s32'null', 1
 	spriteEnd: 
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_toki_hit01_yobidousa' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectPosition: (ENEMY)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn402_toki_hit01'
 		ignoreFinishStop: 1
 	} endUpon: 
 	sprite: s32'null', 1
 	spriteEnd: 
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_toki_hit01_bougyomusi' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectPosition: (ENEMY)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn_toki_hit02'
 		setAngle: 60000
 		ignoreFinishStop: 1
 	} endUpon: 
 	sprite: s32'null', 1
 	spriteEnd: 
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_toki_hit01_cold_eye' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectPosition: (ENEMY)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn407_toki_hit01'
 	} endUpon: 
 	sprite: s32'null', 1
 	spriteEnd: 
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_toki_hit01_hakki' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByAll: 1
 		linkObjectPosition: (ENEMY)
 		linkParticle: s32'htn430_toki_hit01'
 	} endUpon: 
 	sprite: s32'null', 1
 	spriteEnd: 
 	sprite: s32'null', 60
 	linkObjectPosition: (NULL)
 	characterSfx: s32'ARC_BTL_HTN_Hakki_ExeHit'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_toki_hit01_Seichowo' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectStop: (NULL)
 		addXPosition: 50000
 	} endUpon: 
 	sprite: s32'null', 2
 	sprite: s32'null', 60
 	linkParticle: s32'htn432_toki_hit01'
 	moveToPosType: (ENEMY_MAIN_PLAYER), (ABDOMINAL)
 } endState: 
 
 beginState: s32'htn_toki_hit01_Seichowo_sub' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectPosition: (NULL)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn432_toki_hit01_sub'
 		setAngle: 7000
 		ignoreFinishStop: 1
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'htn_5Beff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn203_eff01'
 } endState: 
 
 beginState: s32'htn_2Beff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn232_eff01'
 } endState: 
 
 beginState: s32'htn_Air5Beff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn261_eff01'
 } endState: 
 
 beginState: s32'htn_atemi_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PLAYER)
 		linkObjectPosition: (PLAYER)
 		linkObjectStop: (PLAYER)
 		jumpToStateInObject: (SELF), s32'end'
 		upon: (SIGNAL1) {
 			clearRegisteredUponCode: (SIGNAL1)
 			linkObjectPosition: (NULL)
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn205_toki_aura01'
 	label: s32'end'
 	sprite: s32'null', 30
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	linkObjectStop: (NULL)
 	ignoreWorldStopByAll: 1
 	linkObjectChangeToEliminate: (NULL)
 	linkParticle: s32'htn205_toki_aura01_end'
 } endState: 
 
 beginState: s32'htn_202_hit_camera03' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectStop: (NULL)
 		ignoreFinishStop: 1
 		linkParticle: s32'htn_toki_hit02'
 		moveToPosType: (PLAYER), (EX_POINT_0)
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'htn_204_hit_camera' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn_toki_hit02'
 		setAngle: 7000
 		ignoreFinishStop: 1
 		moveToPosType: (PLAYER), (EX_POINT_0)
 	} endUpon: 
 	sprite: s32'null', 60
 	addZPosition: -30000
 } endState: 
 
 beginState: s32'htn_205_hit_camera' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn_toki_hit01'
 		ignoreFinishStop: 1
 		moveToPosType: (PLAYER), (EX_POINT_0)
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'htn_403_hit_camera' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn_toki_hit02'
 		setAngle: 60000
 		moveToPosType: (PLAYER), (EX_POINT_0)
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'htn_401_hit_camera' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectStop: (NULL)
 		ignoreFinishStop: 1
 		moveToPosType: (PLAYER), (EX_POINT_0)
 	} endUpon: 
 	sprite: s32'null', 1
 	sprite: s32'null', 60
 	linkParticle: s32'htn_toki_hit01'
 } endState: 
 
 beginState: s32'htn_402_hit_camera' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn402_toki_hit01'
 		ignoreFinishStop: 1
 		moveToPosType: (PLAYER), (EX_POINT_0)
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'htn_403_hit_camera' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		linkObjectChangeToEliminate: (NULL)
 		linkObjectStop: (NULL)
 		linkParticle: s32'htn_toki_hit02'
 		setAngle: 60000
 		moveToPosType: (PLAYER), (EX_POINT_0)
 	} endUpon: 
 	sprite: s32'null', 60
 } endState: 
 
 beginState: s32'htn_hajiki' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn_hajiki'
 } endState: 
 
 beginState: s32'htn_400_aura01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		if: var(ObjectParameter1) {
 			linkObjectPosition: (NULL)
 			upon: (NO_STOP_IDLING) {
 				copyVarFromParent: var(PosXRaw), var(PosXRaw)
 				copyVarFromParent: var(PosZ), var(PosZ)
 				copyVarFromObject: (SELF), var(act0), (PLAYER), var(PlayerArmorHit)
 				ifOperation: (IS_GREATER_OR_EQUAL), var(PlayerArmorHit), int(0) {
 					modifyVar: (DIV), var(act0), int(600000)
 					modifyVar: (ADD), var(act0), int(-1)
 					ifOperation: (IS_LESS_OR_EQUAL), var(PlayerArmorHit), int(600000) {
 						modifyVar: (MUL), var(act0), int(50000)
 						modifyVar: (SUB), var(PosXRaw), var(act0)
 					} endIf: 
 					else:  {
 						modifyVar: (MUL), var(act0), int(30000)
 						modifyVar: (SUB), var(PosXRaw), var(act0)
 					} endElse: 
 				} endIf: 
 				else:  {
 					addXPosition: 70000
 				} endElse: 
 			} endUpon: 
 		} endIf: 
 	} endUpon: 
 	sprite: s32'null', 90
 	linkParticle: s32'htn400_eff01'
 } endState: 
 
 beginState: s32'htn_401_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn401_eff01'
 } endState: 
 
 beginState: s32'htn_401_eff01_hit' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn401_toki_hit01'
 } endState: 
 
 beginState: s32'htn_402_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: 20000
 		percentageXScale: 120
 		addYPosition: 120000
 		percentageYScale: 60
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn402_eff01'
 } endState: 
 
 beginState: s32'htn_403_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	addXPosition: 20000
 	percentageYScale: 150
 	addZPosition: -30000
 	linkParticle: s32'htn403_eff01'
 } endState: 
 
 beginState: s32'236CHaseiA_atk' {
 	upon: (IMMEDIATE) {
 		setActionTimeMax: 300
 		linkParticle: s32'htn404_eff01'
 		setLinkParticleNotStop: 1
 		addXPosition: 200000
 		addYPosition: 100000
 	} endUpon: 
 	sprite: s32'236CHaseiA_atk', 10
 	sprite: s32'null', 6
 	sprite: s32'null', 20
 	setLinkParticleNotStop: 0
 } endState: 
 
 beginState: s32'htn_405_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn405_eff01'
 } endState: 
 
 beginState: s32'236CHasei8A_atk' {
 	upon: (IMMEDIATE) {
 		setActionTimeMax: 300
 		linkParticle: s32'htn404_eff01'
 		setLinkParticleNotStop: 1
 		addXPosition: 100000
 		addYPosition: 200000
 		setAngle: 60000
 	} endUpon: 
 	sprite: s32'236CHaseiNaname_atk', 10
 	sprite: s32'null', 6
 	sprite: s32'null', 20
 	setLinkParticleNotStop: 0
 } endState: 
 
 beginState: s32'htn_406_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn406_eff01'
 } endState: 
 
 beginState: s32'236CHasei2A_atk' {
 	upon: (IMMEDIATE) {
 		setActionTimeMax: 300
 		linkParticle: s32'htn404_eff01'
 		setLinkParticleNotStop: 1
 		addXPosition: 50000
 		addYPosition: -100000
 		setAngle: -60000
 	} endUpon: 
 	sprite: s32'236CHaseiNaname_atk', 10
 	sprite: s32'null', 6
 	sprite: s32'null', 20
 	setLinkParticleNotStop: 0
 } endState: 
 
 beginState: s32'Cold_Eye_Matome' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PLAYER)
 		linkObjectChangeToEliminate: (PLAYER)
 		setWallCollision: 1
 		copyVar: var(act0), int(200000)
 		upon: (BEFORE_EXIT) {
 			stopBankSE: 0, 20
 		} endUpon: 
 	} endUpon: 
 	sprite: s32'null', 1
 	sprite: s32'null', 17
 	spriteOverrideTime: 21, var(ObjectParameter1)
 	spriteOverrideTime: 15, var(ObjectParameter2)
 	setEX_POINT_F: (PLAYER_LAND), 0, 0
 	createObject: s32'Cold_Eye_Range', (EX_POINT_F)
 	privateBankSE: 0, s32'ARC_BTL_HTN_ColdEye_Ice', 100
 	spriteEnd: 
 	gotoLabel: s32'loop'
 	label: s32'loop'
 	sprite: s32'null', 15
 	setEX_POINT_F: (PLAYER_LAND), 0, 0
 	modifyVar: (ADD), var(act1), var(act0)
 	addEX_POINT_F: var(act1), 0, 0
 	createObject: s32'Cold_Eye_Range', (EX_POINT_F)
 	setEX_POINT_F: (PLAYER_LAND), 0, 0
 	modifyVar: (MUL), var(act1), int(-1)
 	addEX_POINT_F: var(act1), 0, 0
 	createObject: s32'Cold_Eye_Range', (EX_POINT_F)
 	applyTo: (PREVIOUS) {
 		flipDirection: 
 	} endApply: 
 	modifyVar: (MUL), var(act1), int(-1)
 	spriteEnd: 
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(LoopCount), int(300)
 } endState: 
 
 beginState: s32'Cold_Eye_Range' {
 	upon: (IMMEDIATE) {
 		specialAttack: 
 		callSubroutine: s32'cmnSpecialShot_AtkInit'
 		callSubroutine: s32'cmnAtkTemplNageLand'
 		callSubroutine: s32'cmn_KuzushiHosei'
 		canGrab: 0
 		setProjectileDurability: 0
 		hitstop: 20
 		groundKnockbackEffect: (FORCE_STANDLOW)
 		atkFrontDirection: 1
 		attackKillType: (CANNOT_KILL)
 		atkRange: 250000, -300000, 0, 0
 		atkHitPositionType: 4, 50000, 0
 		atkShakeCamera: s32'ShakeTate', 1500, 1, 20, 0
 		linkObjectStop: (PLAYER)
 		linkParticle: s32'htn_field_Loop'
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_PRE_CTRL) {
 			clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_PRE_CTRL)
 			ifVarOfObjectOperation: (IS_LESS_OR_EQUAL), (PLAYER), var(EnemyFrontDistance), int(1000) {
 				atkDirectionReverse: 1
 				copyVar: var(act0), int(1)
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Cold_Eye_A'
 			if: var(Result) {
 				sendSignalName: (PLAYER), s32'Cold_Eye_A_Hit', 0
 				copyVarFromObject: (PLAYER), var(act0), (SELF), var(act0)
 				jumpToLabel: s32'hit'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Cold_Eye_B'
 			if: var(Result) {
 				sendSignalName: (PLAYER), s32'Cold_Eye_B_Hit', 0
 				copyVarFromObject: (PLAYER), var(act0), (SELF), var(act0)
 				jumpToLabel: s32'hit'
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Cold_Eye_C'
 			if: var(Result) {
 				sendSignalName: (PLAYER), s32'Cold_Eye_C_Hit', 0
 				copyVarFromObject: (PLAYER), var(act0), (SELF), var(act0)
 				jumpToLabel: s32'hit'
 			} endIf: 
 		} endUpon: 
 		jumpToStateInObject: (SELF), s32'end'
 		jumpToStateInObject: 47, s32'end'
 		jumpToStateInObject: (ATTACKER), s32'atk'
 		jumpToStateInObject: (DEFENDER), s32'hit'
 		initializeDestroy: 
 		numberOfHits: 1
 		destroyOnPlayerDamage: 1
 		jumpToStateInObject: 45, s32'end'
 	} endUpon: 
 	sprite: s32'cold_eye', 200
 	label: s32'atk'
 	sprite: s32'cold_eye', 4
 	if: var(IN_BATTLE) {
 		ifNot: var(CameraCombo0) {
 			enableHitbox: 
 		} endIf: 
 	} endIf: 
 	sprite: s32'null', 2
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'hit'
 	sprite: s32'null', 200
 	clearRegisteredUponCode: (SIGNAL1)
 	clearRegisteredUponCode: (SIGNAL2)
 	clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_PRE_CTRL)
 	grabOrRelease: 
 	checkCurrentStateName: (PLAYER), s32'Cold_Eye_A'
 	if: var(Result) {
 		applyAttackLockWithStateName: s32'Cold_Eye_Atk'
 	} endIf: 
 	checkCurrentStateName: (PLAYER), s32'Cold_Eye_B'
 	if: var(Result) {
 		applyAttackLockWithStateName: s32'Cold_Eye_Atk'
 	} endIf: 
 	checkCurrentStateName: (PLAYER), s32'Cold_Eye_C'
 	if: var(Result) {
 		applyAttackLockWithStateName: s32'Cold_Eye_Atk_C'
 	} endIf: 
 	spriteEnd: 
 	label: s32'end'
 	sprite: s32'null', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	clearRegisteredUponCode: (SIGNAL2)
 	clearRegisteredUponCode: (AFTER_EXIT)
 	clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE_PRE_CTRL)
 	clearRegisteredUponCode: (DESTROY)
 	recoveryState: 
 	linkObjectStop: (NULL)
 	ignoreWorldStopByAll: 1
 	sprite: s32'null', 20
 	characterSfx: s32'ARC_BTL_HTN_ColdEye_End'
 	linkParticle: s32'htn_field_End'
 } endState: 
 
 beginState: s32'Cold_Eye_Atk' {
 	upon: (IMMEDIATE) {
 		specialAttack: 
 		callSubroutine: s32'cmnAtkLevel_3_AtkInit'
 		callSubroutine: s32'cmnSpecialShot_AtkInit'
 		callSubroutine: s32'cmnAtkTemplNageExe'
 		callSubroutine: s32'cmn_KuzushiHosei'
 		damage: 230
 		tensionGainOnConnect: 250
 		airHitstun: 50
 		groundKnockbackEffect: (ExDamageBallMof)
 		airKnockbackEffect: (ExDamageBallMof)
 		atkDamageCell: 24
 		attackXKnockback: 0
 		attackYKnockback: 0
 		attackGuardtype: (NONE)
 		atkFrontDirection: 2
 		attackDamageEffect: 1, s32''
 		applyAttackLockWithStateName: s32'Cold_Eye_Atk'
 		attackKillType: (CANNOT_KILL)
 		damageSoundEffect: s32'dammy'
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'htn407_toki_hit01'
 		setLinkParticleNotStop: 1
 		autoHitSignalSending: 1
 		worldOutToEliminate: 1
 		faceOpponent: 
 		copyVar: var(CameraCombo0), int(1)
 		jumpToStateInObject: (SELF), s32'finish'
 	} endUpon: 
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 2
 	enableHitbox: 
 	characterSfx: s32'ARC_BTL_HTN_ColdEye_Hit'
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 16
 	enableHitbox: 
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 9
 	enableHitbox: 
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 2
 	enableHitbox: 
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 16
 	enableHitbox: 
 	sprite: s32'cold_eye_tadan', 3
 	groundKnockbackEffect: (FORCE_STAND)
 	airKnockbackEffect: (FORCE_STAND)
 	enableHitbox: 
 	sprite: s32'null', 60
 	label: s32'finish'
 	sprite: s32'cold_eye_tadan', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	enableHitbox: 
 	attackKillType: (CAN_KILL)
 	hitstop: 20
 	groundKnockbackEffect: (CRUMPLE)
 	airKnockbackEffect: (CRUMPLE)
 	atkShakeCamera: s32'ShakeTate', 1000, 1, 20, 0
 	sprite: s32'cold_eye_tadan', 1
 	ifBeginAliveEnemyPlayer: 0 {
 		enableHitbox: 
 		damage: 0
 		tensionGainOnConnect: 0
 		atkHoseiTadan: 1
 		noHitstunScaling: 0
 		damageScalingOverride: 500
 		groundKnockbackEffect: (AIR_FACE_UP)
 		airKnockbackEffect: (AIR_FACE_UP)
 		attackXKnockback: 15000
 		attackYKnockback: 18000
 		dontAddComboCount: 1
 		applyAttackLockWithStateName: s32''
 		disableHitstopForSelf: 0
 		autoHitStopSending: 1
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			checkCurrentStateName: (PLAYER), s32'Cold_Eye_A'
 			if: var(Result) {
 				sendSignalName: (PLAYER), s32'Cold_Eye_A_Finish_Hit', 0
 			} endIf: 
 			checkCurrentStateName: (PLAYER), s32'Cold_Eye_B'
 			if: var(Result) {
 				sendSignalName: (PLAYER), s32'Cold_Eye_B_Finish_Hit', 0
 			} endIf: 
 		} endUpon: 
 	} endIf: 
 	else:  {
 		recoveryState: 
 	} endElse: 
 	sprite: s32'null', 60
 	recoveryState: 
 } endState: 
 
 beginState: s32'Cold_Eye_Atk_C' {
 	upon: (IMMEDIATE) {
 		specialAttack: 
 		callSubroutine: s32'cmnAtkLevel_3_AtkInit'
 		callSubroutine: s32'cmnSpecialShot_AtkInit'
 		callSubroutine: s32'cmnAtkTemplNageExe'
 		callSubroutine: s32'cmn_KuzushiHosei'
 		callSubroutine: s32'cmnSpecialAttackEx_ShotInit'
 		damage: 260
 		airHitstun: 50
 		groundKnockbackEffect: (ExDamageBallMof)
 		airKnockbackEffect: (ExDamageBallMof)
 		atkDamageCell: 24
 		attackXKnockback: 0
 		attackYKnockback: 0
 		attackGuardtype: (NONE)
 		atkFrontDirection: 2
 		attackDamageEffect: 1, s32''
 		applyAttackLockWithStateName: s32'Cold_Eye_Atk_C'
 		attackKillType: (CANNOT_KILL)
 		damageSoundEffect: s32'dammy'
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkParticle: s32'htn407_toki_hit01'
 		setLinkParticleNotStop: 1
 		autoHitSignalSending: 1
 		worldOutToEliminate: 1
 		faceOpponent: 
 		copyVar: var(CameraCombo0), int(1)
 		upon: (SIGNAL1) {
 			clearRegisteredUponCode: (SIGNAL1)
 			copyVar: var(act0), int(1)
 		} endUpon: 
 		jumpToStateInObject: (SELF), s32'finish'
 	} endUpon: 
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 2
 	enableHitbox: 
 	characterSfx: s32'ARC_BTL_HTN_ColdEye_Hit'
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 9
 	enableHitbox: 
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 16
 	enableHitbox: 
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 2
 	enableHitbox: 
 	sprite: s32'cold_eye_tadan', 3
 	atkDamageCell: 16
 	enableHitbox: 
 	sprite: s32'cold_eye_tadan', 3
 	groundKnockbackEffect: (FORCE_STAND)
 	airKnockbackEffect: (FORCE_STAND)
 	enableHitbox: 
 	sprite: s32'null', 60
 	label: s32'finish'
 	sprite: s32'cold_eye_tadan', 1
 	clearRegisteredUponCode: (SIGNAL0)
 	enableHitbox: 
 	attackKillType: (CAN_KILL)
 	hitstop: 20
 	groundKnockbackEffect: (CRUMPLE)
 	airKnockbackEffect: (CRUMPLE)
 	atkShakeCamera: s32'ShakeTate', 1000, 1, 20, 0
 	sprite: s32'cold_eye_tadan', 1
 	ifBeginAliveEnemyPlayer: 0 {
 		enableHitbox: 
 		damage: 0
 		tensionGainOnConnect: 0
 		atkHoseiTadan: 1
 		noHitstunScaling: 0
 		damageScalingOverride: 500
 		groundKnockbackEffect: (AIR_FACE_UP)
 		airKnockbackEffect: (AIR_FACE_UP)
 		attackXKnockback: 15000
 		attackYKnockback: 18000
 		dontAddComboCount: 1
 		applyAttackLockWithStateName: s32''
 		if: var(act0) {
 			attackXKnockback: 5000
 			attackYKnockback: 40000
 			groundKnockbackEffect: (KirimomiLand)
 			airKnockbackEffect: (KirimomiLand)
 			wallBounceXVelocityPercent: 0
 			slidingKnockdownDuration: 0
 		} endIf: 
 		disableHitstopForSelf: 0
 		autoHitStopSending: 1
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			checkCurrentStateName: (PLAYER), s32'Cold_Eye_C'
 			if: var(Result) {
 				sendSignalName: (PLAYER), s32'Cold_Eye_C_Finish_Hit', 0
 			} endIf: 
 		} endUpon: 
 	} endIf: 
 	else:  {
 		recoveryState: 
 	} endElse: 
 	sprite: s32'null', 60
 	recoveryState: 
 } endState: 
 
 beginState: s32'htn_407_lines01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn407_lines01'
 } endState: 
 
 beginState: s32'htn_407_lines02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn407_lines02'
 } endState: 
 
 beginState: s32'htn_408_eff01' {
 	upon: (IMMEDIATE) {
 		addXPosition: -80000
 	} endUpon: 
 	sprite: s32'null', 20
 	linkParticle: s32'htn408_eff01'
 	sprite: s32'null', 70
 	characterSfx: s32'ARC_BTL_HTN_ColdEye_End_Short'
 } endState: 
 
 beginState: s32'htn_430_E_Light' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 1000
 	linkParticle: s32'htn_430_E_Light'
 } endState: 
 
 beginState: s32'htn_430_tokibg01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addYPosition: 100000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn430_tokibg01'
 } endState: 
 
 beginState: s32'htn_430_tokibg02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addYPosition: 100000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn430_tokibg02'
 } endState: 
 
 beginState: s32'htn_430_aura01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 90
 	linkParticle: s32'htn430_eff01'
 } endState: 
 
 beginState: s32'htn_430_aura01_end' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn430_eff01_end'
 } endState: 
 
 beginState: s32'htn_430_aura01_air' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 90
 	linkParticle: s32'htn430_eff02'
 } endState: 
 
 beginState: s32'htn_430_aura01_air_end' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn430_eff02_end'
 } endState: 
 
 beginState: s32'htn_430_hibi01a' {
 	upon: (IMMEDIATE) {
 	} endUpon: 
 	sprite: s32'null', 25
 	linkParticle: s32'htn430_hibi_eff01'
 	sprite: s32'null', 35
 	characterSfx: s32'ARC_BTL_HTN_ColdEye_End'
 } endState: 
 
 beginState: s32'htn_430_hibi01b' {
 	upon: (IMMEDIATE) {
 		addXPosition: 150000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn430_hibi_eff01'
 } endState: 
 
 beginState: s32'htn_430_hibi01c' {
 	upon: (IMMEDIATE) {
 		addXPosition: 300000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn430_hibi_eff01'
 } endState: 
 
 beginState: s32'htn_430_atk_eff01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 3
 	linkParticle: s32'htn430_atk_eff01'
 	sprite: s32'null', 30
 	linkObjectPosition: (NULL)
 } endState: 
 
 beginState: s32'htn_430_slash01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addYPosition: 50000
 	} endUpon: 
 	sprite: s32'null', 3
 	linkParticle: s32'htn430_slash01'
 	sprite: s32'null', 30
 	linkObjectPosition: (NULL)
 } endState: 
 
 beginState: s32'htn_430_Air_Add' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (ENEMY)
 		linkObjectStop: (PARENT)
 		setAngle: -10000
 		setScale: 2000
 	} endUpon: 
 	sprite: s32'null', 3
 	linkParticle: s32'htn430_add'
 	sprite: s32'null', 30
 	linkObjectPosition: (NULL)
 } endState: 
 
 beginState: s32'htn_431_E_Light' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 1000
 	linkParticle: s32'htn_431_E_Light'
 } endState: 
 
 beginState: s32'htn_431_aura01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn431_aura01'
 } endState: 
 
 beginState: s32'koroshinoichigeki_hibi' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		ignoreWorldStopByAll: 1
 		ignoreFinishStop: 1
 	} endUpon: 
 	sprite: s32'null', 20
 	linkParticle: s32'htn431_hibi01'
 	sprite: s32'null', 2
 	linkObjectStop: (PLAYER)
 	ignoreWorldStopByAll: 0
 	ignoreFinishStop: 0
 	sprite: s32'null', 200
 	linkObjectStop: (NULL)
 	ignoreWorldStopByAll: 1
 	ignoreFinishStop: 1
 } endState: 
 
 beginState: s32'koroshinoichigeki_atk' {
 	upon: (IMMEDIATE) {
 		overdriveAttack: 
 		callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 		callSubroutine: s32'cmnUltimateShot_AtkInit'
 		damage: 2564
 		minimalDamagePercent: 22
 		atkFrontDirection: 2
 		groundKnockbackEffect: (AIR_STRONG)
 		airKnockbackEffect: (AIR_STRONG)
 		attackXKnockback: 80000
 		attackYKnockback: 18000
 		airHitstun: 50
 		wallBounceCount: 0
 		wallBounceXVelocityPercent: 0
 		slidingKnockdownDuration: 40
 		groundBounceUntechTime: 30000
 		hitstop: 12
 		disableHitstopForSelf: 0
 		autoHitStopSending: 1
 		disableAttackClashing: 1
 		atkMainPlayerHitOnly: 1
-		applyAttackLockWithStateName: s32'CmnNeutral'
 		atkRange: 50000000, 10000, 0, 0
 		attackDamageEffect: 4, s32'cmn_BomHit'
 		damageSoundEffect: s32'ARC_BTL_HTN_Kyusho_Hit'
 		atkShakeCamera: s32'ShakeYoko', 3000, 0, 25, 5
 		ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(1) {
 			damage: 3564
 			minimalDamagePercent: 35
 			hitstop: 15
 			attackXKnockback: 90000
 			atkShakeCamera: s32'ShakeYoko', 4000, 0, 30, 30
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(2) {
 			damage: 4564
 			minimalDamagePercent: 39
 			hitstop: 18
 			attackXKnockback: 100000
 			atkShakeCamera: s32'ShakeYoko', 4500, 0, 35, 30
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(3) {
 			damage: 5064
 			minimalDamagePercent: 43
 			hitstop: 21
 			attackXKnockback: 110000
 			atkShakeCamera: s32'ShakeYoko', 5000, 0, 39, 30
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(4) {
 			damage: 5641
 			minimalDamagePercent: 46
 			hitstop: 24
 			attackXKnockback: 120000
 			atkShakeCamera: s32'ShakeYoko', 5000, 0, 42, 30
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(5) {
 			damage: 5956
 			minimalDamagePercent: 53
 			hitstop: 27
 			attackXKnockback: 130000
 			atkShakeCamera: s32'ShakeYoko', 5000, 0, 45, 30
 		} endIf: 
 		ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(6) {
 			damage: 6564
 			minimalDamagePercent: 56
 			hitstop: 30
 			attackXKnockback: 150000
 			atkShakeCamera: s32'ShakeYoko', 6000, 0, 50, 30
 		} endIf: 
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
 			callSubroutine: s32'cmnUltimate_CameraCombo'
 			ifNot: var(act0) {
 				checkCurrentStateName: (PLAYER), s32'KoroshinoIchigeki'
 				if: var(Result) {
 					sendSignalName: (PLAYER), s32'Koroshi_Hit', 0
 				} endIf: 
 			} endIf: 
 			copyVar: var(CameraCombo1), int(1)
 			ifOperation: (IS_GREATER_OR_EQUAL), var(ObjectParameter1), int(4) {
 				createObject: s32'htn431_Max_Eff', (ENEMY_CENTER)
 			} endIf: 
 		} endUpon: 
 		upon: (IDLING) {
 			if: var(CameraCombo1) {
 				disableHitbox: 
 			} endIf: 
 		} endUpon: 
 		upon: (AFTER_EXIT) {
 			clearRegisteredUponCode: (AFTER_EXIT)
 			copyVar: var(act0), int(1)
 		} endUpon: 
 		linkParticle: s32'htn431_atk01'
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		linkObjectChangeToEliminate: (PARENT)
 		setActionTimeMax: 300
 		autoHitSignalSending: 1
 	} endUpon: 
 	sprite: s32'koroshinoichigeki_atk', 4
 	enableHitbox: 
 	sprite: s32'null', 30
 	ignoreWorldStopByAll: 1
 } endState: 
 
 beginState: s32'htn_431_smoke_group' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		ignoreWorldStopByAll: 1
 		setYPosition: 0
 		addXPosition: 190000
 	} endUpon: 
 	sprite: s32'null', 15
 	createObject: s32'htn_431_smoke01', (ZERO)
 	createObject: s32'htn_431_smoke02', (ZERO)
 	createObject: s32'htn_431_smoke03', (ZERO)
 } endState: 
 
 beginState: s32'htn_431_smoke01' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		ignoreWorldStopByAll: 1
 		ignoreFinishStop: 1
 		setAngleY: 180000
 		setScale: 1250
 		addZPosition: 150000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bg_undersmoke00'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_431_smoke02' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		ignoreWorldStopByAll: 1
 		ignoreFinishStop: 1
 		setAngleY: 180000
 		setScale: 1250
 		addXPosition: 500000
 		addZPosition: 150000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bg_undersmoke00'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn_431_smoke03' {
 	upon: (IMMEDIATE) {
 		linkObjectPosition: (PARENT)
 		ignoreWorldStopByAll: 1
 		ignoreFinishStop: 1
 		setAngleY: 180000
 		setScale: 1250
 		addXPosition: 1000000
 		addZPosition: 150000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bg_undersmoke00'
 	spriteEnd: 
 } endState: 
 
 beginState: s32'htn431_Max_Eff' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PLAYER)
 		addXPosition: -150000
 		setYPosition: 300000
 		setScale: 875
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn431_Max'
 	linkObjectStop: (NULL)
 	ignoreWorldStopByAll: 1
 	ignoreFinishStop: 1
 	spriteEnd: 
 } endState: 
 
 beginState: s32'432cs_atk' {
 	upon: (IMMEDIATE) {
 		overdriveAttack: 
 		callSubroutine: s32'cmnAtkLevel_4_AtkInit'
 		callSubroutine: s32'cmnUltimateShot_AtkInit'
 		callSubroutine: s32'cmnAtkTemplNageExe'
 		damage: 80
 		minimalDamagePercent: 33
 		hitstop: 0
 		groundKnockbackEffect: (ExDamageBallMof)
 		airKnockbackEffect: (ExDamageBallMof)
 		atkFrontDirection: 1
 		atkDamageToRecovery: 1
 		disableHitsparks: 1
 		atkShutUp: 1
 		attackKillType: (CANNOT_KILL)
 		applyAttackLockWithStateName: s32'432cs_atk'
 		linkObjectStop: (NULL)
 		linkObjectChangeToEliminate: (PARENT)
 		setActionTimeMax: 400
 		ignoreWorldStopByFriend: 1
 		upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE) {
 			callSubroutine: s32'cmnUltimate_CameraCombo'
 		} endUpon: 
 		jumpToStateInObject: (SELF), s32'futtobu'
 	} endUpon: 
 	sprite: s32'htn432cs_atk', 28
 	atkDamageCell: 24
 	enableHitbox: 
 	spriteEnd: 
 	sprite: s32'htn432cs_atk', 77
 	atkDamageCell: 26
 	enableHitbox: 
 	spriteEnd: 
 	label: s32'loop'
 	sprite: s32'htn432cs_atk', 3
 	moveToPosType: (ENEMY_MAIN_PLAYER), (CENTER)
 	atkDamageCell: 24
 	enableHitbox: 
 	atkAdditionalHitCount: 1
 	sprite: s32'htn432cs_atk', 3
 	atkDamageCell: 26
 	enableHitbox: 
 	sprite: s32'htn432cs_atk', 3
 	atkDamageCell: 25
 	enableHitbox: 
 	sprite: s32'htn432cs_atk', 3
 	atkDamageCell: 27
 	enableHitbox: 
 	atkAdditionalHitCount: 0
 	spriteEnd: 
 	modifyVar: (ADD), var(act0), int(1)
 	gotoLabelIfOperation: s32'loop', (IS_LESS_OR_EQUAL), var(act0), int(5)
 	sprite: s32'htn432cs_atk', 3
 	moveToPosType: (ENEMY_MAIN_PLAYER), (CENTER)
 	atkDamageCell: 24
 	enableHitbox: 
 	atkAdditionalHitCount: 2
 	sprite: s32'htn432cs_atk', 3
 	atkDamageCell: 26
 	enableHitbox: 
 	sprite: s32'htn432cs_atk', 3
 	atkDamageCell: 25
 	enableHitbox: 
 	sprite: s32'htn432cs_atk', 3
 	atkDamageCell: 27
 	enableHitbox: 
 	atkAdditionalHitCount: 0
 	applyAttackLockWithStateName: s32'SeichowoTsuzuketeiru'
 	spriteEnd: 
 	sprite: s32'null', 2147483647
 	spriteEnd: 
 	label: s32'futtobu'
 	sprite: s32'htn432cs_atk', 3
 	clearRegisteredUponCode: (SIGNAL0)
 	moveToPosType: (ENEMY_MAIN_PLAYER), (CENTER)
 	enableHitbox: 
 	damage: 50
 	minimalDamagePercent: 100
 	groundKnockbackEffect: (AIR_STRONG)
 	airKnockbackEffect: (AIR_STRONG)
 	attackXKnockback: 100000
 	attackYKnockback: 15000
 	wallBounceCount: 1
 	wallBounceXVelocityPercent: 0
 	airHitstun: 60
 	attackFinishBlowback: 1
 	dontAddComboCount: 1
 	atkShutUp: 0
 	disableOpponentRecovery: 0
 	atkAdditionalHitCount: 0
 	attackKillType: (CAN_KILL)
 	atkHoseiTadan: 1
 	callSubroutine: s32'cmn_UltimateHosei'
 	applyAttackLockWithStateName: s32'CmnNeutral'
 } endState: 
 
 beginState: s32'htn_432_E_Light' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 1000
 	linkParticle: s32'htn_432_E_Light'
 } endState: 
 
 beginState: s32'htn_432_aura01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		jumpToStateInObject: (SELF), s32'hide'
 		jumpToStateInObject: (ATTACKER), s32'aura'
 		addZPosition: -40000
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn432_aura01'
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'hide'
 	sprite: s32'null', 200
 	setColorAlpha: 0
 	spriteEnd: 
 	gotoLabel: s32'end'
 	label: s32'aura'
 	sprite: s32'null', 200
 	linkParticle: s32'htn432_aura01'
 	setColorAlpha: 255
 	spriteEnd: 
 	label: s32'end'
 	sprite: s32'null', 40
 	clearRegisteredUponCode: (SIGNAL0)
 	clearRegisteredUponCode: (SIGNAL1)
 	setColorAlpha: 255
 } endState: 
 
 beginState: s32'htn_432_atk01_sub' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn432_aura01_sub'
 } endState: 
 
 beginState: s32'htn_432_atk01_end' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn432_aura01_end'
 } endState: 
 
 beginState: s32'htn_432_atk01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 30
 	linkParticle: s32'htn432_atk_eff01'
 } endState: 
 
 beginState: s32'htn_432cs_hit01_Loop' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn432cs_hit01_Loop'
 } endState: 
 
 beginState: s32'htn_432cs_atk01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn432cs_atk01'
 } endState: 
 
 beginState: s32'htn_432cs_atk02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addYPosition: 100000
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn432cs_atk02'
 } endState: 
 
 beginState: s32'htn_432cs_tokitobasi_bg02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addYPosition: 300000
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'htn432cs_tokibg02'
 } endState: 
 
 beginState: s32'htn_432cs_tokitobasi_bg01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectStop: (PARENT)
 		addYPosition: 300000
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn432cs_tokibg01'
 } endState: 
 
 beginState: s32'htn_432cs_bg_dash01' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 60
 	linkParticle: s32'bg_dash_default'
 } endState: 
 
 beginState: s32'htn_432cs_bg_dash02' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'bg_htn432_smoke01'
 } endState: 
 
 beginState: s32'htn_432_atk03' {
 	upon: (IMMEDIATE) {
 		linkObjectChangeToEliminate: (PARENT)
 		linkObjectPosition: (PARENT)
 		linkObjectStop: (PARENT)
 		addXPosition: 150000
 	} endUpon: 
 	sprite: s32'null', 90
 	linkParticle: s32'htn432_atk03'
 } endState: 
 
 beginState: s32'htn_432_tokitobasi_bg01_end' {
 	upon: (IMMEDIATE) {
 		linkObjectStop: (PARENT)
 		addYPosition: 300000
 		addXPosition: 150000
 	} endUpon: 
 	sprite: s32'null', 200
 	linkParticle: s32'htn432_tokibg01_end'
 	sprite: s32'null', 180
 } endState: 
 
 beginState: s32'FinishCamera_NmlAtk2C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cameraMotion: -150114, 130721, 20293, 4575, 9710, 0, 72, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_NmlAtkAir2C' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cameraMotion: -164349, 355539, 43634, 1920, 5996, 0, 54, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_236CHasei' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	moveCameraEx: (ENEMY_MAIN_PLAYER), -152353, 442351, 109291, 37, 3436, 0, 54, 0, 90, 30, 0, 0, 2
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'FinishCamera_HakkisuruChikara' {
 	upon: (IMMEDIATE) {
 		ignoreWorldStopByFriend: 1
 		ignoreFinishStop: 1
 		resetCamera: 
 		linkCommonPawn: 
 	} endUpon: 
 	sprite: s32'null', 60
 	createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
 	cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
 	cameraShake: s32'ShakeTate', 1500, 3, 25, 3
 	cameraMotion: 270812, 195105, 123067, -637, -10579, 0, 54, 0, 90, 30, 1
 	sprite: s32'null', 30
 	deactivateObjectByName: s32'cmn_syutyusen'
 } endState: 
 
 beginState: s32'DUMMYDUMMY' {
 	sprite: s32'null', 1
 } endState: 
 
